<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Installation &mdash; psiFinder v1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Quality Control" href="Quality%20Control.html" />
    <link rel="prev" title="psiFinder" href="../index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> psiFinder
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#download-psifinder">Download psiFinder</a></li>
<li class="toctree-l2"><a class="reference internal" href="#download-genome-sequence-and-annotation-file">Download genome sequence and annotation file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#download-example-data">Download example data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#install-conda">Install conda</a></li>
<li class="toctree-l2"><a class="reference internal" href="#starting-the-software">Starting the software</a></li>
<li class="toctree-l2"><a class="reference internal" href="#configuration">Configuration</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html">Ψ-sites snoRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20and%20snoRNA%20Interaction.html">Ψ-sites and snoRNA Interaction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Motif%20Searching.html">Ψ-sites Motif Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html">Ψ-sites SNPs and SNVs Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Differential%20Modification.html">Ψ-sites Differential Modification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="Accessory%20Tools.html">Accessory Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">psiFinder</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Installation</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Installation.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this heading"></a></h1>
<p><code class="docutils literal notranslate"><span class="pre">Remember</span></code> – psiFinder are available for <strong>Linux os</strong> as well as for
<strong>Mac os</strong>!</p>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#download-psifinder" id="id2">Download psiFinder</a></p></li>
<li><p><a class="reference internal" href="#download-genome-sequence-and-annotation-file" id="id3">Download genome sequence and annotation file</a></p></li>
<li><p><a class="reference internal" href="#download-example-data" id="id4">Download example data</a></p></li>
<li><p><a class="reference internal" href="#install-conda" id="id5">Install conda</a></p></li>
<li><p><a class="reference internal" href="#starting-the-software" id="id6">Starting the software</a></p></li>
<li><p><a class="reference internal" href="#configuration" id="id7">Configuration</a></p></li>
</ul>
</div>
<section id="download-psifinder">
<h2><a class="toc-backref" href="#id2">Download psiFinder</a><a class="headerlink" href="#download-psifinder" title="Permalink to this heading"></a></h2>
<p><em>psiFinder is free and open to all users.</em></p>
<p>If you’re here for the first time, download psiFinder first, and decompress it into a directory whatever you like.</p>
<p><a class="reference external" href="https://mega.nz/file/wDdwyCZY#KasVu7WPJfKLDpSh_nnGfrBk5ho14QWnToQDHgraqaU">Download psiFinder v.0.1 for Linux x64</a></p>
<p><a class="reference external" href="https://mega.nz/file/ROsBkYiY#IFZ56zYR-3j7dCuz-34UF3r-LU7GZx-TkHdURTJ-5zI">Download psiFinder v.0.1 for Mac OSX</a></p>
<p>Windows support is currently a work-in-progress. Stay tuned and visit back for a Windows compatible executable.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For users who want to remote operation a linux system in windows, the SSH client must be capable of X11-Forwarding, such as <a class="reference external" href="https://mobaxterm.mobatek.net/">MobaXterm</a> et al.</p>
</div>
</section>
<section id="download-genome-sequence-and-annotation-file">
<h2><a class="toc-backref" href="#id3">Download genome sequence and annotation file</a><a class="headerlink" href="#download-genome-sequence-and-annotation-file" title="Permalink to this heading"></a></h2>
<p>Users should download the genome file and annotation file for the species of interest.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Current version of psiFinder mainly support genome data derived from human, more support for other species please contact us for new version development!</p>
</div>
<p><a class="reference external" href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/">Download hg38 genome &lt;fa&gt;</a></p>
<p><a class="reference external" href="https://www.gencodegenes.org/human/">Download hg38 annotation &lt;gtf&gt;</a></p>
</section>
<section id="download-example-data">
<h2><a class="toc-backref" href="#id4">Download example data</a><a class="headerlink" href="#download-example-data" title="Permalink to this heading"></a></h2>
<p>psiFinder provide some example data (human) for the user to test. If want to do so, you should download the example data.</p>
<p><a class="reference external" href="https://mega.nz/fm/wacTDQQK">Download example data</a></p>
</section>
<section id="install-conda">
<h2><a class="toc-backref" href="#id5">Install conda</a><a class="headerlink" href="#install-conda" title="Permalink to this heading"></a></h2>
<p>psiFinder needs to use <code class="docutils literal notranslate"><span class="pre">conda</span></code> to configure the environment, please install conda first:</p>
<ul class="simple">
<li><p><em>Linux</em></p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget -c https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
chmod <span class="m">777</span> Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh
<span class="nb">source</span> ~/.bashrc
</pre></div>
</div>
<ul class="simple">
<li><p><em>Mac OS</em></p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
chmod <span class="m">777</span> Miniconda3-latest-MacOSX-x86_64.sh
sh Miniconda3-latest-MacOSX-x86_64.sh
<span class="nb">source</span> ~/.bash_profile
</pre></div>
</div>
</section>
<section id="starting-the-software">
<h2><a class="toc-backref" href="#id6">Starting the software</a><a class="headerlink" href="#starting-the-software" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><p><em>Linux</em></p></li>
</ul>
<p>Once the psiFinder mian program and the Genome annotation are downloaded and unzipped properly, go to the directory of psiFinder:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/unzipped_psiFinder
$ tree -L <span class="m">1</span>
.
├── AppRun -&gt; psiFinder
├── CPAT-3.0.0
├── genome
├── default.desktop
├── psiFinder
├── psiFinder.sh
├── lib
├── libexec
├── libXss.so.1
├── pack.sh
├── plugins
├── qss
├── qt.conf
├── resources
├── snakemake
└── translations

<span class="m">9</span> directories, <span class="m">7</span> files
</pre></div>
</div>
<p>The operation of psiFinder needs to grant executable permission to the script of the software. Please execute the following code before running the software:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ chmod <span class="m">777</span> ./snakemake/script/*
$ chmod <span class="m">777</span> ./psiFinder
</pre></div>
</div>
<p>Then, upon executed the ./psiFinder, the following window appears:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ ./psiFinder.sh
</pre></div>
</div>
<img alt="../_images/psiFinder_window.png" src="../_images/psiFinder_window.png" />
<ul class="simple">
<li><p><em>Mac OS</em></p></li>
</ul>
<p>For Mac users, after decompressing the file, open the <code class="docutils literal notranslate"><span class="pre">Command</span> <span class="pre">Line</span></code> file, cd to the <code class="docutils literal notranslate"><span class="pre">psiFinder.app</span> <span class="pre">directory</span></code>, and input <code class="docutils literal notranslate"><span class="pre">./psiFinder.app/Content/Macos/psiFinder</span></code> at Command file to run.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For Mac, you should place the downloaded genome file and/or the example data in the same directory of <code class="docutils literal notranslate"><span class="pre">psiFinder</span></code> flie mentioned above. (like <code class="docutils literal notranslate"><span class="pre">psiFinder/genome</span></code> and/or <code class="docutils literal notranslate"><span class="pre">psiFinder/data</span></code>)</p>
</div>
</section>
<section id="configuration">
<h2><a class="toc-backref" href="#id7">Configuration</a><a class="headerlink" href="#configuration" title="Permalink to this heading"></a></h2>
<p>Before uploading data, you should first configure the operating environment of the software, this step can be achieved through clicking the option <code class="docutils literal notranslate"><span class="pre">global-&gt;configuration</span></code> at the upper of the main window.</p>
<img alt="../_images/configuration.png" src="../_images/configuration.png" />
<p>Once click <code class="docutils literal notranslate"><span class="pre">CHECK</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">0_configuration.sh</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">0_configuration.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>

Help<span class="o">()</span> <span class="o">{</span>
<span class="nb">echo</span>
<span class="s2">&quot;0_software_test.sh example:</span>
<span class="s2"> bash 0_software_test.sh</span>
<span class="s2">&quot;</span>
<span class="o">}</span>


usage<span class="o">()</span> <span class="o">{</span>                                      <span class="c1"># Function: Print a help message.</span>
  <span class="nb">echo</span> <span class="s2">&quot;Usage: </span><span class="nv">$0</span><span class="s2"> [ -h help]&quot;</span> <span class="m">1</span>&gt;<span class="p">&amp;</span><span class="m">2</span>
<span class="o">}</span>
exit_abnormal<span class="o">()</span> <span class="o">{</span>                              <span class="c1"># Function: Exit with error.</span>
  usage
  <span class="nb">exit</span> <span class="m">1</span>
<span class="o">}</span>


<span class="c1">#psiFinder requires pre-installation of mupdf, wkhtmltopdf, conda, perl, R (optparse, dplyr, tidyr, tidyverse, ggrepel, stringr, ggplot2, RColorBrewer, caTools, cowplot, pROC, e1071,</span>
<span class="c1"># neuralnet, venndiagram, reshape2, limma, edgeR, gprofiler2, ggrepel, pathview, bedr, scales, motifStack), snakemake, seqtk, cutadapt, STAR, samtools, rtsSeeker, bedAnnotator, ACAscan,</span>
<span class="c1"># miranda, homer, gtftogenepred, bedtools, minion, please click &#39;CHECK&#39; button to check if it is already installed; if not installed, required softwares will be installed or set into environment</span>
<span class="c1"># variables by conda.</span>

<span class="k">while</span> <span class="nb">getopts</span> <span class="s2">&quot;:h:&quot;</span> options<span class="p">;</span> <span class="k">do</span>

  <span class="k">case</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">options</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">in</span>
    h<span class="p">|</span>:<span class="o">)</span>
      usage
      Help
      <span class="nb">exit</span> <span class="m">0</span>
      <span class="p">;;</span>
    <span class="se">\?</span><span class="o">)</span> <span class="c1"># incorrect option</span>
      <span class="nb">echo</span> <span class="s2">&quot;Error: -</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span><span class="s2"> Invalid option&quot;</span>
      exit_abnormal
      <span class="p">;;</span>
  <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">shift</span> <span class="k">$((</span><span class="nv">$OPTIND</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span>
date  <span class="c1">## echo the date at start</span>
<span class="nb">echo</span> <span class="s2">&quot;Starting: software checking...&quot;</span>

<span class="c1">#install conda</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which conda<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;conda is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nb">eval</span> <span class="s2">&quot;cd pipline_script/&quot;</span>
<span class="nv">conda_setup</span><span class="o">=</span><span class="s2">&quot;bash conda_setup.sh&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_setup</span>
<span class="nb">wait</span>
<span class="nb">eval</span> <span class="s2">&quot;cd ../&quot;</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;conda(miniconda3) is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install perl</span>
<span class="k">if</span> <span class="o">[</span> <span class="sb">`</span>which perl <span class="p">|</span> grep -c <span class="s2">&quot;miniconda3/bin/perl&quot;</span><span class="sb">`</span> <span class="o">==</span> <span class="m">0</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;perl is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_perl</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c anaconda perl&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_perl</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;perl(miniconda3) is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install R</span>
<span class="k">if</span> <span class="o">[</span> <span class="sb">`</span>which R <span class="p">|</span> grep -c <span class="s2">&quot;miniconda3/bin/R&quot;</span><span class="sb">`</span> <span class="o">==</span> <span class="m">0</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;R is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_R</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c r r-base&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_R</span>
<span class="nb">wait</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;R(miniconda3) is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install optparse R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing optparse...&quot;</span>
conda install --yes -c bioconda r-optparse
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install openxlsx R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing openxlsx...&quot;</span>
conda install --yes -c bioconda r-openxlsx
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install dplyr R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing dplyr...&quot;</span>
conda install --yes -c r r-dplyr
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install tidyr R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing tidyr...&quot;</span>
conda install --yes -c r r-tidyr
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install tidyverse R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing tidyverse...&quot;</span>
conda install --yes -c r r-tidyverse
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install stringr R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing stringr...&quot;</span>
conda install --yes -c r r-stringr
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install ggplot2 R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing ggplot2...&quot;</span>
conda install --yes -c r r-ggplot2
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install ggpol R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing ggpol...&quot;</span>
conda install --yes -c conda-forge r-ggpol
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install ggpubr R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing ggpubr...&quot;</span>
conda install --yes -c conda-forge r-ggpubr
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install RColorBrewer R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing RColorBrewer...&quot;</span>
conda install --yes -c r r-rcolorbrewer
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install cowplot R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing cowplot...&quot;</span>
conda install --yes -c conda-forge r-cowplot
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install gridextra R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing gridextra...&quot;</span>
conda install --yes -c r r-gridextra
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install pROC R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing pROC...&quot;</span>
conda install --yes -c r r-proc
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install e1071 R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing e1071...&quot;</span>
conda install --yes -c r r-e1071
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install neuralnet R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing neuralnet...&quot;</span>
conda install --yes -c conda-forge r-neuralnet
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install caTools R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing caTools...&quot;</span>
conda install --yes -c r r-catools
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install factoextra R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing factoextra...&quot;</span>
conda install --yes -c conda-forge r-factoextra
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install venndiagram R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing venndiagram...&quot;</span>
conda install --yes -c bioconda r-venndiagram
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install reshape2 R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing reshape2...&quot;</span>
conda install --yes -c r r-reshape2
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install limma R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing limma...&quot;</span>
conda install --yes -c bioconda bioconductor-limma
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install edger R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing edger...&quot;</span>
conda install --yes -c bioconda bioconductor-edger
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install gprofiler2 R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing gprofiler2...&quot;</span>
conda install --yes -c conda-forge r-gprofiler2
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install ggrepel R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing ggrepel...&quot;</span>
conda install --yes -c bioconda r-ggrepel
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install pathview R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing pathview...&quot;</span>
conda install --yes -c bioconda bioconductor-pathview
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install bedr R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing bedr...&quot;</span>
conda install --yes -c bioconda r-bedr
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install scales R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing scales...&quot;</span>
conda install --yes -c r r-scales
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install motifstack R package</span>
<span class="nb">echo</span> <span class="s2">&quot;installing motifstack...&quot;</span>
conda install --yes -c bioconda bioconductor-motifstack
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install snakemake</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which snakemake<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;snakemake is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_snakemake</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda snakemake&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_snakemake</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;snakemake is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install mupdf-x11</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which  mupdf-x11<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;mupdf is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_mupdf</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c conda-forge mupdf&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_mupdf</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;mupdf is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install wkhtmltopdf</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which  wkhtmltopdf<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;wkhtmltopdf is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_wkhtmltopdf</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c conda-forge wkhtmltopdf&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_wkhtmltopdf</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;wkhtmltopdf is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install fastx_collapser</span>
<span class="c1"># if [ -z  `which  fastx_collapser` ]</span>
<span class="c1"># then</span>
<span class="c1"># echo &quot;fastx_toolkit is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="c1"># conda_fastx_toolkit=&quot;conda install --yes -c bioconda fastx_toolkit&quot;</span>
<span class="c1"># eval $conda_fastx_toolkit</span>
<span class="c1"># else</span>
<span class="c1"># echo &quot;fastx_toolkit is installed.&quot;</span>
<span class="c1"># fi</span>

<span class="c1">#install cutadapt</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which cutadapt<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;cutadapt is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_cutadapt</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda cutadapt&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_cutadapt</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;cutadapt is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install seqtk</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which seqtk<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;seqtk is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_seqtk</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda seqtk&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_seqtk</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;seqtk is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install STAR</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which STAR<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;STAR is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_STAR</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda star&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_STAR</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;STAR is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install samtools</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which samtools<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;samtools is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_samtools</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda samtools&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_samtools</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;samtools is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install gtfToGenePred</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which gtfToGenePred<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;gtfToGenePred is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_gtfToGenePred</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda ucsc-gtftogenepred&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_gtfToGenePred</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;gtfToGenePred is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install bedtools</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which bedtools<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;bedtools is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_bedtools</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda bedtools&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_bedtools</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;bedtools is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#locate rtsSeeker</span>
<span class="k">if</span> <span class="o">[</span> <span class="sb">`</span><span class="nb">test</span> -e /script/rtsSeeker<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;rtsSeeker is not exist, please check if script/rtsSeeker exist in your current directory (directory with psiFinder) or re-download the .zip package and unzip it...&quot;</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;rtsSeeker exist (Note: if you copy the psiFinder to other directory, make sure you copy the rtsSeeker/ directory with it as well).&quot;</span>
<span class="k">fi</span>

<span class="c1">#locate bedAnnotator</span>
<span class="k">if</span> <span class="o">[</span> <span class="sb">`</span><span class="nb">test</span> -e /script/bedAnnotator<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;bedAnnotator is not exist, please check if script/bedAnnotator exist in your current directory (directory with psiFinder) or re-download the .zip package and unzip it...&quot;</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;bedAnnotator exist (Note: if you copy the psiFinder to other directory, make sure you copy the annotation/ directory with it as well).&quot;</span>
<span class="k">fi</span>

<span class="c1">#locate ACAscan</span>
<span class="k">if</span> <span class="o">[</span> <span class="sb">`</span><span class="nb">test</span> -e /script/ACAscan<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;ACAscan is not exist, please check if annotation/ACAscan exist in your current directory (directory with psiFinder) or re-download the .zip package and unzip it...&quot;</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;ACAscan exist (Note: if you copy the psiFinder to other directory, make sure you copy the annotation/ directory with it as well).&quot;</span>
<span class="k">fi</span>

<span class="c1">#install viennarna</span>
<span class="nb">echo</span> <span class="s2">&quot;installing viennarna...&quot;</span>
conda install --yes -c bioconda viennarna
<span class="nb">echo</span> -e <span class="s2">&quot;\n&quot;</span>

<span class="c1">#install miranda</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which miranda<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;miranda is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_miranda</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda miranda&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_miranda</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;miranda is installed.&quot;</span>
<span class="k">fi</span>

<span class="c1">#install homer</span>
<span class="k">if</span> <span class="o">[</span> -z  <span class="sb">`</span>which homer<span class="sb">`</span> <span class="o">]</span>
<span class="k">then</span>
<span class="nb">echo</span> <span class="s2">&quot;homer is not installed, or not in environment variables. Configuration check will add it to environment variables...&quot;</span>
<span class="nv">conda_homer</span><span class="o">=</span><span class="s2">&quot;conda install --yes -c bioconda homer&quot;</span>
<span class="nb">eval</span> <span class="nv">$conda_homer</span>
<span class="k">else</span>
<span class="nb">echo</span> <span class="s2">&quot;homer is installed.&quot;</span>
<span class="k">fi</span>

<span class="nb">echo</span> <span class="s2">&quot;Finished: All software checking is done!&quot;</span>
<span class="nb">exit</span> <span class="m">0</span>  <span class="c1"># Exit normally.</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This step will automatically install some software through <a class="reference external" href="https://docs.conda.io/en/latest/">conda</a>.</p>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For support or questions please make a post on <a class="reference external" href="http://biostars.org">Biostars</a>. For feature requests or bug reports please open an issue on <a class="reference external" href="https://github.com/worsteggs/psiFinder_readthedocs/issues">github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../index.html" class="btn btn-neutral float-left" title="psiFinder" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Quality%20Control.html" class="btn btn-neutral float-right" title="Quality Control" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, chenzhr23.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>