
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>snakemake pipline &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/haiku.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Accessory Tools" href="Accessory%20Tools.html" />
    <link rel="prev" title="Ψ-sites Functional Enrichment" href="%CE%A8-sites%20Functional%20Enrichment.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="../index.html">
          <span>psiFinder v1.0 documentation</span></a></h1>
        <h2 class="heading"><span>snakemake pipline</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        «&#160;&#160;<a href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="Accessory%20Tools.html">Accessory Tools</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <section id="snakemake-pipline">
<h1>snakemake pipline<a class="headerlink" href="#snakemake-pipline" title="Permalink to this heading">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p>
<ul>
<li><p><a class="reference internal" href="#select-genome" id="id2">Select Genome</a></p></li>
<li><p><a class="reference internal" href="#data-upload" id="id3">Data upload</a></p></li>
<li><p><a class="reference internal" href="#basic-analyses" id="id4">Basic analyses</a></p></li>
<li><p><a class="reference internal" href="#mode-options" id="id5">Mode options</a></p></li>
<li><p><a class="reference internal" href="#filter-threshold" id="id6">Filter threshold</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#snakemake-rules" id="id7">snakemake rules</a></p></li>
<li><p><a class="reference internal" href="#output" id="id8">Output</a></p></li>
</ul>
</div>
<p>For a quick start of Ψ-sites data analyses, we developed a snakemake pipline and designed corresponding QT widget for easy running.</p>
<img alt="../_images/quickstart_snakemake.png" src="../_images/quickstart_snakemake.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading">¶</a></h2>
<section id="select-genome">
<h3><a class="toc-backref" href="#id2">Select Genome</a><a class="headerlink" href="#select-genome" title="Permalink to this heading">¶</a></h3>
<p>Current version of psiFinder snakemake pipline mainly support <span class="red">Group: Mammal-&gt;Genome: Homo_sapiens-&gt;Assembly: hg38</span>, more support for other species will coming soon.</p>
</section>
<section id="data-upload">
<h3><a class="toc-backref" href="#id3">Data upload</a><a class="headerlink" href="#data-upload" title="Permalink to this heading">¶</a></h3>
<p>Users should <code class="docutils literal notranslate"><span class="pre">browse</span></code> and locate <cite>fastq|fasta fastq.gz/fasta.gz fq/fa fq.gz/fa.gz</cite> file for both CMC-input and CMC-treated samples.</p>
<ul class="simple">
<li><p>If <code class="docutils literal notranslate"><span class="pre">Barcode</span> <span class="pre">removal</span></code> is checked, it will enable 5’/3’ barcode setting, remove <code class="docutils literal notranslate"><span class="pre">LENGTH</span></code> (<a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/guide.html">cutadapt</a>) bases from reads.</p></li>
<li><p>If <code class="docutils literal notranslate"><span class="pre">Adapter</span> <span class="pre">removal</span></code> is checked, it will enable 5’/3’ adapter setting, sequence of an adapter ligated to the 5’/3’ end will be cutadapt.</p></li>
</ul>
</section>
<section id="basic-analyses">
<h3><a class="toc-backref" href="#id4">Basic analyses</a><a class="headerlink" href="#basic-analyses" title="Permalink to this heading">¶</a></h3>
<p>Current version of psiFinder snakemake pipline provides four basic analysis module of Ψ-sites: (1) identification; (2) annotation; (3) distribution; (4) snoRNA-guided target.</p>
</section>
<section id="mode-options">
<h3><a class="toc-backref" href="#id5">Mode options</a><a class="headerlink" href="#mode-options" title="Permalink to this heading">¶</a></h3>
<p>Different mode options are provided for selection on SVM/ANN/User-defined approaches.</p>
</section>
<section id="filter-threshold">
<h3><a class="toc-backref" href="#id6">Filter threshold</a><a class="headerlink" href="#filter-threshold" title="Permalink to this heading">¶</a></h3>
<p>If User-defined <code class="docutils literal notranslate"><span class="pre">Mode</span> <span class="pre">options</span></code> is checked, then <code class="docutils literal notranslate"><span class="pre">Filter</span> <span class="pre">thresholds</span></code> is enable to customize the threshold of the six key metrics.</p>
</section>
</section>
<section id="snakemake-rules">
<h2><a class="toc-backref" href="#id7">snakemake rules</a><a class="headerlink" href="#snakemake-rules" title="Permalink to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will firstly run <code class="docutils literal notranslate"><span class="pre">cp_unzip.sh</span></code> to unzip and redirect the input sequencing reads files. Then, <code class="docutils literal notranslate"><span class="pre">psiFinder.sh</span></code> will be secondly run to apply snakemake rules selected by <code class="docutils literal notranslate"><span class="pre">quickstart_config.yml</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">cp_unzip.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -ne <span class="m">5</span> <span class="o">]]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; input_sample input_path treat_sample treat_path log_file&#39;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="k">fi</span>

<span class="nv">input_sample</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">input_path</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">treat_sample</span><span class="o">=</span><span class="nv">$3</span>
<span class="nv">treat_path</span><span class="o">=</span><span class="nv">$4</span>
<span class="nv">log_file</span><span class="o">=</span><span class="nv">$5</span>


<span class="nb">echo</span> -e <span class="s2">&quot;clear psiFinder.sh history&quot;</span>
<span class="nb">echo</span> -e <span class="s2">&quot;Your quickstart job is starting, please wait...&quot;</span> &gt; <span class="nv">$log_file</span>

<span class="c1">#.fastq.gz</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$input_sample</span>  &gt; <span class="nv">$input_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$treat_sample</span>  &gt; <span class="nv">$treat_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fasta.gz</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$input_sample</span>  &gt; <span class="nv">$input_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$input_path</span> &gt; <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$treat_sample</span>  &gt; <span class="nv">$treat_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$treat_path</span> &gt; <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fq.gz</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fq.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$input_sample</span>  &gt; <span class="nv">$input_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fq.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$treat_sample</span>  &gt; <span class="nv">$treat_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fa.gz</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fa.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$input_sample</span>  &gt; <span class="nv">$input_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$input_path</span> &gt; <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fa.gz <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    gunzip -c <span class="nv">$treat_sample</span>  &gt; <span class="nv">$treat_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$treat_path</span> &gt; <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fastq</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$input_sample</span> <span class="nv">$input_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$treat_sample</span> <span class="nv">$treat_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fasta</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$input_sample</span> <span class="nv">$input_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$input_path</span> &gt; <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$treat_sample</span> <span class="nv">$treat_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$treat_path</span> &gt; <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fq</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fq <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$input_sample</span> <span class="nv">$input_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fq <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fastq file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$treat_sample</span> <span class="nv">$treat_path</span> <span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fa</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fa <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$input_sample</span> <span class="nv">$input_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$input_path</span> &gt; <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span> <span class="o">==</span> *.fa <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="k">$(</span>tput setaf <span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fasta file...\n</span><span class="k">$(</span>tput sgr <span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    cp -R <span class="nv">$treat_sample</span> <span class="nv">$treat_path</span>
    seqtk seq -F <span class="s1">&#39;#&#39;</span> <span class="nv">$treat_path</span> &gt; <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
    rm <span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="nb">wait</span>


<span class="nb">echo</span> -e <span class="s2">&quot;Starting snakemake workflow...&quot;</span>
<span class="c1">#snakemake</span>
<span class="c1">#fastq</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="k">$(</span>basename <span class="si">${</span><span class="nv">input_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq <span class="o">]]</span> <span class="o">&amp;&amp;</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="k">$(</span>basename <span class="si">${</span><span class="nv">treat_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span> <span class="o">==</span> *.fastq <span class="o">]]</span><span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;check info: Input is fastq files&quot;</span>  &gt;&gt; <span class="nv">$log_file</span>
    nohup bash snakemake/psiFinder.sh -i <span class="k">$(</span>basename <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fastq</span><span class="si">}</span><span class="k">)</span> -t <span class="k">$(</span>basename <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fastq</span><span class="si">}</span><span class="k">)</span> -c quickstart_config.yml &gt;&gt; <span class="nv">$log_file</span> <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
    <span class="nb">wait</span>
<span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;check info: Input is not fastq files&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
<span class="k">fi</span>

<span class="c1">#fasta</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="k">$(</span>basename <span class="si">${</span><span class="nv">input_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta <span class="o">]]</span> <span class="o">&amp;&amp;</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="k">$(</span>basename <span class="si">${</span><span class="nv">treat_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span> <span class="o">==</span> *.fasta <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;check info: Input is fasta files&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
    nohup bash snakemake/psiFinder.sh -i <span class="k">$(</span>basename <span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span><span class="k">)</span> -t <span class="k">$(</span>basename <span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span><span class="k">)</span> -c quickstart_config.yml &gt;&gt; <span class="nv">$log_file</span> <span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">&amp;</span>
    <span class="nb">wait</span>
<span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;check info: Input is not fasta files&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
<span class="k">fi</span>

<span class="nb">echo</span> -e <span class="s2">&quot;Finished snakemake workflow!&quot;</span> &gt;&gt; <span class="nv">$log_file</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">psiFinder.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>

usage<span class="o">()</span> <span class="o">{</span>                                      <span class="c1"># Function: Print a help message.</span>
  <span class="nb">echo</span> <span class="s2">&quot;Usage: </span><span class="nv">$0</span><span class="s2"> [ -h help] [ -i input ] [ -t treat ] [ -c quickstart_config.yml ]&quot;</span> <span class="m">1</span>&gt;<span class="p">&amp;</span><span class="m">2</span>
<span class="o">}</span>
exit_abnormal<span class="o">()</span> <span class="o">{</span>                              <span class="c1"># Function: Exit with error.</span>
  usage
  <span class="nb">exit</span> <span class="m">1</span>
<span class="o">}</span>


<span class="k">while</span> <span class="nb">getopts</span> <span class="s2">&quot;:h:i:t:c:&quot;</span> options<span class="p">;</span> <span class="k">do</span>

  <span class="k">case</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">options</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">in</span>
    h<span class="p">|</span>:<span class="o">)</span>
      usage
      Help
      <span class="nb">exit</span> <span class="m">0</span>
      <span class="p">;;</span>
    i<span class="o">)</span>
      <span class="nv">input</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$input</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t set the input sample&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    t<span class="o">)</span>
      <span class="nv">treat</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$treat</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t set the treat sample&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    c<span class="o">)</span>
      <span class="nv">config</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$config</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t set the config yml file&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    <span class="se">\?</span><span class="o">)</span> <span class="c1"># incorrect option</span>
      <span class="nb">echo</span> <span class="s2">&quot;Error: -</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span><span class="s2"> Invalid option&quot;</span>
      exit_abnormal
      <span class="p">;;</span>
  <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">shift</span> <span class="k">$((</span><span class="nv">$OPTIND</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span>

<span class="nv">quickstart_config</span><span class="o">=</span><span class="k">$(</span>cat <span class="nv">$config</span><span class="k">)</span>
<span class="nv">quickstart_array</span><span class="o">=(</span><span class="k">$(</span><span class="nb">echo</span> <span class="nv">$quickstart_config</span> <span class="p">|</span> tr <span class="s2">&quot;:&quot;</span> <span class="s2">&quot;\n&quot;</span><span class="k">)</span><span class="o">)</span>

<span class="nb">echo</span> -e <span class="s2">&quot;==Starting a quickstart job...==:\n&quot;</span>

<span class="c1">#cutadapt-se</span>
<span class="nb">echo</span> -e <span class="s2">&quot;ad_remove_input:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[41]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[41]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting input cutadapt fastq...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/cutadapt/input/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>.fastq
<span class="k">else</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Moving input cutadapt fastq...\n&quot;</span>
    mv snakemake/reads/input/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>.fastq snakemake/cutadapt/input
<span class="k">fi</span>

<span class="nb">echo</span> -e <span class="s2">&quot;ad_remove_treat:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[43]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[43]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting treat cutadapt fastq...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/cutadapt/treat/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.fastq
<span class="k">else</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Moving treat cutadapt fastq...\n&quot;</span>
    mv snakemake/reads/treat/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.fastq snakemake/cutadapt/treat
<span class="k">fi</span>

<span class="c1">#STAR-index</span>
<span class="nv">temp</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span> -e <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[5]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="p">|</span> sed <span class="s2">&quot;s/\&quot;//g&quot;</span>  <span class="k">)</span>
<span class="nv">DIRECTORY</span><span class="o">=</span><span class="s2">&quot;snakemake/genome/</span><span class="si">${</span><span class="nv">temp</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[</span> -d <span class="s2">&quot;</span><span class="nv">$DIRECTORY</span><span class="s2">&quot;</span> <span class="o">]</span><span class="p">;</span> <span class="k">then</span>
  <span class="nb">echo</span> <span class="s2">&quot;</span><span class="nv">$DIRECTORY</span><span class="s2"> is not empty, STAR index have been built&quot;</span>

<span class="k">else</span>
  <span class="nb">echo</span> -e <span class="s2">&quot;No STAR index in </span><span class="nv">$DIRECTORY</span><span class="s2">, generating STAR index...\n&quot;</span>
  snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/genome/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">temp</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">fi</span>

<span class="c1">#rtsSeeker result</span>
<span class="nb">echo</span> -e <span class="s2">&quot;sites_identification:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[31]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[31]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting rtsSeeker result for Ψ sites identification...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/sites_identification/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker ann</span>
<span class="nb">echo</span> -e <span class="s2">&quot;ann:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[49]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[49]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting ann result for Ψ sites identification...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/ann/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_ann_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker svm</span>
<span class="nb">echo</span> -e <span class="s2">&quot;svm:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[45]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[45]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting svm result for Ψ sites identification...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/svm/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_svm_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker user-defined</span>
<span class="nb">echo</span> -e <span class="s2">&quot;user-defined:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[51]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[51]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting user-defined result for Ψ sites identification...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/user_defined/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_user_defined_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#bedAnnotator</span>
<span class="nb">echo</span> -e <span class="s2">&quot;sites_annotation:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[33]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[33]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting bedAnnotator result for Ψ sites annotation...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/sites_annotation/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_add_seq_group_uniq.bed
<span class="k">fi</span>

<span class="c1">#metagene</span>
<span class="nb">echo</span> -e <span class="s2">&quot;metagene:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[47]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[47]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting metagene result for Ψ sites identification...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/meta_gene/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_metagene_pseudoU.pdf
<span class="k">fi</span>

<span class="c1">#ACAscan</span>
<span class="nb">echo</span> -e <span class="s2">&quot;sites_target_prediction:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[35]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span> <span class="o">[[</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[35]</span><span class="si">}</span><span class="s2">&quot;</span> <span class="o">=</span>~ <span class="s2">&quot;true&quot;</span>  <span class="o">]]</span> <span class="p">;</span>
<span class="k">then</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;Getting ACAscan result for Ψ sites target prediction...\n&quot;</span>
    snakemake -s snakemake/Snakefile --cores <span class="m">8</span> snakemake/output/sites_target_prediction/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_fa.out
<span class="k">fi</span>

<span class="c1">#emit done signal</span>
<span class="nb">echo</span> -e <span class="s2">&quot;==All done!==&quot;</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">quickstart_config.yml</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>group: <span class="s2">&quot;Mammal&quot;</span>
genome: <span class="s2">&quot;Homo_sapiens&quot;</span>
assembly: <span class="s2">&quot;hg38&quot;</span>
se_input: <span class="s2">&quot;/public/home/chenzr/PSI_Seq_brainCell/A1-A12-totalRNA-result/A1.cutadapt.extendedFrags.collapse.cutBarcodes.fa.gz&quot;</span>
se_treat: <span class="s2">&quot;/public/home/chenzr/PSI_Seq_brainCell/A1-A12-totalRNA-result/A2.cutadapt.extendedFrags.collapse.cutBarcodes.fa.gz&quot;</span>
se_input_prefix: <span class="s2">&quot;A1&quot;</span>
se_treat_prefix: <span class="s2">&quot;A2&quot;</span>
se_br5_input: <span class="s2">&quot;0&quot;</span>
se_br3_input: <span class="s2">&quot;0&quot;</span>
se_ad5_input: <span class="s2">&quot;&quot;</span>
se_ad3_input: <span class="s2">&quot;&quot;</span>
se_br5_treat: <span class="s2">&quot;0&quot;</span>
se_br3_treat: <span class="s2">&quot;0&quot;</span>
se_ad5_treat: <span class="s2">&quot;&quot;</span>
se_ad3_treat: <span class="s2">&quot;&quot;</span>
sites_identification: <span class="s2">&quot;true&quot;</span>
sites_annotation: <span class="s2">&quot;true&quot;</span>
sites_target_prediction: <span class="s2">&quot;true&quot;</span>
br_remove_input: <span class="s2">&quot;false&quot;</span>
br_remove_treat: <span class="s2">&quot;false&quot;</span>
ad_remove_input: <span class="s2">&quot;false&quot;</span>
ad_remove_treat: <span class="s2">&quot;false&quot;</span>
svm: <span class="s2">&quot;true&quot;</span>
metagene: <span class="s2">&quot;false&quot;</span>
ann: <span class="s2">&quot;false&quot;</span>
user_defined: <span class="s2">&quot;false&quot;</span>
treatpre_Fold_thres: <span class="s2">&quot;0&quot;</span>
preFold_FC_thres: <span class="s2">&quot;0&quot;</span>
treataft_Fold_thres: <span class="s2">&quot;0&quot;</span>
aftFold_FC_thres: <span class="s2">&quot;0&quot;</span>
treatstoprate_thres: <span class="s2">&quot;0&quot;</span>
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id8">Output</a><a class="headerlink" href="#output" title="Permalink to this heading">¶</a></h2>
<p>snakemake result are output to specific directory where psiFinder package is unziped.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/psiFinder/snakemake/
$ tree -L <span class="m">1</span>
.
├── cp_unzip.sh
├── genome
├── logs
├── output
├── psiFinder.sh
├── reads
├── script
├── Snakefile
├── star
└── cutadapt

<span class="m">7</span> directories, <span class="m">3</span> files

$ <span class="nb">cd</span> /the/directory/of/psiFinder/snakemake/output
$ tree -L <span class="m">1</span>
.
├── ann
├── meta_gene
├── sites_annotation
├── sites_identification
├── sites_target_prediction
├── svm
└── user_defined

<span class="m">7</span> directories, <span class="m">0</span> files
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_quickstart_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        «&#160;&#160;<a href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="Accessory%20Tools.html">Accessory Tools</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 5.1.1.
    </div>
  </body>
</html>