
<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>snakemake pipline &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d75fae25" />
    <link rel="stylesheet" type="text/css" href="../_static/bizstyle.css?v=5283bb3d" />
    
    <script src="../_static/documentation_options.js?v=e160b93e"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Accessory Tools" href="Accessory%20Tools.html" />
    <link rel="prev" title="Ψ-sites Functional Enrichment" href="%CE%A8-sites%20Functional%20Enrichment.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="Accessory%20Tools.html" title="Accessory Tools"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Functional%20Enrichment.html" title="Ψ-sites Functional Enrichment"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">snakemake pipline</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="snakemake-pipline">
<h1>snakemake pipline<a class="headerlink" href="#snakemake-pipline" title="Link to this heading">¶</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p>
<ul>
<li><p><a class="reference internal" href="#select-genome" id="id2">Select Genome</a></p></li>
<li><p><a class="reference internal" href="#data-upload" id="id3">Data upload</a></p></li>
<li><p><a class="reference internal" href="#basic-analyses" id="id4">Basic analyses</a></p></li>
<li><p><a class="reference internal" href="#mode-options" id="id5">Mode options</a></p></li>
<li><p><a class="reference internal" href="#filter-threshold" id="id6">Filter threshold</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#snakemake-rules" id="id7">snakemake rules</a></p></li>
<li><p><a class="reference internal" href="#output" id="id8">Output</a></p></li>
</ul>
</nav>
<p>For a quick start of Ψ-sites data analyses, we developed a snakemake pipline and designed corresponding QT widget for easy running.</p>
<img alt="../_images/quickstart_snakemake.png" src="../_images/quickstart_snakemake.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Input</a><a class="headerlink" href="#input" title="Link to this heading">¶</a></h2>
<section id="select-genome">
<h3><a class="toc-backref" href="#id2" role="doc-backlink">Select Genome</a><a class="headerlink" href="#select-genome" title="Link to this heading">¶</a></h3>
<p>Current version of psiFinder snakemake pipline mainly support <span class="red">Group: Mammal-&gt;Genome: Homo_sapiens-&gt;Assembly: hg38</span>, more support for other species will coming soon.</p>
</section>
<section id="data-upload">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">Data upload</a><a class="headerlink" href="#data-upload" title="Link to this heading">¶</a></h3>
<p>Users should <code class="docutils literal notranslate"><span class="pre">browse</span></code> and locate <cite>fastq|fasta fastq.gz/fasta.gz fq/fa fq.gz/fa.gz</cite> file for both CMC-input and CMC-treated samples.</p>
<ul class="simple">
<li><p>If <code class="docutils literal notranslate"><span class="pre">Barcode</span> <span class="pre">removal</span></code> is checked, it will enable 5’/3’ barcode setting, remove <code class="docutils literal notranslate"><span class="pre">LENGTH</span></code> (<a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/guide.html">cutadapt</a>) bases from reads.</p></li>
<li><p>If <code class="docutils literal notranslate"><span class="pre">Adapter</span> <span class="pre">removal</span></code> is checked, it will enable 5’/3’ adapter setting, sequence of an adapter ligated to the 5’/3’ end will be cutadapt.</p></li>
</ul>
</section>
<section id="basic-analyses">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">Basic analyses</a><a class="headerlink" href="#basic-analyses" title="Link to this heading">¶</a></h3>
<p>Current version of psiFinder snakemake pipline provides four basic analysis module of Ψ-sites: (1) identification; (2) annotation; (3) distribution; (4) snoRNA-guided target.</p>
</section>
<section id="mode-options">
<h3><a class="toc-backref" href="#id5" role="doc-backlink">Mode options</a><a class="headerlink" href="#mode-options" title="Link to this heading">¶</a></h3>
<p>Different mode options are provided for selection on SVM/ANN/User-defined approaches.</p>
</section>
<section id="filter-threshold">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Filter threshold</a><a class="headerlink" href="#filter-threshold" title="Link to this heading">¶</a></h3>
<p>If User-defined <code class="docutils literal notranslate"><span class="pre">Mode</span> <span class="pre">options</span></code> is checked, then <code class="docutils literal notranslate"><span class="pre">Filter</span> <span class="pre">thresholds</span></code> is enable to customize the threshold of the six key metrics.</p>
</section>
</section>
<section id="snakemake-rules">
<h2><a class="toc-backref" href="#id7" role="doc-backlink">snakemake rules</a><a class="headerlink" href="#snakemake-rules" title="Link to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will firstly run <code class="docutils literal notranslate"><span class="pre">cp_unzip.sh</span></code> to unzip and redirect the input sequencing reads files. Then, <code class="docutils literal notranslate"><span class="pre">psiFinder.sh</span></code> will be secondly run to apply snakemake rules selected by <code class="docutils literal notranslate"><span class="pre">quickstart_config.yml</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">cp_unzip.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="nv">$#</span><span class="w"> </span>-ne<span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span><span class="w"> </span><span class="s1">&#39; input_sample input_path treat_sample treat_path log_file&#39;</span>
<span class="w">    </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
<span class="k">fi</span>

<span class="nv">input_sample</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">input_path</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">treat_sample</span><span class="o">=</span><span class="nv">$3</span>
<span class="nv">treat_path</span><span class="o">=</span><span class="nv">$4</span>
<span class="nv">log_file</span><span class="o">=</span><span class="nv">$5</span>


<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;clear psiFinder.sh history&quot;</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Your quickstart job is starting, please wait...&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$log_file</span>

<span class="c1">#.fastq.gz</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$input_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$treat_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fasta.gz</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$input_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$input_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$treat_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$treat_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fq.gz</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fq.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$input_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fq.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$treat_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fa.gz</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fa.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$input_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$input_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fa.gz<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: unzipping and redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>gunzip<span class="w"> </span>-c<span class="w"> </span><span class="nv">$treat_sample</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$treat_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fastq</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$input_sample</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$treat_sample</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fasta</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$input_sample</span><span class="w"> </span><span class="nv">$input_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$treat_sample</span><span class="w"> </span><span class="nv">$treat_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="c1">#.fq</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fq<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$input_sample</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fq<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fastq file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$treat_sample</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span><span class="p">&amp;</span>
<span class="k">fi</span>

<span class="c1">#.fa</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$input_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fa<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">input: redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$input_sample</span><span class="w"> </span><span class="nv">$input_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$input_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$input_path</span>
<span class="k">fi</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$treat_sample</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fa<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>tput<span class="w"> </span>setaf<span class="w"> </span><span class="m">2</span><span class="k">)</span><span class="s2">treat: redirecting fasta file...\n</span><span class="k">$(</span>tput<span class="w"> </span>sgr<span class="w"> </span><span class="m">0</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>cp<span class="w"> </span>-R<span class="w"> </span><span class="nv">$treat_sample</span><span class="w"> </span><span class="nv">$treat_path</span>
<span class="w">    </span>seqtk<span class="w"> </span>seq<span class="w"> </span>-F<span class="w"> </span><span class="s1">&#39;#&#39;</span><span class="w"> </span><span class="nv">$treat_path</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span>.fastq
<span class="w">    </span>rm<span class="w"> </span><span class="nv">$treat_path</span>
<span class="k">fi</span>

<span class="nb">wait</span>


<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Starting snakemake workflow...&quot;</span>
<span class="c1">#snakemake</span>
<span class="c1">#fastq</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fastq<span class="w"> </span><span class="o">]]</span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;check info: Input is fastq files&quot;</span><span class="w">  </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>nohup<span class="w"> </span>bash<span class="w"> </span>snakemake/psiFinder.sh<span class="w"> </span>-i<span class="w"> </span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fastq</span><span class="si">}</span><span class="k">)</span><span class="w"> </span>-t<span class="w"> </span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fastq</span><span class="si">}</span><span class="k">)</span><span class="w"> </span>-c<span class="w"> </span>quickstart_config.yml<span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span><span class="p">&amp;</span>
<span class="w">    </span><span class="nb">wait</span>
<span class="k">else</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;check info: Input is not fastq files&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="k">fi</span>

<span class="c1">#fasta</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">==</span><span class="w"> </span>*.fasta<span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;check info: Input is fasta files&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="w">    </span>nohup<span class="w"> </span>bash<span class="w"> </span>snakemake/psiFinder.sh<span class="w"> </span>-i<span class="w"> </span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">input_path</span><span class="p">%.fasta</span><span class="si">}</span><span class="k">)</span><span class="w"> </span>-t<span class="w"> </span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">treat_path</span><span class="p">%.fasta</span><span class="si">}</span><span class="k">)</span><span class="w"> </span>-c<span class="w"> </span>quickstart_config.yml<span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span><span class="p">&amp;</span>
<span class="w">    </span><span class="nb">wait</span>
<span class="k">else</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;check info: Input is not fasta files&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
<span class="k">fi</span>

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Finished snakemake workflow!&quot;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$log_file</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">psiFinder.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>

usage<span class="o">()</span><span class="w"> </span><span class="o">{</span><span class="w">                                      </span><span class="c1"># Function: Print a help message.</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Usage: </span><span class="nv">$0</span><span class="s2"> [ -h help] [ -i input ] [ -t treat ] [ -c quickstart_config.yml ]&quot;</span><span class="w"> </span><span class="m">1</span>&gt;<span class="p">&amp;</span><span class="m">2</span>
<span class="o">}</span>
exit_abnormal<span class="o">()</span><span class="w"> </span><span class="o">{</span><span class="w">                              </span><span class="c1"># Function: Exit with error.</span>
<span class="w">  </span>usage
<span class="w">  </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
<span class="o">}</span>


<span class="k">while</span><span class="w"> </span><span class="nb">getopts</span><span class="w"> </span><span class="s2">&quot;:h:i:t:c:&quot;</span><span class="w"> </span>options<span class="p">;</span><span class="w"> </span><span class="k">do</span>

<span class="w">  </span><span class="k">case</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">options</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="k">in</span>
<span class="w">    </span>h<span class="p">|</span>:<span class="o">)</span>
<span class="w">      </span>usage
<span class="w">      </span>Help
<span class="w">      </span><span class="nb">exit</span><span class="w"> </span><span class="m">0</span>
<span class="w">      </span><span class="p">;;</span>
<span class="w">    </span>i<span class="o">)</span>
<span class="w">      </span><span class="nv">input</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
<span class="w">      </span><span class="k">if</span><span class="w"> </span>!<span class="w"> </span><span class="o">[[</span><span class="w"> </span>-n<span class="w"> </span><span class="nv">$input</span><span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;You didn&#39;t set the input sample&quot;</span>
<span class="w">      </span><span class="k">fi</span>
<span class="w">      </span><span class="p">;;</span>
<span class="w">    </span>t<span class="o">)</span>
<span class="w">      </span><span class="nv">treat</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
<span class="w">      </span><span class="k">if</span><span class="w"> </span>!<span class="w"> </span><span class="o">[[</span><span class="w"> </span>-n<span class="w"> </span><span class="nv">$treat</span><span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;You didn&#39;t set the treat sample&quot;</span>
<span class="w">      </span><span class="k">fi</span>
<span class="w">      </span><span class="p">;;</span>
<span class="w">    </span>c<span class="o">)</span>
<span class="w">      </span><span class="nv">config</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
<span class="w">      </span><span class="k">if</span><span class="w"> </span>!<span class="w"> </span><span class="o">[[</span><span class="w"> </span>-n<span class="w"> </span><span class="nv">$config</span><span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">        </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;You didn&#39;t set the config yml file&quot;</span>
<span class="w">      </span><span class="k">fi</span>
<span class="w">      </span><span class="p">;;</span>
<span class="w">    </span><span class="se">\?</span><span class="o">)</span><span class="w"> </span><span class="c1"># incorrect option</span>
<span class="w">      </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Error: -</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span><span class="s2"> Invalid option&quot;</span>
<span class="w">      </span>exit_abnormal
<span class="w">      </span><span class="p">;;</span>
<span class="w">  </span><span class="k">esac</span>
<span class="k">done</span>

<span class="nb">shift</span><span class="w"> </span><span class="k">$((</span><span class="nv">$OPTIND</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span><span class="k">))</span>

<span class="nv">quickstart_config</span><span class="o">=</span><span class="k">$(</span>cat<span class="w"> </span><span class="nv">$config</span><span class="k">)</span>
<span class="nv">quickstart_array</span><span class="o">=(</span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span><span class="nv">$quickstart_config</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>tr<span class="w"> </span><span class="s2">&quot;:&quot;</span><span class="w"> </span><span class="s2">&quot;\n&quot;</span><span class="k">)</span><span class="o">)</span>

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;==Starting a quickstart job...==:\n&quot;</span>

<span class="c1">#cutadapt-se</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;ad_remove_input:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[41]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[41]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting input cutadapt fastq...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/cutadapt/input/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>.fastq
<span class="k">else</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Moving input cutadapt fastq...\n&quot;</span>
<span class="w">    </span>mv<span class="w"> </span>snakemake/reads/input/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>.fastq<span class="w"> </span>snakemake/cutadapt/input
<span class="k">fi</span>

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;ad_remove_treat:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[43]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[43]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting treat cutadapt fastq...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/cutadapt/treat/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.fastq
<span class="k">else</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Moving treat cutadapt fastq...\n&quot;</span>
<span class="w">    </span>mv<span class="w"> </span>snakemake/reads/treat/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.fastq<span class="w"> </span>snakemake/cutadapt/treat
<span class="k">fi</span>

<span class="c1">#STAR-index</span>
<span class="nv">temp</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[5]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span><span class="s2">&quot;s/\&quot;//g&quot;</span><span class="w">  </span><span class="k">)</span>
<span class="nv">DIRECTORY</span><span class="o">=</span><span class="s2">&quot;snakemake/genome/</span><span class="si">${</span><span class="nv">temp</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$DIRECTORY</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$DIRECTORY</span><span class="s2"> is not empty, STAR index have been built&quot;</span>

<span class="k">else</span>
<span class="w">  </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;No STAR index in </span><span class="nv">$DIRECTORY</span><span class="s2">, generating STAR index...\n&quot;</span>
<span class="w">  </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/genome/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">temp</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">fi</span>

<span class="c1">#rtsSeeker result</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;sites_identification:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[31]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[31]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting rtsSeeker result for Ψ sites identification...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/sites_identification/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker ann</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;ann:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[49]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[49]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting ann result for Ψ sites identification...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/ann/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_ann_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker svm</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;svm:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[45]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[45]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting svm result for Ψ sites identification...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/svm/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_svm_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#rtsSeeker user-defined</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;user-defined:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[51]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[51]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting user-defined result for Ψ sites identification...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/user_defined/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_user_defined_psi_prediction.bed
<span class="k">fi</span>

<span class="c1">#bedAnnotator</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;sites_annotation:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[33]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[33]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting bedAnnotator result for Ψ sites annotation...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/sites_annotation/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_add_seq_group_uniq.bed
<span class="k">fi</span>

<span class="c1">#metagene</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;metagene:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[47]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[47]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting metagene result for Ψ sites identification...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/meta_gene/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_metagene_pseudoU.pdf
<span class="k">fi</span>

<span class="c1">#ACAscan</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;sites_target_prediction:</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[35]</span><span class="si">}</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">quickstart_array</span><span class="p">[35]</span><span class="si">}</span><span class="s2">&quot;</span><span class="w"> </span><span class="o">=</span>~<span class="w"> </span><span class="s2">&quot;true&quot;</span><span class="w">  </span><span class="o">]]</span><span class="w"> </span><span class="p">;</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Getting ACAscan result for Ψ sites target prediction...\n&quot;</span>
<span class="w">    </span>snakemake<span class="w"> </span>-s<span class="w"> </span>snakemake/Snakefile<span class="w"> </span>--cores<span class="w"> </span><span class="m">8</span><span class="w"> </span>snakemake/output/sites_target_prediction/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="si">${</span><span class="nv">input</span><span class="si">}</span><span class="s2">&quot;</span>_versus_<span class="s2">&quot;</span><span class="si">${</span><span class="nv">treat</span><span class="si">}</span><span class="s2">&quot;</span>_fa.out
<span class="k">fi</span>

<span class="c1">#emit done signal</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;==All done!==&quot;</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">quickstart_config.yml</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>group:<span class="w"> </span><span class="s2">&quot;Mammal&quot;</span>
genome:<span class="w"> </span><span class="s2">&quot;Homo_sapiens&quot;</span>
assembly:<span class="w"> </span><span class="s2">&quot;hg38&quot;</span>
se_input:<span class="w"> </span><span class="s2">&quot;/public/home/chenzr/PSI_Seq_brainCell/A1-A12-totalRNA-result/A1.cutadapt.extendedFrags.collapse.cutBarcodes.fa.gz&quot;</span>
se_treat:<span class="w"> </span><span class="s2">&quot;/public/home/chenzr/PSI_Seq_brainCell/A1-A12-totalRNA-result/A2.cutadapt.extendedFrags.collapse.cutBarcodes.fa.gz&quot;</span>
se_input_prefix:<span class="w"> </span><span class="s2">&quot;A1&quot;</span>
se_treat_prefix:<span class="w"> </span><span class="s2">&quot;A2&quot;</span>
se_br5_input:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
se_br3_input:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
se_ad5_input:<span class="w"> </span><span class="s2">&quot;&quot;</span>
se_ad3_input:<span class="w"> </span><span class="s2">&quot;&quot;</span>
se_br5_treat:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
se_br3_treat:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
se_ad5_treat:<span class="w"> </span><span class="s2">&quot;&quot;</span>
se_ad3_treat:<span class="w"> </span><span class="s2">&quot;&quot;</span>
sites_identification:<span class="w"> </span><span class="s2">&quot;true&quot;</span>
sites_annotation:<span class="w"> </span><span class="s2">&quot;true&quot;</span>
sites_target_prediction:<span class="w"> </span><span class="s2">&quot;true&quot;</span>
br_remove_input:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
br_remove_treat:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
ad_remove_input:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
ad_remove_treat:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
svm:<span class="w"> </span><span class="s2">&quot;true&quot;</span>
metagene:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
ann:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
user_defined:<span class="w"> </span><span class="s2">&quot;false&quot;</span>
treatpre_Fold_thres:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
preFold_FC_thres:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
treataft_Fold_thres:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
aftFold_FC_thres:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
treatstoprate_thres:<span class="w"> </span><span class="s2">&quot;0&quot;</span>
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id8" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading">¶</a></h2>
<p>snakemake result are output to specific directory where psiFinder package is unziped.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>/the/directory/of/psiFinder/snakemake/
$<span class="w"> </span>tree<span class="w"> </span>-L<span class="w"> </span><span class="m">1</span>
.
├──<span class="w"> </span>cp_unzip.sh
├──<span class="w"> </span>genome
├──<span class="w"> </span>logs
├──<span class="w"> </span>output
├──<span class="w"> </span>psiFinder.sh
├──<span class="w"> </span>reads
├──<span class="w"> </span>script
├──<span class="w"> </span>Snakefile
├──<span class="w"> </span>star
└──<span class="w"> </span>cutadapt

<span class="m">7</span><span class="w"> </span>directories,<span class="w"> </span><span class="m">3</span><span class="w"> </span>files

$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>/the/directory/of/psiFinder/snakemake/output
$<span class="w"> </span>tree<span class="w"> </span>-L<span class="w"> </span><span class="m">1</span>
.
├──<span class="w"> </span>ann
├──<span class="w"> </span>meta_gene
├──<span class="w"> </span>sites_annotation
├──<span class="w"> </span>sites_identification
├──<span class="w"> </span>sites_target_prediction
├──<span class="w"> </span>svm
└──<span class="w"> </span>user_defined

<span class="m">7</span><span class="w"> </span>directories,<span class="w"> </span><span class="m">0</span><span class="w"> </span>files
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_quickstart_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html">Ψ-sites snoRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20and%20snoRNA%20Interaction.html">Ψ-sites and snoRNA Interaction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Motif%20Searching.html">Ψ-sites Motif Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html">Ψ-sites SNPs and SNVs Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Differential%20Modification.html">Ψ-sites Differential Modification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">snakemake pipline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#select-genome">Select Genome</a></li>
<li class="toctree-l3"><a class="reference internal" href="#data-upload">Data upload</a></li>
<li class="toctree-l3"><a class="reference internal" href="#basic-analyses">Basic analyses</a></li>
<li class="toctree-l3"><a class="reference internal" href="#mode-options">Mode options</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filter-threshold">Filter threshold</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#snakemake-rules">snakemake rules</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Accessory%20Tools.html">Accessory Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="Accessory%20Tools.html" title="Accessory Tools"
             >next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Functional%20Enrichment.html" title="Ψ-sites Functional Enrichment"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">snakemake pipline</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
    &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 8.2.3.
    </div>
  </body>
</html>