<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Transcripts Alignment &mdash; psiFinder v1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Identification" href="%CE%A8-sites%20Identification.html" />
    <link rel="prev" title="Quality Control" href="Quality%20Control.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> psiFinder
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Transcripts Alignment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l2"><a class="reference internal" href="#perform-transcripts-mapping">Perform transcripts mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Target%20Prediction.html">Ψ-sites Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">psiFinder</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Transcripts Alignment</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Transcripts Alignment.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="transcripts-alignment">
<h1>Transcripts Alignment<a class="headerlink" href="#transcripts-alignment" title="Permalink to this heading"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#perform-transcripts-mapping" id="id2">Perform transcripts mapping</a></p></li>
<li><p><a class="reference internal" href="#output" id="id3">Output</a></p></li>
</ul>
</div>
<p>The functions of this interface utilized <a class="reference external" href="https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf">STAR</a> to map sequencing reads to reference genome location and consist of two parts:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Generating</span> <span class="pre">genome</span> <span class="pre">indexes</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Running</span> <span class="pre">mapping</span> <span class="pre">jobs</span></code></p></li>
</ul>
<p>Once click <code class="docutils literal notranslate"><span class="pre">BUILD</span></code>, then psiFinder will run <code class="docutils literal notranslate"><span class="pre">2.0_STAR_index.sh</span></code>.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

Help<span class="o">()</span> <span class="o">{</span>
<span class="nb">echo</span> <span class="s1">&#39;Usage: &#39;</span><span class="nv">$0</span> <span class="s1">&#39; -a threads -b genome -c gtf -o outputpath&#39;</span>
<span class="o">}</span>


usage<span class="o">()</span> <span class="o">{</span>                                      <span class="c1"># Function: Print a help message.</span>
  <span class="nb">echo</span> <span class="s2">&quot;Usage: </span><span class="nv">$0</span><span class="s2"> [ -h help] [ -a threads ] [ -b genome ] [ -c gtf ] [ -o outputpath ]&quot;</span> <span class="m">1</span>&gt;<span class="p">&amp;</span><span class="m">2</span>
<span class="o">}</span>
exit_abnormal<span class="o">()</span> <span class="o">{</span>                              <span class="c1"># Function: Exit with error.</span>
  usage
  <span class="nb">exit</span> <span class="m">1</span>
<span class="o">}</span>


<span class="k">while</span> <span class="nb">getopts</span> <span class="s2">&quot;:h:a:b:c:o:&quot;</span> options<span class="p">;</span> <span class="k">do</span>

  <span class="k">case</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">options</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">in</span>
    h<span class="p">|</span>:<span class="o">)</span>
      usage
      Help
      <span class="nb">exit</span> <span class="m">0</span>
      <span class="p">;;</span>
    a<span class="o">)</span>
      <span class="nv">threads</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$threads</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t input the threads&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    b<span class="o">)</span>
      <span class="nv">genome</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$genome</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t input the genome&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    c<span class="o">)</span>
      <span class="nv">gtf</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$gtf</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t input the gtf&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    o<span class="o">)</span>
      <span class="nv">outputpath</span><span class="o">=</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span>
      <span class="k">if</span> ! <span class="o">[[</span> -n <span class="nv">$outputpath</span> <span class="o">]]</span> <span class="p">;</span> <span class="k">then</span>
        <span class="nb">echo</span> <span class="s2">&quot;You didn&#39;t input the outputpath&quot;</span>
      <span class="k">fi</span>
      <span class="p">;;</span>
    <span class="se">\?</span><span class="o">)</span> <span class="c1"># incorrect option</span>
      <span class="nb">echo</span> <span class="s2">&quot;Error: -</span><span class="si">${</span><span class="nv">OPTARG</span><span class="si">}</span><span class="s2"> Invalid option&quot;</span>
      exit_abnormal
      <span class="p">;;</span>
  <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">shift</span> <span class="k">$((</span><span class="nv">$OPTIND</span> <span class="o">-</span> <span class="m">1</span><span class="k">))</span>
date  <span class="c1">## echo the date at start</span>


<span class="nb">echo</span> star indexing start
<span class="nb">echo</span> STAR --runThreadN <span class="nv">$threads</span> --runMode genomeGenerate --genomeDir <span class="nv">$outputpath</span> --genomeFastaFiles <span class="nv">$genome</span> --sjdbGTFfile <span class="nv">$gtf</span> --sjdbOverhang <span class="m">99</span>
    STAR <span class="se">\</span>
    --runThreadN <span class="nv">$threads</span> <span class="se">\</span>
    --runMode genomeGenerate <span class="se">\</span>
    --genomeDir <span class="nv">$outputpath</span><span class="se">\</span>
    --genomeFastaFiles <span class="nv">$genome</span> <span class="se">\</span>
    --sjdbGTFfile <span class="nv">$gtf</span> <span class="se">\</span>
    --sjdbOverhang <span class="m">99</span>
<span class="nb">echo</span> star indexing end
</pre></div>
</div>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, then psiFinder will run <code class="docutils literal notranslate"><span class="pre">2.1_STAR_align.sh</span></code>.</p>
<img alt="../_images/Transcripts_Alignment.png" src="../_images/Transcripts_Alignment.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading"></a></h2>
<p>Firstly, upload downloaded genome sequence and annotation file to build genome indexes. Next, locate the directory where your indexes were generated, and <code class="docutils literal notranslate"><span class="pre">browse</span></code> the syntropic read (the sequencing reads with the same strand direction as the genome) file for both CMC-input and CMC-treat samples.</p>
</section>
<section id="perform-transcripts-mapping">
<h2><a class="toc-backref" href="#id2">Perform transcripts mapping</a><a class="headerlink" href="#perform-transcripts-mapping" title="Permalink to this heading"></a></h2>
<p>Built-in STAR argument for <code class="docutils literal notranslate"><span class="pre">Running</span> <span class="pre">mapping</span> <span class="pre">jobs</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>STAR <span class="se">\</span>
    --genomeDir <span class="nv">$starIndex</span> <span class="se">\</span>
    --readFilesIn <span class="nv">$input</span> <span class="se">\</span>
    --runThreadN <span class="nv">$threads</span> <span class="se">\</span>
    --genomeLoad NoSharedMemory <span class="se">\</span>
    --alignEndsType EndToEnd <span class="se">\</span>
    --outSAMtype BAM SortedByCoordinate <span class="se">\</span>
    --outFileNamePrefix <span class="si">${</span><span class="nv">input</span><span class="p">%.fastq</span><span class="si">}</span> <span class="se">\</span>
    --outStd Log <span class="se">\</span>
    --limitBAMsortRAM <span class="m">60000000000</span> <span class="se">\</span>
    --outFilterType BySJout <span class="se">\</span>
    --outFilterMultimapScoreRange <span class="m">0</span> <span class="se">\</span>
    --outFilterMultimapNmax <span class="m">30</span> <span class="se">\</span>
    --outFilterMismatchNmax <span class="m">15</span> <span class="se">\</span>
    --outFilterMismatchNoverLmax <span class="m">0</span>.1 <span class="se">\</span>
    --outFilterScoreMin <span class="m">0</span> <span class="se">\</span>
    --outFilterScoreMinOverLread <span class="m">0</span> <span class="se">\</span>
    --outFilterMatchNmin <span class="m">15</span> <span class="se">\</span>
    --outFilterMatchNminOverLread <span class="m">0</span>.8 <span class="se">\</span>
    --alignIntronMin <span class="m">20</span> <span class="se">\</span>
    --alignIntronMax <span class="m">1000000</span> <span class="se">\</span>
    --alignMatesGapMax <span class="m">1000000</span> <span class="se">\</span>
    --seedSearchStartLmax <span class="m">15</span> <span class="se">\</span>
    --seedSearchStartLmaxOverLread <span class="m">1</span> <span class="se">\</span>
    --seedSearchLmax <span class="m">0</span> <span class="se">\</span>
    --seedMultimapNmax <span class="m">20000</span> <span class="se">\</span>
    --seedPerReadNmax <span class="m">1000</span> <span class="se">\</span>
    --seedPerWindowNmax <span class="m">100</span> <span class="se">\</span>
    --seedNoneLociPerWindow <span class="m">20</span> <span class="se">\</span>
    --alignSJoverhangMin <span class="m">8</span> <span class="se">\</span>
    --alignSJDBoverhangMin <span class="m">1</span> <span class="se">\</span>
    --quantMode TranscriptomeSAM GeneCounts <span class="se">\</span>
    --outSAMmode Full <span class="se">\</span>
    --outSAMattributes All <span class="se">\</span>
    --outSAMunmapped None <span class="se">\</span>
    --outSAMorder Paired <span class="se">\</span>
    --outSAMprimaryFlag AllBestScore <span class="se">\</span>
    --outSAMreadID Standard <span class="se">\</span>
    --outReadsUnmapped Fastx <span class="se">\</span>
    --alignEndsProtrude <span class="m">150</span> ConcordantPair <span class="se">\</span>
    --limitOutSJcollapsed <span class="m">5000000</span>
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id3">Output</a><a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>Results in bam format (with <code class="docutils literal notranslate"><span class="pre">Aligned.sortedByCoord.out.bam</span></code> suffix) will be output to the target directory.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffixes <code class="docutils literal notranslate"><span class="pre">_STAR_build_config.txt</span></code> and <code class="docutils literal notranslate"><span class="pre">_STAR_align_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="Quality%20Control.html" class="btn btn-neutral float-left" title="Quality Control" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="%CE%A8-sites%20Identification.html" class="btn btn-neutral float-right" title="Ψ-sites Identification" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, chenzhr23.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>