

<!doctype html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ψ-sites Differential Modification &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/bizstyle.css" />
    
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Functional Enrichment" href="%CE%A8-sites%20Functional%20Enrichment.html" />
    <link rel="prev" title="Ψ-sites SNPs and SNVs Searching" href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20Functional%20Enrichment.html" title="Ψ-sites Functional Enrichment"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html" title="Ψ-sites SNPs and SNVs Searching"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Differential Modification</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="sites-differential-modification">
<h1>Ψ-sites Differential Modification<a class="headerlink" href="#sites-differential-modification" title="Permalink to this heading">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#compare-sites-modification-abundance" id="id2">Compare Ψ-sites modification abundance</a></p></li>
<li><p><a class="reference internal" href="#optional-parameters" id="id3">Optional Parameters</a></p>
<ul>
<li><p><a class="reference internal" href="#up-log2fc" id="id4">up-log2FC</a></p></li>
<li><p><a class="reference internal" href="#down-log2fc" id="id5">down-log2FC</a></p></li>
<li><p><a class="reference internal" href="#p-value" id="id6">P.Value</a></p></li>
<li><p><a class="reference internal" href="#min-fold" id="id7">min-fold</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id8">Output</a></p>
<ul>
<li><p><a class="reference internal" href="#information" id="id9">Information</a></p></li>
<li><p><a class="reference internal" href="#diagram" id="id10">Diagram</a></p></li>
</ul>
</li>
</ul>
</div>
<p>psiFinder <code class="docutils literal notranslate"><span class="pre">Ψ-sites</span> <span class="pre">Differential</span> <span class="pre">Modification</span></code> utilize <a class="reference external" href="https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf">limma</a> to generate result for Ψ-sites differential modification based on <code class="docutils literal notranslate"><span class="pre">stopRatioFC</span></code> of input rtsSeeker result files.</p>
<img alt="../_images/Differential_Modification.png" src="../_images/Differential_Modification.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading">¶</a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result with <code class="docutils literal notranslate"><span class="pre">_all.bed</span></code> suffix) to <code class="docutils literal notranslate"><span class="pre">pseUdiff</span></code> QT widget in bed format, , to construct Ψ-sites differential modification comparison.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Group1</strong>: normally the control group, input required at least two samples; <strong>Group2</strong>: normally the experiment group , input required at least two samples.</p>
</div>
</section>
<section id="compare-sites-modification-abundance">
<h2><a class="toc-backref" href="#id2">Compare Ψ-sites modification abundance</a><a class="headerlink" href="#compare-sites-modification-abundance" title="Permalink to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">pseUdiff.sh</span></code>, <code class="docutils literal notranslate"><span class="pre">pseUdiff.r</span></code>, <code class="docutils literal notranslate"><span class="pre">limma_bed_annotation.sh</span></code>, and <code class="docutils literal notranslate"><span class="pre">limma_remove_redundancy.r</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">pseUdiff.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -ne <span class="m">9</span> <span class="o">]]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: &#39;</span><span class="nv">$0</span> <span class="s1">&#39; group1_name group2_name group1filelist group2filelist uplog2FC downlog2FC pval mincount outdir&#39;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="k">fi</span>

<span class="nv">group1_name</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">group2_name</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">group1filelist</span><span class="o">=</span><span class="nv">$3</span>
<span class="nv">group2filelist</span><span class="o">=</span><span class="nv">$4</span>
<span class="nv">uplog2FC</span><span class="o">=</span><span class="nv">$5</span>
<span class="nv">downlog2FC</span><span class="o">=</span><span class="nv">$6</span>
<span class="nv">pval</span><span class="o">=</span><span class="nv">$7</span>
<span class="nv">mincount</span><span class="o">=</span><span class="nv">$8</span>
<span class="nv">outdir</span><span class="o">=</span><span class="nv">$9</span>


Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/pseudiff.r -d <span class="nv">$group1_name</span> -e <span class="nv">$group2_name</span> -f <span class="nv">$group1filelist</span> -g <span class="nv">$group2filelist</span> -l <span class="nv">$mincount</span> -o <span class="nv">$outdir</span> &gt; <span class="s2">&quot;</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span><span class="s2">/</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span>_limma.log
cat <span class="s2">&quot;</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span><span class="s2">/</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span>_limma.log
bash <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/limma_bed_annotation.sh <span class="nv">$uplog2FC</span> <span class="nv">$downlog2FC</span> <span class="nv">$pval</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span> <span class="s2">&quot;</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span><span class="s2">/</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span>_treat_control_overall_filt.bed <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6 <span class="nv">$group1filelist</span> <span class="nv">$group2filelist</span> &gt; <span class="s2">&quot;</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span><span class="s2">/</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span>_limma_bed_annotation.log
cat <span class="s2">&quot;</span><span class="si">${</span><span class="nv">outdir</span><span class="si">}</span><span class="s2">/</span><span class="si">${</span><span class="nv">group2_name</span><span class="si">}</span><span class="s2">-</span><span class="si">${</span><span class="nv">group1_name</span><span class="si">}</span><span class="s2">&quot;</span>_limma_bed_annotation.log
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">pseUdiff.r</span></code></p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/usr/bin/env Rscript</span>
<span class="nf">options</span><span class="p">(</span><span class="n">warn</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;optparse&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;openxlsx&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;limma&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;edgeR&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;dplyr&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;tidyverse&quot;</span><span class="p">))</span>

<span class="nf">options</span><span class="p">(</span><span class="n">warn</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span>

<span class="n">option_list</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-d&quot;</span><span class="p">,</span> <span class="s">&quot;--group1_name&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;file name of group1 for limma [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-e&quot;</span><span class="p">,</span> <span class="s">&quot;--group2_name&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;file name of group1 for limma [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-f&quot;</span><span class="p">,</span> <span class="s">&quot;--group1filelist&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;file list of group1 for limma [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-g&quot;</span><span class="p">,</span> <span class="s">&quot;--group2filelist&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;file list of group2 for limma [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-l&quot;</span><span class="p">,</span> <span class="s">&quot;--mincount&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;mincount for limma [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-o&quot;</span><span class="p">,</span> <span class="s">&quot;--outdir&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;output file name [default= %default]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">)</span>
<span class="p">);</span>
<span class="n">opt_parser</span> <span class="o">=</span> <span class="nf">OptionParser</span><span class="p">(</span><span class="n">option_list</span><span class="o">=</span><span class="n">option_list</span><span class="p">);</span>
<span class="n">opt</span> <span class="o">=</span> <span class="nf">parse_args</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>

<span class="nf">if </span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group1_name</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group2_name</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group1filelist</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group2filelist</span><span class="p">)</span>  <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">mincount</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">outdir</span><span class="p">)</span> <span class="p">){</span>
  <span class="nf">print_help</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>
  <span class="nf">stop</span><span class="p">(</span><span class="s">&quot;Please provide -d group1_name, -e group2_name, -f group1filelist, -g group2filelist, -l mincount, and -o outdir option&quot;</span><span class="p">,</span> <span class="n">call.</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">);</span>
<span class="p">}</span>

<span class="n">group1_name</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group1_name</span>
<span class="n">group2_name</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group2_name</span>
<span class="n">group1filelist</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group1filelist</span>
<span class="n">group1filelist</span><span class="o">&lt;-</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">,</span><span class="s">&quot;,&quot;</span><span class="p">))</span>
<span class="n">group2filelist</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group2filelist</span>
<span class="n">group2filelist</span><span class="o">&lt;-</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">,</span><span class="s">&quot;,&quot;</span><span class="p">))</span>
<span class="n">mincount</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">mincount</span>
<span class="n">outdir</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">outdir</span>

<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;name for group1:&quot;</span><span class="p">,</span><span class="n">group1_name</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;name for group2:&quot;</span><span class="p">,</span><span class="n">group2_name</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;control group (group1) list:&quot;</span><span class="p">,</span><span class="n">group1filelist</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;experiment group (group2) list:&quot;</span><span class="p">,</span><span class="n">group2filelist</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;down-regulated log2(fold change):&quot;</span><span class="p">,</span><span class="n">downlog2FC</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;minimum count for each group (replicates):&quot;</span><span class="p">,</span><span class="n">mincount</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;output directory&quot;</span><span class="p">,</span><span class="n">outdir</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">print</span><span class="p">(</span><span class="s">&quot;limma program start&quot;</span><span class="p">)</span>

<span class="c1">####group1####</span>
<span class="nf">if</span><span class="p">(</span><span class="o">!</span><span class="nf">is.null</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)){</span>
  <span class="nf">print</span><span class="p">(</span><span class="s">&quot;getting group 1 foldChange&quot;</span><span class="p">)</span>
  <span class="n">datalist</span> <span class="o">=</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="nf">read.delim</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">))</span>
  <span class="nf">names</span><span class="p">(</span><span class="n">datalist</span><span class="p">)</span><span class="o">&lt;-</span><span class="n">group1filelist</span>
  <span class="nf">if</span><span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;*.bed&quot;</span><span class="p">,</span><span class="nf">basename</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">))){</span>
    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;input files are bed format&quot;</span><span class="p">)</span>
    <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="nf">colnames</span><span class="p">(</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]])[</span><span class="m">1</span><span class="p">]</span><span class="o">&lt;&lt;-</span><span class="s">&quot;chrom&quot;</span>
  <span class="p">}))</span>
  <span class="p">}</span><span class="n">else</span><span class="p">{</span>
    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;input files are txt format&quot;</span><span class="p">)</span>
  <span class="p">}</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">$</span><span class="n">base</span><span class="o">&lt;&lt;-</span><span class="nf">str_replace_all</span><span class="p">(</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">$</span><span class="n">base</span><span class="p">,</span><span class="s">&quot;TRUE&quot;</span><span class="p">,</span><span class="s">&quot;T&quot;</span><span class="p">)</span>
  <span class="p">}))</span>
  <span class="n">foldChange_datalist</span><span class="o">&lt;-</span><span class="nf">list</span><span class="p">()</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">foldChange_datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">&lt;&lt;-</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">base</span><span class="o">==</span><span class="s">&quot;T&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span><span class="n">chromStart</span><span class="p">,</span><span class="n">chromEnd</span><span class="p">,</span><span class="n">name</span><span class="p">,</span><span class="n">foldChange</span><span class="p">,</span><span class="n">treatStopNum</span><span class="p">,</span><span class="n">treatPreRpmFold</span><span class="p">,</span><span class="n">preRpmFoldRatio</span><span class="p">,</span><span class="n">treatAfterRpmFold</span><span class="p">,</span><span class="n">afterRpmFoldRatio</span><span class="p">,</span><span class="n">treatStopRatio</span><span class="p">,</span><span class="n">stopRatioFC</span><span class="p">,</span><span class="n">strand</span><span class="p">,</span><span class="n">extendSeq</span><span class="p">)</span>
  <span class="p">}))</span>
  <span class="c1">#give names to datalist</span>
  <span class="nf">names</span><span class="p">(</span><span class="n">foldChange_datalist</span><span class="p">)</span><span class="o">&lt;-</span><span class="n">group1filelist</span>
  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist</span> <span class="o">%&gt;%</span> <span class="nf">reduce</span><span class="p">(</span><span class="n">full_join</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">)</span>

  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="nf">distinct</span><span class="p">(</span><span class="n">extendSeq</span><span class="p">,</span><span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
  <span class="n">col_name</span><span class="o">&lt;-</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">)</span>

  <span class="n">chrom_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chrom$|chrom[.]&quot;</span><span class="p">))]</span>
  <span class="n">chrom_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chrom_col_name</span><span class="p">])</span>

  <span class="n">chromStart_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chromStart$|chromStart[.]&quot;</span><span class="p">))]</span>
  <span class="n">chromStart_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chromStart_col_name</span><span class="p">])</span>

  <span class="n">chromEnd_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chromEnd$|chromEnd[.]&quot;</span><span class="p">))]</span>
  <span class="n">chromEnd_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chromEnd_col_name</span><span class="p">])</span>

  <span class="n">strand_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^strand$|strand[.]&quot;</span><span class="p">))]</span>
  <span class="n">strand_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">strand_col_name</span><span class="p">])</span>

  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chrom.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chrom_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromStart.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chromStart_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromEnd.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chromEnd_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">strand.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">strand_col_name_list</span><span class="p">))</span>

  <span class="n">foldChange_datalist_df_chrloc</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">chrom.x</span><span class="p">,</span><span class="n">chromStart.x</span><span class="p">,</span><span class="n">chromEnd.x</span><span class="p">,</span><span class="n">strand.x</span><span class="p">)</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;foldChange&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatStopNum</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopNum&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatPreRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatPreRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_preRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;preRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatAfterRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatAfterRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_afterRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;afterRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatStopRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_stopRatioFC</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;stopRatioFC&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_extendSeq</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">foldChange_datalist_df_chrloc</span><span class="p">,</span><span class="n">foldChange_datalist_df_extendSeq</span><span class="p">,</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatStopNum</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatPreRpmFold</span><span class="p">,</span><span class="n">foldChange_datalist_df_preRpmFoldRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatAfterRpmFold</span><span class="p">,</span><span class="n">foldChange_datalist_df_afterRpmFoldRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatStopRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_stopRatioFC</span><span class="p">)</span>

  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)))]</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;.[xy]&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)))]),</span><span class="nf">rep</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;rep&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="m">1</span><span class="p">,</span><span class="n">each</span><span class="o">=</span><span class="m">1</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_overall</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df_foldChange</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">group1_overall</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom&quot;</span><span class="p">,</span><span class="s">&quot;chromStart&quot;</span><span class="p">,</span><span class="s">&quot;chromEnd&quot;</span><span class="p">,</span><span class="s">&quot;strand&quot;</span><span class="p">)</span>

<span class="p">}</span>

<span class="c1">####group2####</span>
<span class="nf">if</span><span class="p">(</span><span class="o">!</span><span class="nf">is.null</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)){</span>
  <span class="nf">print</span><span class="p">(</span><span class="s">&quot;getting group 2 foldChange&quot;</span><span class="p">)</span>
  <span class="n">datalist</span> <span class="o">=</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="nf">read.delim</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">))</span>
  <span class="nf">names</span><span class="p">(</span><span class="n">datalist</span><span class="p">)</span><span class="o">&lt;-</span><span class="n">group2filelist</span>
  <span class="nf">if</span><span class="p">(</span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;*.bed&quot;</span><span class="p">,</span><span class="nf">basename</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">))){</span>
    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;input files are bed format&quot;</span><span class="p">)</span>
    <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="nf">colnames</span><span class="p">(</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]])[</span><span class="m">1</span><span class="p">]</span><span class="o">&lt;&lt;-</span><span class="s">&quot;chrom&quot;</span>
  <span class="p">}))</span>
  <span class="p">}</span><span class="n">else</span><span class="p">{</span>
    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;input files are txt format&quot;</span><span class="p">)</span>
  <span class="p">}</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">$</span><span class="n">base</span><span class="o">&lt;&lt;-</span><span class="nf">str_replace_all</span><span class="p">(</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">$</span><span class="n">base</span><span class="p">,</span><span class="s">&quot;TRUE&quot;</span><span class="p">,</span><span class="s">&quot;T&quot;</span><span class="p">)</span>
  <span class="p">}))</span>
  <span class="n">foldChange_datalist</span><span class="o">&lt;-</span><span class="nf">list</span><span class="p">()</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">datalist</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">foldChange_datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">&lt;&lt;-</span><span class="n">datalist</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">base</span><span class="o">==</span><span class="s">&quot;T&quot;</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span><span class="n">chromStart</span><span class="p">,</span><span class="n">chromEnd</span><span class="p">,</span><span class="n">name</span><span class="p">,</span><span class="n">foldChange</span><span class="p">,</span><span class="n">treatStopNum</span><span class="p">,</span><span class="n">treatPreRpmFold</span><span class="p">,</span><span class="n">preRpmFoldRatio</span><span class="p">,</span><span class="n">treatAfterRpmFold</span><span class="p">,</span><span class="n">afterRpmFoldRatio</span><span class="p">,</span><span class="n">treatStopRatio</span><span class="p">,</span><span class="n">stopRatioFC</span><span class="p">,</span><span class="n">strand</span><span class="p">,</span><span class="n">extendSeq</span><span class="p">)</span>
  <span class="p">}))</span>
  <span class="c1">#give names to datalist</span>
  <span class="nf">names</span><span class="p">(</span><span class="n">foldChange_datalist</span><span class="p">)</span><span class="o">&lt;-</span><span class="n">group2filelist</span>
  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist</span> <span class="o">%&gt;%</span> <span class="nf">reduce</span><span class="p">(</span><span class="n">full_join</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">)</span>

  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="nf">distinct</span><span class="p">(</span><span class="n">extendSeq</span><span class="p">,</span><span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
  <span class="n">col_name</span><span class="o">&lt;-</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">)</span>

  <span class="n">chrom_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chrom$|chrom[.]&quot;</span><span class="p">))]</span>
  <span class="n">chrom_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chrom_col_name</span><span class="p">])</span>

  <span class="n">chromStart_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chromStart$|chromStart[.]&quot;</span><span class="p">))]</span>
  <span class="n">chromStart_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chromStart_col_name</span><span class="p">])</span>

  <span class="n">chromEnd_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^chromEnd$|chromEnd[.]&quot;</span><span class="p">))]</span>
  <span class="n">chromEnd_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">chromEnd_col_name</span><span class="p">])</span>

  <span class="n">strand_col_name</span><span class="o">&lt;-</span><span class="n">col_name</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">col_name</span><span class="p">,</span><span class="s">&quot;^strand$|strand[.]&quot;</span><span class="p">))]</span>
  <span class="n">strand_col_name_list</span><span class="o">&lt;-</span><span class="nf">as.list</span><span class="p">(</span><span class="n">foldChange_datalist_df</span><span class="p">[,</span><span class="n">strand_col_name</span><span class="p">])</span>

  <span class="n">foldChange_datalist_df</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chrom.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chrom_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromStart.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chromStart_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromEnd.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">chromEnd_col_name_list</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">strand.x</span> <span class="o">=</span><span class="nf">do.call</span><span class="p">(</span><span class="n">coalesce</span><span class="p">,</span><span class="n">strand_col_name_list</span><span class="p">))</span>

  <span class="n">foldChange_datalist_df_chrloc</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">chrom.x</span><span class="p">,</span><span class="n">chromStart.x</span><span class="p">,</span><span class="n">chromEnd.x</span><span class="p">,</span><span class="n">strand.x</span><span class="p">)</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;foldChange&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatStopNum</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopNum&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatPreRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatPreRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_preRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;preRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatAfterRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatAfterRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_afterRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;afterRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_treatStopRatio</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_stopRatioFC</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;stopRatioFC&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_extendSeq</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">foldChange_datalist_df_chrloc</span><span class="p">,</span><span class="n">foldChange_datalist_df_extendSeq</span><span class="p">,</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatStopNum</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatPreRpmFold</span><span class="p">,</span><span class="n">foldChange_datalist_df_preRpmFoldRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatAfterRpmFold</span><span class="p">,</span><span class="n">foldChange_datalist_df_afterRpmFoldRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_treatStopRatio</span><span class="p">,</span><span class="n">foldChange_datalist_df_stopRatioFC</span><span class="p">)</span>

  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)))]</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;.[xy]&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)))]),</span><span class="nf">rep</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;rep&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="m">1</span><span class="p">,</span><span class="n">each</span><span class="o">=</span><span class="m">1</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>
  <span class="n">foldChange_datalist_df_foldChange</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_datalist_df_foldChange</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group2_overall</span><span class="o">&lt;-</span><span class="n">foldChange_datalist_df_foldChange</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">group2_overall</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom&quot;</span><span class="p">,</span><span class="s">&quot;chromStart&quot;</span><span class="p">,</span><span class="s">&quot;chromEnd&quot;</span><span class="p">,</span><span class="s">&quot;strand&quot;</span><span class="p">)</span>

<span class="p">}</span>

<span class="nf">if</span><span class="p">(</span><span class="o">!</span><span class="nf">is.null</span><span class="p">(</span><span class="n">group1_overall</span><span class="p">)</span> <span class="o">&amp;</span> <span class="o">!</span><span class="nf">is.null</span><span class="p">(</span><span class="n">group2_overall</span><span class="p">)){</span>
  <span class="nf">print</span><span class="p">(</span><span class="s">&quot;calculating contrast significance of group2/group1 using limma voom&quot;</span><span class="p">)</span>
  <span class="n">group_list</span><span class="o">&lt;-</span><span class="nf">list</span><span class="p">(</span><span class="n">group1_overall</span><span class="p">,</span><span class="n">group2_overall</span><span class="p">)</span>
  <span class="nf">names</span><span class="p">(</span><span class="n">group_list</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;group1&quot;</span><span class="p">,</span><span class="s">&quot;group2&quot;</span><span class="p">)</span>
  <span class="n">foldChange_group_list_df</span><span class="o">&lt;-</span><span class="n">group_list</span> <span class="o">%&gt;%</span> <span class="nf">reduce</span><span class="p">(</span><span class="n">full_join</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df</span><span class="p">)</span>

  <span class="n">foldChange_group_list_df</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="nf">distinct</span><span class="p">(</span><span class="n">extendSeq</span><span class="p">,</span><span class="n">.keep_all</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
  <span class="n">foldChange_group_list_df</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chrom.x</span> <span class="o">=</span> <span class="nf">coalesce</span><span class="p">(</span><span class="n">chrom.x</span><span class="p">,</span><span class="n">chrom.y</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromStart.x</span> <span class="o">=</span> <span class="nf">coalesce</span><span class="p">(</span><span class="n">chromStart.x</span><span class="p">,</span><span class="n">chromStart.y</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">chromEnd.x</span> <span class="o">=</span> <span class="nf">coalesce</span><span class="p">(</span><span class="n">chromEnd.x</span><span class="p">,</span><span class="n">chromEnd.y</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">strand.x</span> <span class="o">=</span> <span class="nf">coalesce</span><span class="p">(</span><span class="n">strand.x</span><span class="p">,</span><span class="n">strand.y</span><span class="p">))</span>

  <span class="n">foldChange_group_list_df_chrloc</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">chrom.x</span><span class="p">,</span><span class="n">chromStart.x</span><span class="p">,</span><span class="n">chromEnd.x</span><span class="p">,</span><span class="n">strand.x</span><span class="p">)</span>
  <span class="n">foldChange_group_list_df_foldChange</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;foldChange&quot;</span><span class="p">))</span>

  <span class="c1">#treatStopNum</span>
  <span class="n">foldChange_group_list_df_treatStopNum</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopNum&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_treatStopNum</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopNum</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_treatStopNumBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopNum</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_treatStopNumBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopNum</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#treatPreRpmFold</span>
  <span class="n">foldChange_group_list_df_treatPreRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatPreRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_treatPreRpmFold</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatPreRpmFold</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_treatPreRpmFoldBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatPreRpmFold</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_treatPreRpmFoldBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatPreRpmFold</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#preRpmFoldRatio</span>
  <span class="n">foldChange_group_list_df_preRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;preRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_preRpmFoldRatio</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_preRpmFoldRatio</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_preRpmFoldRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_preRpmFoldRatio</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_preRpmFoldRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_preRpmFoldRatio</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#treatAfterRpmFold</span>
  <span class="n">foldChange_group_list_df_treatAfterRpmFold</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatAfterRpmFold&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_treatAfterRpmFold</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatAfterRpmFold</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_treatAfterRpmFoldBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatAfterRpmFold</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_treatAfterRpmFoldBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatAfterRpmFold</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#afterRpmFoldRatio</span>
  <span class="n">foldChange_group_list_df_afterRpmFoldRatio</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;afterRpmFoldRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_afterRpmFoldRatio</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_afterRpmFoldRatio</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_afterRpmFoldRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_afterRpmFoldRatio</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_afterRpmFoldRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_afterRpmFoldRatio</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#treatStopRatio</span>
  <span class="n">foldChange_group_list_df_treatStopRatio</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;treatStopRatio&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_treatStopRatio</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopRatio</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_treatStopRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopRatio</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_treatStopRatioBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_treatStopRatio</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="c1">#stopRatioFC</span>
  <span class="n">foldChange_group_list_df_stopRatioFC</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;stopRatioFC&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_stopRatioFC</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_stopRatioFC</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>
  <span class="n">group1_stopRatioFCBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_stopRatioFC</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)])</span>
  <span class="n">group2_stopRatioFCBaseMean</span><span class="o">&lt;-</span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">foldChange_group_list_df_stopRatioFC</span><span class="p">[,(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))])</span>

  <span class="n">foldChange_group_list_df_extendSeq</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="nf">contains</span><span class="p">(</span><span class="s">&quot;extendSeq&quot;</span><span class="p">))</span>
  <span class="n">foldChange_group_list_df_foldChange</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">foldChange_group_list_df_chrloc</span><span class="p">,</span><span class="n">foldChange_group_list_df_extendSeq</span><span class="p">,</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)</span>

  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)))]</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;.[xy]&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)))]),</span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group&quot;</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">times</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)),</span><span class="nf">rep</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group&quot;</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">times</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>
  <span class="n">foldChange_group_list_df_foldChange</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)]</span> <span class="o">=</span> <span class="m">0</span>

  <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom&quot;</span><span class="p">,</span><span class="s">&quot;chromStart&quot;</span><span class="p">,</span><span class="s">&quot;chromEnd&quot;</span><span class="p">,</span><span class="s">&quot;strand&quot;</span><span class="p">)</span>

  <span class="c1">#construct sample design</span>
  <span class="n">design</span> <span class="o">&lt;-</span> <span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span> <span class="m">0</span><span class="o">+</span><span class="nf">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)),</span><span class="nf">rep</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)))))</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;group1&quot;</span><span class="p">,</span> <span class="s">&quot;group2&quot;</span><span class="p">)</span>
  <span class="n">foldChange</span><span class="o">&lt;-</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">[,</span><span class="m">6</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">)]</span>
  <span class="nf">rownames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">colnames</span><span class="p">(</span><span class="n">foldChange</span><span class="p">)</span>

  <span class="n">foldChange</span><span class="o">&lt;-</span><span class="n">foldChange</span><span class="m">+0.001</span>
  <span class="n">log2foldChange</span> <span class="o">&lt;-</span> <span class="nf">log2</span><span class="p">(</span><span class="n">foldChange</span><span class="p">)</span>
  <span class="n">fit</span> <span class="o">&lt;-</span> <span class="nf">lmFit</span><span class="p">(</span><span class="n">log2foldChange</span><span class="p">,</span> <span class="n">design</span><span class="p">)</span>
  <span class="n">cont.matrix</span> <span class="o">&lt;-</span> <span class="nf">makeContrasts</span><span class="p">(</span><span class="n">group2</span><span class="o">-</span><span class="n">group1</span><span class="p">,</span> <span class="n">levels</span><span class="o">=</span><span class="n">design</span><span class="p">)</span>
  <span class="n">fit2</span> <span class="o">&lt;-</span> <span class="nf">contrasts.fit</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">cont.matrix</span><span class="p">)</span>
  <span class="n">fit2</span> <span class="o">&lt;-</span> <span class="nf">eBayes</span><span class="p">(</span><span class="n">fit2</span><span class="p">,</span><span class="n">trend</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="n">robust</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>

  <span class="c1">####overall results####</span>
  <span class="n">overall</span><span class="o">=</span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit2</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="kc">Inf</span><span class="p">,</span> <span class="n">sort.by</span><span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span>
  <span class="n">overall</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">foldChange_group_list_df_foldChange</span><span class="p">,</span><span class="n">overall</span><span class="p">)</span>
  <span class="n">overall</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">overall</span><span class="p">,</span><span class="n">group1_treatStopNumBaseMean</span><span class="p">,</span><span class="n">group2_treatStopNumBaseMean</span><span class="p">,</span><span class="n">group1_treatPreRpmFoldBaseMean</span><span class="p">,</span><span class="n">group2_treatPreRpmFoldBaseMean</span><span class="p">,</span><span class="n">group1_preRpmFoldRatioBaseMean</span><span class="p">,</span><span class="n">group2_preRpmFoldRatioBaseMean</span><span class="p">,</span><span class="n">group1_treatAfterRpmFoldBaseMean</span><span class="p">,</span><span class="n">group2_treatAfterRpmFoldBaseMean</span><span class="p">,</span><span class="n">group1_afterRpmFoldRatioBaseMean</span><span class="p">,</span><span class="n">group2_afterRpmFoldRatioBaseMean</span><span class="p">,</span><span class="n">group1_treatStopRatioBaseMean</span><span class="p">,</span><span class="n">group2_treatStopRatioBaseMean</span><span class="p">,</span><span class="n">group1_stopRatioFCBaseMean</span><span class="p">,</span><span class="n">group2_stopRatioFCBaseMean</span><span class="p">)</span>
  <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">overall</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outdir</span><span class="p">,</span><span class="s">&quot;/&quot;</span><span class="p">,</span><span class="n">group2_name</span><span class="p">,</span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="n">group1_name</span><span class="p">,</span><span class="s">&quot;_treat_control_overall.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">overwrite</span> <span class="o">=</span><span class="bp">T</span><span class="p">)</span>

  <span class="n">overall_filt</span><span class="o">&lt;-</span><span class="n">overall</span> <span class="o">%&gt;%</span>
    <span class="nf">rowwise</span><span class="p">()</span> <span class="o">%&gt;%</span>
    <span class="nf">mutate</span><span class="p">(</span>
      <span class="n">aux_condition1</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="nf">c_across</span><span class="p">(</span><span class="n">cols</span> <span class="o">=</span> <span class="nf">ends_with</span><span class="p">(</span><span class="s">&quot;_group1&quot;</span><span class="p">))</span> <span class="o">&gt;</span> <span class="n">mincount</span><span class="p">),</span><span class="c1">#group1 aux_condition1</span>
      <span class="n">aux_condition2</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="nf">c_across</span><span class="p">(</span><span class="n">cols</span> <span class="o">=</span> <span class="nf">ends_with</span><span class="p">(</span><span class="s">&quot;_group1&quot;</span><span class="p">))</span> <span class="o">==</span> <span class="m">0</span><span class="p">),</span><span class="c1">#group1 aux_condition2</span>
      <span class="n">aux_condition3</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="nf">c_across</span><span class="p">(</span><span class="n">cols</span> <span class="o">=</span> <span class="nf">ends_with</span><span class="p">(</span><span class="s">&quot;_group2&quot;</span><span class="p">))</span> <span class="o">&gt;</span> <span class="n">mincount</span><span class="p">),</span><span class="c1">#group2 aux_condition3</span>
      <span class="n">aux_condition4</span> <span class="o">=</span> <span class="nf">sum</span><span class="p">(</span><span class="nf">c_across</span><span class="p">(</span><span class="n">cols</span> <span class="o">=</span> <span class="nf">ends_with</span><span class="p">(</span><span class="s">&quot;_group2&quot;</span><span class="p">))</span> <span class="o">==</span> <span class="m">0</span><span class="p">)</span><span class="c1">#group2 aux_condition4</span>
    <span class="p">)</span> <span class="o">%&gt;%</span>
    <span class="nf">ungroup</span><span class="p">()</span> <span class="o">%&gt;%</span>
    <span class="nf">filter</span><span class="p">((</span><span class="n">aux_condition1</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">aux_condition3</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))</span><span class="o">|</span>
           <span class="p">(</span><span class="n">aux_condition1</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">aux_condition4</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">))</span><span class="o">|</span>
           <span class="c1"># (aux_condition3 == length(group1filelist) &amp; aux_condition1 == length(group2filelist))|</span>
           <span class="p">(</span><span class="n">aux_condition3</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">aux_condition2</span> <span class="o">==</span> <span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)))</span> <span class="o">%&gt;%</span> <span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="nf">starts_with</span><span class="p">(</span><span class="s">&quot;aux_&quot;</span><span class="p">))</span>
  <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">overall_filt</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outdir</span><span class="p">,</span><span class="s">&quot;/&quot;</span><span class="p">,</span><span class="n">group2_name</span><span class="p">,</span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="n">group1_name</span><span class="p">,</span><span class="s">&quot;_treat_control_overall_filt.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">overwrite</span> <span class="o">=</span><span class="bp">T</span><span class="p">)</span>

  <span class="n">overall_filt</span><span class="o">$</span><span class="n">name</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group2-group1-&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">overall_filt</span><span class="o">$</span><span class="n">chrom</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>
  <span class="n">overall_filt_bed</span><span class="o">&lt;-</span><span class="n">overall_filt</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">overall_filt</span><span class="p">),</span><span class="m">5</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)</span><span class="m">+1</span><span class="p">,</span><span class="m">4</span><span class="o">:</span><span class="m">5</span><span class="p">,</span><span class="m">6</span><span class="o">:</span><span class="p">(</span><span class="m">6</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)</span><span class="m">-1</span><span class="p">),(</span><span class="m">5</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span><span class="o">+</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)</span><span class="m">+1+1</span><span class="p">)</span><span class="o">:</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">overall_filt</span><span class="p">)</span><span class="m">-1</span><span class="p">))]</span>
  <span class="nf">write.table</span><span class="p">(</span><span class="n">overall_filt_bed</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outdir</span><span class="p">,</span><span class="s">&quot;/&quot;</span><span class="p">,</span><span class="n">group2_name</span><span class="p">,</span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="n">group1_name</span><span class="p">,</span><span class="s">&quot;_treat_control_overall_filt.bed&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">row.names</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span><span class="n">quote</span> <span class="o">=</span> <span class="bp">F</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="p">}</span>

<span class="nf">print</span><span class="p">(</span><span class="s">&quot;limma program end&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">limma_bed_annotation.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -ne <span class="m">8</span> <span class="o">]]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; uplog2FC downlog2FC pval group_name bedfile bed6file group1filelist group2filelist&#39;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="k">fi</span>

<span class="nv">uplog2FC</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">downlog2FC</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">pval</span><span class="o">=</span><span class="nv">$3</span>
<span class="nv">group_name</span><span class="o">=</span><span class="nv">$4</span>
<span class="nv">bedfile</span><span class="o">=</span><span class="nv">$5</span>
<span class="nv">bed6file</span><span class="o">=</span><span class="nv">$6</span>
<span class="nv">group1filelist</span><span class="o">=</span><span class="nv">$7</span>
<span class="nv">group2filelist</span><span class="o">=</span><span class="nv">$8</span>

<span class="nb">echo</span> <span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
<span class="nv">bedAnnotator_cmd1</span><span class="o">=</span><span class="s2">&quot;</span><span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span><span class="s2">/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="s2">6 -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
<span class="nv">bedAnnotator_cmd2</span><span class="o">=</span><span class="s2">&quot;</span><span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span><span class="s2">/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="nv">$bedfile</span><span class="s2"> -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno_append.bed&quot;</span>
<span class="nb">eval</span> <span class="nv">$bedAnnotator_cmd1</span> <span class="p">&amp;</span>
<span class="nb">eval</span> <span class="nv">$bedAnnotator_cmd2</span> <span class="p">&amp;</span>
<span class="nb">wait</span>

<span class="c1">#add sequence and remove redundancy</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/limma_remove_redundancy.r -l <span class="nv">$uplog2FC</span> -m <span class="nv">$downlog2FC</span> -n <span class="nv">$pval</span> -q <span class="nv">$group_name</span> -i <span class="nv">$group1filelist</span> -j <span class="nv">$group2filelist</span> -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed -g <span class="nv">$bedfile</span> -s <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed -e <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/human.hg38.Pseudo.result.col29.xlsx -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>
<span class="nb">echo</span> -e <span class="s2">&quot;Finished: bedAnnotator done\n&quot;</span>

<span class="nb">echo</span> -e <span class="s2">&quot;bedAnnotator result in </span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>

mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_DGE_volcano_plot.pdf <span class="p">&amp;</span>&gt; /dev/null
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_gene_biotype_piechart.pdf <span class="p">&amp;</span>&gt; /dev/null
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_plot_table.pdf <span class="p">&amp;</span>&gt; /dev/null
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">limma_remove_redundancy.r</span></code></p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/usr/bin/env Rscript</span>
<span class="nf">options</span><span class="p">(</span><span class="n">warn</span><span class="o">=</span><span class="m">-1</span><span class="p">)</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;optparse&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;dplyr&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;stringr&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;ggrepel&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;openxlsx&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;RColorBrewer&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;tidyr&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;gridExtra&quot;</span><span class="p">))</span>


<span class="n">option_list</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-l&quot;</span><span class="p">,</span> <span class="s">&quot;--uplog2FC&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;uplog2FC&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-m&quot;</span><span class="p">,</span> <span class="s">&quot;--downlog2FC&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;downlog2FC&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-n&quot;</span><span class="p">,</span> <span class="s">&quot;--pval&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;pval&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;numeric&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-q&quot;</span><span class="p">,</span> <span class="s">&quot;--group_name&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;group_name&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-i&quot;</span><span class="p">,</span> <span class="s">&quot;--group1filelist&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;group1filelist&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-j&quot;</span><span class="p">,</span> <span class="s">&quot;--group2filelist&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;group2filelist&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-f&quot;</span><span class="p">,</span> <span class="s">&quot;--infile1&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;input file 1 [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-g&quot;</span><span class="p">,</span> <span class="s">&quot;--infile2&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;input file 2 [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-s&quot;</span><span class="p">,</span> <span class="s">&quot;--infile3&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;input file 3 [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-e&quot;</span><span class="p">,</span> <span class="s">&quot;--infile4&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;input file 4 [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-o&quot;</span><span class="p">,</span> <span class="s">&quot;--outfile&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;output file name [default= %default]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">)</span>
<span class="p">);</span>
<span class="n">opt_parser</span> <span class="o">=</span> <span class="nf">OptionParser</span><span class="p">(</span><span class="n">option_list</span><span class="o">=</span><span class="n">option_list</span><span class="p">);</span>
<span class="n">opt</span> <span class="o">=</span> <span class="nf">parse_args</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>

<span class="nf">if </span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">uplog2FC</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">downlog2FC</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">pval</span><span class="p">)</span> <span class="o">||</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group_name</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group1filelist</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">group2filelist</span><span class="p">)</span> <span class="o">||</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">infile1</span><span class="p">)</span> <span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">infile2</span><span class="p">)</span> <span class="o">||</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">infile3</span><span class="p">)</span> <span class="o">||</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">infile4</span><span class="p">)</span> <span class="o">||</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">outfile</span><span class="p">)</span> <span class="p">){</span>
  <span class="nf">print_help</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>
  <span class="nf">stop</span><span class="p">(</span><span class="s">&quot;Please provide -l uplog2FC, -m downlog2FC, -n pval, -g group_name, -i group1filelist, -j group2filelist, -f infile1 -g infile2 and -s infile3, -e infile4, and -o outfile option&quot;</span><span class="p">,</span> <span class="n">call.</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">);</span>
<span class="p">}</span>

<span class="n">uplog2FC</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">uplog2FC</span>
<span class="n">downlog2FC</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">downlog2FC</span>
<span class="n">pval</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">pval</span>
<span class="n">group_name</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group_name</span>
<span class="n">group1filelist</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group1filelist</span>
<span class="n">group1filelist</span><span class="o">&lt;-</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">,</span><span class="s">&quot;,&quot;</span><span class="p">))</span>
<span class="n">group2filelist</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">group2filelist</span>
<span class="n">group2filelist</span><span class="o">&lt;-</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">,</span><span class="s">&quot;,&quot;</span><span class="p">))</span>
<span class="n">anno.biotype.bed.file</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">infile1</span>
<span class="n">input.bed.file</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">infile2</span>
<span class="n">hg38.psiU.SingleSites.bed.file</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">infile3</span>
<span class="n">human.hg38.Pseudo.result.col29.xlsx.file</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">infile4</span>
<span class="n">outFile</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">outfile</span>

<span class="nf">print</span><span class="p">(</span><span class="n">uplog2FC</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">downlog2FC</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">pval</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">group_name</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">anno.biotype.bed.file</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">input.bed.file</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed.file</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">human.hg38.Pseudo.result.col29.xlsx.file</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">outFile</span><span class="p">)</span>

<span class="n">anno.biotype.bed</span><span class="o">&lt;-</span><span class="nf">read.delim</span><span class="p">(</span><span class="n">anno.biotype.bed.file</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="n">head</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="n">anno.biotype.bed</span><span class="o">&lt;-</span><span class="nf">separate</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">anno.biotype.bed</span><span class="p">,</span> <span class="n">col</span> <span class="o">=</span> <span class="n">V7</span><span class="p">,</span> <span class="n">into</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Transcript_id&quot;</span><span class="p">,</span> <span class="s">&quot;Transcript_name&quot;</span><span class="p">,</span><span class="s">&quot;Gene_id&quot;</span><span class="p">,</span><span class="s">&quot;Gene_name&quot;</span><span class="p">,</span><span class="s">&quot;type&quot;</span><span class="p">,</span><span class="s">&quot;Region&quot;</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;\\|&quot;</span><span class="p">)</span>

<span class="c1">#pie chart for all snoRNA-guided RNAs</span>
<span class="n">mRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;mRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;protein_coding&quot;</span><span class="p">))</span>
<span class="n">pseudogene</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;pseudogene&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;rRNA_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;translated_processed_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;unprocessed_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;processed_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;transcribed_processed_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;unitary_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;polymorphic_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;transcribed_unitary_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;TR_V_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;TR_J_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;IG_V_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;IG_C_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;IG_D_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;IG_pseudogene&quot;</span><span class="p">,</span> <span class="s">&quot;IG_J_pseudogene&quot;</span><span class="p">))</span>
<span class="n">lncRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;lncRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;processed_transcript&quot;</span><span class="p">,</span><span class="s">&quot;lincRNA&quot;</span><span class="p">,</span><span class="s">&quot;non_coding&quot;</span><span class="p">,</span><span class="s">&quot;3prime_overlapping_ncRNA&quot;</span><span class="p">,</span><span class="s">&quot;3prime_overlapping_ncrna&quot;</span><span class="p">,</span><span class="s">&quot;sense_intronic&quot;</span><span class="p">,</span><span class="s">&quot;antisense&quot;</span><span class="p">,</span><span class="s">&quot;sense_overlapping&quot;</span><span class="p">,</span><span class="s">&quot;known_ncrna&quot;</span><span class="p">,</span><span class="s">&quot;macro_lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;bidirectional_promoter_lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;retained_intron&quot;</span><span class="p">,</span><span class="s">&quot;TEC&quot;</span><span class="p">))</span>
<span class="n">sncRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;sncRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;snRNA&quot;</span><span class="p">,</span><span class="s">&quot;snoRNA&quot;</span><span class="p">,</span><span class="s">&quot;misc_RNA&quot;</span><span class="p">,</span><span class="s">&quot;miscRNA&quot;</span><span class="p">,</span><span class="s">&quot;miRNA&quot;</span><span class="p">,</span><span class="s">&quot;ribozyme&quot;</span><span class="p">,</span><span class="s">&quot;sRNA&quot;</span><span class="p">,</span><span class="s">&quot;scRNA&quot;</span><span class="p">,</span><span class="s">&quot;scaRNA&quot;</span><span class="p">,</span><span class="s">&quot;srpRNA&quot;</span><span class="p">,</span><span class="s">&quot;tRNA-Deu&quot;</span><span class="p">,</span><span class="s">&quot;tRNA-RTE&quot;</span><span class="p">,</span><span class="s">&quot;piRNA&quot;</span><span class="p">,</span><span class="s">&quot;siRNA&quot;</span><span class="p">))</span>
<span class="n">rRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;rRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;rRNA&quot;</span><span class="p">,</span><span class="s">&quot;Mt_rRNA&quot;</span><span class="p">))</span>
<span class="n">tRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;tRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;tRNA&quot;</span><span class="p">,</span><span class="s">&quot;Mt_tRNA&quot;</span><span class="p">,</span><span class="s">&quot;vaultRNA&quot;</span><span class="p">))</span>
<span class="n">IG_gene</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;IG_gene&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;IG_C_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_D_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_J_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_LV_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_V_gene&quot;</span><span class="p">))</span>
<span class="n">TR_gene</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;TR_gene&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;TR_C_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_J_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_V_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_D_gene&quot;</span><span class="p">))</span>
<span class="n">repeatMasker</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;repeatMasker&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;5S-Deu-L2&quot;</span><span class="p">,</span><span class="s">&quot;Alu&quot;</span><span class="p">,</span><span class="s">&quot;centr&quot;</span><span class="p">,</span><span class="s">&quot;CR1&quot;</span><span class="p">,</span><span class="s">&quot;DNA&quot;</span><span class="p">,</span><span class="s">&quot;DNA?&quot;</span><span class="p">,</span><span class="s">&quot;ERV1&quot;</span><span class="p">,</span><span class="s">&quot;ERV1?&quot;</span><span class="p">,</span><span class="s">&quot;ERVK&quot;</span><span class="p">,</span><span class="s">&quot;ERVL&quot;</span><span class="p">,</span><span class="s">&quot;ERVL?&quot;</span><span class="p">,</span><span class="s">&quot;ERVL-MaLR&quot;</span><span class="p">,</span><span class="s">&quot;Gypsy&quot;</span><span class="p">,</span><span class="s">&quot;Gypsy?&quot;</span><span class="p">,</span><span class="s">&quot;hAT&quot;</span><span class="p">,</span><span class="s">&quot;hAT?&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Ac&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Blackjack&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Charlie&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tag1&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tip100&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tip100?&quot;</span><span class="p">,</span><span class="s">&quot;Helitron&quot;</span><span class="p">,</span><span class="s">&quot;Helitron?&quot;</span><span class="p">,</span><span class="s">&quot;L1&quot;</span><span class="p">,</span><span class="s">&quot;L2&quot;</span><span class="p">,</span><span class="s">&quot;Low_complexity&quot;</span><span class="p">,</span><span class="s">&quot;LTR&quot;</span><span class="p">,</span><span class="s">&quot;LTR?&quot;</span><span class="p">,</span><span class="s">&quot;MIR&quot;</span><span class="p">,</span><span class="s">&quot;MULE-MuDR&quot;</span><span class="p">,</span><span class="s">&quot;nonsense_mediated_decay&quot;</span><span class="p">,</span><span class="s">&quot;non_stop_decay&quot;</span><span class="p">,</span><span class="s">&quot;Penelope&quot;</span><span class="p">,</span><span class="s">&quot;PiggyBac&quot;</span><span class="p">,</span><span class="s">&quot;PiggyBac?&quot;</span><span class="p">,</span><span class="s">&quot;RNA&quot;</span><span class="p">,</span><span class="s">&quot;RTE-BovB&quot;</span><span class="p">,</span><span class="s">&quot;RTE-X&quot;</span><span class="p">,</span><span class="s">&quot;Satellite&quot;</span><span class="p">,</span><span class="s">&quot;Simple_repeat&quot;</span><span class="p">,</span><span class="s">&quot;SVA&quot;</span><span class="p">,</span><span class="s">&quot;TcMar?&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Mariner&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tc2&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tigger&quot;</span><span class="p">,</span><span class="s">&quot;telo&quot;</span><span class="p">,</span><span class="s">&quot;Unknown&quot;</span><span class="p">,</span><span class="s">&quot;acro&quot;</span><span class="p">,</span><span class="s">&quot;Crypton&quot;</span><span class="p">,</span><span class="s">&quot;Dong-R4&quot;</span><span class="p">,</span><span class="s">&quot;I-Jockey&quot;</span><span class="p">,</span><span class="s">&quot;Kolobok&quot;</span><span class="p">,</span><span class="s">&quot;L1-Tx1&quot;</span><span class="p">,</span><span class="s">&quot;Merlin&quot;</span><span class="p">,</span><span class="s">&quot;MULE-MuDR?&quot;</span><span class="p">,</span><span class="s">&quot;PIF-Harbinger&quot;</span><span class="p">,</span><span class="s">&quot;SINE?&quot;</span><span class="p">,</span><span class="s">&quot;TcMar&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Pogo&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tc1&quot;</span><span class="p">))</span>
<span class="n">intergenic</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;intergenic&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;intergenic&quot;</span><span class="p">))</span>
<span class="n">circRNA</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene_biotype</span><span class="o">=</span><span class="s">&quot;circRNA&quot;</span><span class="p">,</span><span class="n">feature_type</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;circRNA&quot;</span><span class="p">))</span>
<span class="n">category</span><span class="o">&lt;-</span><span class="nf">rbind</span><span class="p">(</span><span class="n">mRNA</span><span class="p">,</span><span class="n">pseudogene</span><span class="p">,</span><span class="n">lncRNA</span><span class="p">,</span><span class="n">sncRNA</span><span class="p">,</span><span class="n">rRNA</span><span class="p">,</span><span class="n">tRNA</span><span class="p">,</span><span class="n">IG_gene</span><span class="p">,</span><span class="n">TR_gene</span><span class="p">,</span><span class="n">repeatMasker</span><span class="p">,</span><span class="n">intergenic</span><span class="p">,</span><span class="n">circRNA</span><span class="p">)</span>

<span class="n">anno.biotype.bed</span><span class="o">&lt;-</span><span class="n">anno.biotype.bed</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">category</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;type&quot;</span><span class="o">=</span><span class="s">&quot;feature_type&quot;</span><span class="p">))</span>
<span class="n">anno.biotype.bed</span><span class="o">&lt;-</span><span class="n">anno.biotype.bed</span> <span class="o">%&gt;%</span> <span class="nf">mutate</span><span class="p">(</span><span class="n">type</span> <span class="o">=</span> <span class="nf">ifelse</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">Transcript_id</span><span class="p">)</span> <span class="o">==</span> <span class="s">&quot;intergenic&quot;</span><span class="p">,</span> <span class="s">&quot;intergenic&quot;</span><span class="p">,</span> <span class="nf">as.character</span><span class="p">(</span><span class="n">type</span><span class="p">)))</span>
<span class="n">input.bed</span><span class="o">&lt;-</span><span class="nf">read.delim</span><span class="p">(</span><span class="n">input.bed.file</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="n">probe</span><span class="o">&lt;-</span><span class="nf">which</span><span class="p">(</span><span class="n">input.bed</span><span class="o">$</span><span class="n">V4</span> <span class="o">%in%</span> <span class="n">anno.biotype.bed</span><span class="o">$</span><span class="n">V4</span><span class="p">)</span>
<span class="n">input.bed</span><span class="o">&lt;-</span><span class="n">input.bed</span><span class="p">[</span><span class="n">probe</span><span class="p">,]</span>
<span class="n">add_seq</span> <span class="o">&lt;-</span> <span class="n">anno.biotype.bed</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">input.bed</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;V4&quot;</span><span class="o">=</span><span class="s">&quot;V4&quot;</span><span class="p">))</span>


<span class="n">extendSeq_split</span><span class="o">&lt;-</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">str_split_fixed</span><span class="p">(</span><span class="n">add_seq</span><span class="p">[,</span><span class="s">&quot;V7&quot;</span><span class="p">],</span> <span class="s">&quot;Y&quot;</span><span class="p">,</span> <span class="m">2</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">extendSeq_split</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;extendSeq_bef&quot;</span><span class="p">,</span><span class="s">&quot;extendSeq_aft&quot;</span><span class="p">)</span>
<span class="n">add_seq</span><span class="o">&lt;-</span><span class="nf">cbind</span><span class="p">(</span><span class="n">add_seq</span><span class="p">,</span><span class="n">extendSeq_split</span><span class="p">)</span>
<span class="n">seq_group</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">uniq_seq</span><span class="o">=</span><span class="nf">unique</span><span class="p">(</span><span class="n">add_seq</span><span class="o">$</span><span class="n">extendSeq_aft</span><span class="p">))</span>
<span class="n">seq_group</span><span class="o">$</span><span class="n">seq_id</span><span class="o">&lt;-</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">seq_group</span><span class="o">$</span><span class="n">uniq_seq</span><span class="p">)</span>
<span class="n">add_seq</span> <span class="o">&lt;-</span> <span class="n">add_seq</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">seq_group</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;extendSeq_aft&quot;</span><span class="o">=</span><span class="s">&quot;uniq_seq&quot;</span><span class="p">))</span>

<span class="n">priority</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;rRNA&quot;</span><span class="p">,</span><span class="s">&quot;tRNA&quot;</span><span class="p">,</span><span class="s">&quot;sncRNA&quot;</span><span class="p">,</span><span class="s">&quot;mRNA&quot;</span><span class="p">,</span><span class="s">&quot;lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;circRNA&quot;</span><span class="p">,</span><span class="s">&quot;IG_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_gene&quot;</span><span class="p">,</span><span class="s">&quot;pseudogene&quot;</span><span class="p">,</span><span class="s">&quot;repeatMasker&quot;</span><span class="p">,</span><span class="s">&quot;intergenic&quot;</span><span class="p">)</span>
<span class="n">priority_table</span><span class="o">&lt;-</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">type</span><span class="o">=</span><span class="n">priority</span><span class="p">,</span><span class="n">priority_rank</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="nf">seq</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">priority</span><span class="p">))))</span>
<span class="n">add_seq</span> <span class="o">&lt;-</span> <span class="n">add_seq</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">priority_table</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;gene_biotype&quot;</span><span class="o">=</span><span class="s">&quot;type&quot;</span><span class="p">))</span>
<span class="n">add_seq</span><span class="o">$</span><span class="n">group_id</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group&quot;</span><span class="p">,</span><span class="n">add_seq</span><span class="o">$</span><span class="n">seq_id</span><span class="p">,</span><span class="n">add_seq</span><span class="o">$</span><span class="n">priority_rank</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;-&quot;</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">add_seq</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_add_seq_group.bed&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="nf">write.xlsx</span><span class="p">(</span><span class="n">add_seq</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_add_seq_group.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>


<span class="n">add_seq_group_list</span><span class="o">&lt;-</span><span class="nf">split</span><span class="p">(</span><span class="n">add_seq</span><span class="p">,</span><span class="n">add_seq</span><span class="o">$</span><span class="n">seq_id</span><span class="p">)</span>

<span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">add_seq_group_list</span><span class="p">),</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span>
    <span class="n">add_seq_group_list</span><span class="p">[[</span><span class="n">x</span><span class="p">]]</span><span class="o">&lt;&lt;-</span><span class="nf">head</span><span class="p">(</span><span class="nf">arrange</span><span class="p">(</span><span class="n">add_seq_group_list</span><span class="p">[[</span><span class="n">x</span><span class="p">]],</span><span class="n">priority_rank</span><span class="p">),</span><span class="m">1</span><span class="p">)</span>
    <span class="p">}))</span>

<span class="n">add_seq_group_uniq</span><span class="o">&lt;-</span><span class="nf">do.call</span><span class="p">(</span><span class="n">rbind.data.frame</span><span class="p">,</span> <span class="n">add_seq_group_list</span><span class="p">)</span>
<span class="nf">dim</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="p">)</span>


<span class="nf">colnames</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom&quot;</span><span class="p">,</span><span class="c1">#1</span>
<span class="s">&quot;chromStart&quot;</span><span class="p">,</span><span class="c1">#2</span>
<span class="s">&quot;chromEnd&quot;</span><span class="p">,</span><span class="c1">#3</span>
<span class="s">&quot;name&quot;</span><span class="p">,</span><span class="c1">#4</span>
<span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="c1">#5</span>
<span class="s">&quot;strand&quot;</span><span class="p">,</span><span class="c1">#6</span>
<span class="s">&quot;Transcript_id&quot;</span><span class="p">,</span><span class="c1">#7</span>
<span class="s">&quot;Transcript_name&quot;</span><span class="p">,</span><span class="c1">#8</span>
<span class="s">&quot;Gene_id&quot;</span><span class="p">,</span><span class="c1">#9</span>
<span class="s">&quot;Gene_name&quot;</span><span class="p">,</span><span class="c1">#10</span>
<span class="s">&quot;type&quot;</span><span class="p">,</span><span class="c1">#11</span>
<span class="s">&quot;Region&quot;</span><span class="p">,</span><span class="c1">#12</span>
<span class="s">&quot;gene_feature&quot;</span><span class="p">,</span><span class="c1">#13</span>
<span class="s">&quot;tolBaseNum&quot;</span><span class="p">,</span><span class="c1">#14</span>
<span class="s">&quot;tqueryCov&quot;</span><span class="p">,</span><span class="c1">#15</span>
<span class="s">&quot;tsampCov&quot;</span><span class="p">,</span><span class="c1">#16</span>
<span class="s">&quot;tupDist&quot;</span><span class="p">,</span><span class="c1">#17</span>
<span class="s">&quot;tdownDist&quot;</span><span class="p">,</span><span class="c1">#18</span>
<span class="s">&quot;gene_biotype&quot;</span><span class="p">,</span><span class="c1">#19</span>
<span class="s">&quot;chrom.y&quot;</span><span class="p">,</span><span class="c1">#20</span>
<span class="s">&quot;chromStart.y&quot;</span><span class="p">,</span><span class="c1">#21</span>
<span class="s">&quot;chromEnd.y&quot;</span><span class="p">,</span><span class="c1">#22</span>
<span class="s">&quot;logFC.y&quot;</span><span class="p">,</span><span class="c1">#23</span>
<span class="s">&quot;strand.y&quot;</span><span class="p">,</span><span class="c1">#24</span>
<span class="s">&quot;extendSeq&quot;</span><span class="p">,</span><span class="c1">#25</span>
<span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group1_rep&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group1filelist</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span>
<span class="nf">paste</span><span class="p">(</span><span class="s">&quot;group2_rep&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">group2filelist</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span>
<span class="s">&quot;AveExpr&quot;</span><span class="p">,</span>
<span class="s">&quot;t&quot;</span><span class="p">,</span>
<span class="s">&quot;P.Value&quot;</span><span class="p">,</span>
<span class="s">&quot;adj.P.Val&quot;</span><span class="p">,</span>
<span class="s">&quot;B&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_treatStopNumBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_treatStopNumBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_treatPreRpmFoldBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_treatPreRpmFoldBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_preRpmFoldRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_preRpmFoldRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_treatAfterRpmFoldBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_treatAfterRpmFoldBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_afterRpmFoldRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_afterRpmFoldRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_treatStopRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_treatStopRatioBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group1_stopRatioFCBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;group2_stopRatioFCBaseMean&quot;</span><span class="p">,</span>
<span class="s">&quot;extendSeq_bef&quot;</span><span class="p">,</span>
<span class="s">&quot;extendSeq_aft&quot;</span><span class="p">,</span>
<span class="s">&quot;seq_id&quot;</span><span class="p">,</span>
<span class="s">&quot;priority_rank&quot;</span><span class="p">,</span>
<span class="s">&quot;group_id&quot;</span><span class="p">)</span>
<span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">chrom</span><span class="p">,</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">chromStart</span><span class="p">,</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">chromEnd</span><span class="p">,</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">strand</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>

<span class="c1">#volcano plot visualization</span>
<span class="n">add_seq_group_uniq</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq</span> <span class="o">%&gt;%</span> <span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">logFC</span><span class="p">))</span>

<span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="o">&lt;-</span><span class="nf">as.factor</span><span class="p">(</span>
  <span class="nf">ifelse</span><span class="p">((</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span> <span class="o">&gt;</span> <span class="n">uplog2FC</span> <span class="o">&amp;</span> <span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">P.Value</span> <span class="o">&lt;</span> <span class="n">pval</span><span class="p">),</span><span class="s">&quot;Up_regulated&quot;</span><span class="p">,</span>
         <span class="nf">ifelse</span><span class="p">((</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span> <span class="o">&lt;</span> <span class="n">downlog2FC</span> <span class="o">&amp;</span> <span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">P.Value</span> <span class="o">&lt;</span> <span class="n">pval</span><span class="p">),</span><span class="s">&quot;Down_regulated&quot;</span><span class="p">,</span>
                <span class="nf">ifelse</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">P.Value</span> <span class="o">&gt;</span> <span class="n">pval</span><span class="p">,</span><span class="s">&quot;Not_significant&quot;</span><span class="p">,</span><span class="s">&quot;Not_expected&quot;</span><span class="p">))))</span>
<span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="p">)</span>
<span class="c1"># levels(add_seq_group_uniq$status)&lt;-c(&quot;Down_regulated&quot;,&quot;Up_regulated&quot;,&quot;Not_significant&quot;,&quot;Not_expected&quot;)</span>
<span class="n">mycolors</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="n">Down_regulated</span><span class="o">=</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">6</span><span class="p">,</span><span class="s">&quot;PuBu&quot;</span><span class="p">)[</span><span class="m">6</span><span class="p">],</span><span class="n">Up_regulated</span><span class="o">=</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="s">&quot;Set1&quot;</span><span class="p">)[</span><span class="m">1</span><span class="p">],</span><span class="n">Not_significant</span><span class="o">=</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="s">&quot;Greys&quot;</span><span class="p">)[</span><span class="m">2</span><span class="p">],</span><span class="n">Not_expected</span><span class="o">=</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="s">&quot;YlGn&quot;</span><span class="p">)[</span><span class="m">2</span><span class="p">])</span><span class="c1">#blue/red/grey/green</span>
<span class="c1"># barplot(c(2,5,7,9), col=mycolors)</span>

<span class="n">unique_gene_biotype</span><span class="o">&lt;-</span><span class="nf">unique</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">gene_biotype</span><span class="p">)</span>

<span class="n">top3_Transcript</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="nf">which</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="o">==</span><span class="s">&quot;Up_regulated&quot;</span><span class="p">),</span><span class="m">3</span><span class="p">),</span><span class="nf">tail</span><span class="p">(</span><span class="nf">which</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="o">==</span><span class="s">&quot;Down_regulated&quot;</span><span class="p">),</span><span class="m">3</span><span class="p">))</span>
<span class="c1"># add_seq_group_uniq$Gene_name[setdiff(seq_along(add_seq_group_uniq$Gene_name),top3_gene)]&lt;-&quot;&quot;</span>
<span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">Transcript_name_top</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">Transcript_name</span>
<span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">Transcript_name_top</span><span class="p">[</span><span class="nf">setdiff</span><span class="p">(</span><span class="nf">seq_along</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">Transcript_name_top</span><span class="p">),</span><span class="n">top3_Transcript</span><span class="p">)]</span><span class="o">&lt;-</span><span class="s">&quot;&quot;</span>

<span class="nf">if</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">!=</span> <span class="m">0</span><span class="p">){</span>
  <span class="c1"># Set it globally:</span>
  <span class="nf">options</span><span class="p">(</span><span class="n">ggrepel.max.overlaps</span> <span class="o">=</span> <span class="kc">Inf</span><span class="p">)</span>
  <span class="n">p_volcano</span> <span class="o">&lt;-</span> <span class="nf">ggplot</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="p">,</span>
              <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">logFC</span><span class="p">,</span>
                  <span class="n">y</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="n">P.Value</span><span class="p">),</span>
                  <span class="n">colour</span><span class="o">=</span><span class="nf">str_wrap</span><span class="p">(</span><span class="n">status</span><span class="p">,</span><span class="n">width</span> <span class="o">=</span> <span class="m">2</span><span class="p">),</span>
                  <span class="n">label</span> <span class="o">=</span><span class="n">Transcript_name_top</span><span class="p">)</span> <span class="p">)</span> <span class="o">+</span>
    <span class="nf">geom_jitter</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="m">0.6</span><span class="p">)</span>
  <span class="n">p_volcano</span> <span class="o">&lt;-</span> <span class="n">p_volcano</span> <span class="o">+</span> <span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycolors</span><span class="p">)</span>
  <span class="n">p_volcano</span> <span class="o">&lt;-</span> <span class="n">p_volcano</span> <span class="o">+</span> <span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">downlog2FC</span><span class="p">,</span><span class="n">uplog2FC</span><span class="p">),</span> <span class="n">linetype</span><span class="o">=</span><span class="s">&quot;longdash&quot;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="m">0.2</span><span class="p">)</span>
  <span class="n">p_volcano</span> <span class="o">&lt;-</span> <span class="n">p_volcano</span> <span class="o">+</span> <span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">pval</span><span class="p">)),</span> <span class="n">linetype</span><span class="o">=</span><span class="s">&quot;longdash&quot;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="m">0.2</span><span class="p">)</span>
  <span class="n">p_volcano</span><span class="o">&lt;-</span><span class="n">p_volcano</span><span class="o">+</span>
    <span class="c1"># labs(title = &quot;Group2 - Group1&quot;)+</span>
    <span class="nf">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="n">group_name</span><span class="p">)</span><span class="o">+</span>
    <span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;log2FC&quot;</span><span class="p">)</span><span class="o">+</span>
    <span class="nf">theme_bw</span><span class="p">()</span><span class="o">+</span>
    <span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span> <span class="o">=</span> <span class="s">&quot;top&quot;</span><span class="p">,</span>
          <span class="n">plot.title</span><span class="o">=</span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="o">=</span><span class="m">0.5</span><span class="p">),</span>
          <span class="n">legend.title</span> <span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">legend.key.width</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">1.5</span><span class="p">,</span> <span class="s">&quot;cm&quot;</span><span class="p">),</span>
          <span class="n">legend.key.size</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span> <span class="s">&quot;cm&quot;</span><span class="p">),</span>
          <span class="n">legend.spacing.x</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="s">&quot;cm&quot;</span><span class="p">),</span>
          <span class="n">legend.text.align</span> <span class="o">=</span> <span class="m">0.7</span>
          <span class="c1"># plot.margin=unit(c(1,1,1,1),units=&quot;cm&quot;)</span>
          <span class="p">)</span><span class="o">+</span>
    <span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="o">-</span><span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span><span class="p">)),</span><span class="nf">max</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span><span class="p">)),</span> <span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span><span class="p">)),</span><span class="nf">max</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">logFC</span><span class="p">))))</span><span class="o">+</span>
    <span class="nf">guides</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="nf">guide_legend</span><span class="p">(</span><span class="n">nrow</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="n">ncol</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="n">byrow</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span>

  <span class="n">p_volcano_text</span><span class="o">&lt;-</span><span class="n">p_volcano</span><span class="o">+</span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span>
                            <span class="n">family</span><span class="o">=</span><span class="s">&quot;sans&quot;</span><span class="p">,</span>
                            <span class="n">force</span> <span class="o">=</span> <span class="m">15</span><span class="p">)</span>


  <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_DGE_volcano_plot.pdf&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">width</span><span class="o">=</span><span class="m">8</span><span class="p">,</span><span class="n">height</span><span class="o">=</span><span class="m">8</span><span class="p">)</span>
  <span class="nf">print</span><span class="p">(</span><span class="n">p_volcano_text</span><span class="p">)</span>
  <span class="nf">dev.off</span><span class="p">()</span>
<span class="p">}</span>

<span class="nf">write.xlsx</span><span class="p">(</span><span class="n">add_seq_group_uniq</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_add_seq_group_uniq.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
<span class="n">add_seq_group_uniq_filt</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq</span><span class="p">[</span><span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="o">==</span><span class="s">&quot;Up_regulated&quot;</span> <span class="o">|</span> <span class="n">add_seq_group_uniq</span><span class="o">$</span><span class="n">status</span><span class="o">==</span><span class="s">&quot;Down_regulated&quot;</span><span class="p">,]</span>

<span class="nf">if</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">!=</span> <span class="m">0</span><span class="p">){</span>

  <span class="nf">print</span><span class="p">(</span><span class="s">&quot;pie plot for gene_biotype&quot;</span><span class="p">)</span>
  <span class="n">pseudoU_anno_genetype_num</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">gene_biotype</span><span class="p">)),</span><span class="nf">desc</span><span class="p">(</span><span class="n">Freq</span><span class="p">))</span>
  <span class="nf">write.table</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_anno_gene_biotype_num.sort&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
  <span class="n">percent</span><span class="o">&lt;-</span><span class="nf">round</span><span class="p">(</span><span class="m">100</span><span class="o">*</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">),</span><span class="m">2</span><span class="p">)</span>
  <span class="n">percent</span> <span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;(&#39;</span><span class="p">,</span><span class="n">percent</span><span class="p">,</span> <span class="s">&quot;%&quot;</span><span class="p">,</span> <span class="s">&quot;, &quot;</span><span class="p">,</span> <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">,</span><span class="s">&#39;)&#39;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
  <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">percent</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
  <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">)</span>
  <span class="n">mycol</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Set3&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Paired&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span> <span class="s">&quot;Spectral&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span> <span class="s">&quot;Set1&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Set2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Pastel1&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Pastel2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Dark2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Accent&quot;</span><span class="p">))</span>
  <span class="n">pie_plot</span><span class="o">&lt;-</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">pseudoU_anno_genetype_num</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;Content&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">Freq</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">Var1</span><span class="p">))</span> <span class="o">+</span>
      <span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span> <span class="o">=</span> <span class="s">&#39;identity&#39;</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="s">&#39;stack&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">coord_polar</span><span class="p">(</span><span class="n">theta</span> <span class="o">=</span> <span class="s">&#39;y&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span> <span class="o">+</span>
      <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">panel.background</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.title</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s">&quot;gene biotype&quot;</span><span class="p">))</span><span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycol</span><span class="p">)</span>

  <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_gene_biotype_piechart.pdf&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
  <span class="nf">print</span><span class="p">(</span><span class="n">pie_plot</span><span class="p">)</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>

  <span class="n">pseudoU_anno_genetype_num</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">gene_feature</span><span class="p">)),</span><span class="nf">desc</span><span class="p">(</span><span class="n">Freq</span><span class="p">))</span>
  <span class="n">percent</span><span class="o">&lt;-</span><span class="nf">round</span><span class="p">(</span><span class="m">100</span><span class="o">*</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">),</span><span class="m">2</span><span class="p">)</span>
  <span class="n">percent</span> <span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;(&#39;</span><span class="p">,</span><span class="n">percent</span><span class="p">,</span> <span class="s">&quot;%&quot;</span><span class="p">,</span> <span class="s">&quot;, &quot;</span><span class="p">,</span> <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">,</span><span class="s">&#39;)&#39;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
  <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">percent</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
  <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">pseudoU_anno_genetype_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">)</span>
  <span class="n">mycol</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Set3&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Paired&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span> <span class="s">&quot;Spectral&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span> <span class="s">&quot;Set1&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Set2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Pastel1&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Pastel2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Dark2&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Accent&quot;</span><span class="p">))</span>
  <span class="n">pie_plot</span><span class="o">&lt;-</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">pseudoU_anno_genetype_num</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;Content&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">Freq</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">Var1</span><span class="p">))</span> <span class="o">+</span>
      <span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span> <span class="o">=</span> <span class="s">&#39;identity&#39;</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="s">&#39;stack&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">coord_polar</span><span class="p">(</span><span class="n">theta</span> <span class="o">=</span> <span class="s">&#39;y&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span> <span class="o">+</span>
      <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">panel.background</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.title</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
        <span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
        <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s">&quot;gene feature&quot;</span><span class="p">))</span><span class="o">+</span>
        <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycol</span><span class="p">)</span>

  <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_gene_feature_piechart.pdf&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
  <span class="nf">print</span><span class="p">(</span><span class="n">pie_plot</span><span class="p">)</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>

  <span class="nf">write.table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_add_seq_group_uniq_filt.bed&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
  <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_add_seq_group_uniq_filt.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
  <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_anno_group_redundance.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

  <span class="n">add_seq_group_uniq_filt_high_confid</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">((</span><span class="n">group1_treatPreRpmFoldBaseMean</span> <span class="o">&gt;</span><span class="m">4</span> <span class="o">|</span> <span class="n">group2_treatPreRpmFoldBaseMean</span> <span class="o">&gt;</span><span class="m">4</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">group1_treatAfterRpmFoldBaseMean</span> <span class="o">&gt;</span><span class="m">4</span> <span class="o">|</span> <span class="n">group2_treatAfterRpmFoldBaseMean</span> <span class="o">&gt;</span><span class="m">4</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">group1_stopRatioFCBaseMean</span> <span class="o">&gt;</span><span class="m">4</span> <span class="o">|</span> <span class="n">group2_stopRatioFCBaseMean</span> <span class="o">&gt;</span><span class="m">4</span><span class="p">))</span>
  <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt_high_confid</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_anno_group_redundance_high_confid.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>


  <span class="n">plot_table</span><span class="o">&lt;-</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">gene_feature</span><span class="p">))</span>
  <span class="nf">colnames</span><span class="p">(</span><span class="n">plot_table</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Gene Feature&quot;</span><span class="p">,</span><span class="s">&quot;Num&quot;</span><span class="p">)</span>
  <span class="n">to_plot</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="n">plot_table</span><span class="p">,</span><span class="nf">desc</span><span class="p">(</span><span class="n">Num</span><span class="p">))</span>
  <span class="n">tt3</span> <span class="o">&lt;-</span> <span class="nf">ttheme_minimal</span><span class="p">(</span>
  <span class="n">core</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">bg_params</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="n">blues9</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">],</span> <span class="n">col</span><span class="o">=</span><span class="kc">NA</span><span class="p">),</span>
            <span class="n">fg_params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">fontface</span><span class="o">=</span><span class="m">3</span><span class="p">)),</span>
  <span class="n">colhead</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">fg_params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">col</span><span class="o">=</span><span class="s">&quot;navyblue&quot;</span><span class="p">,</span> <span class="n">fontface</span><span class="o">=</span><span class="m">4L</span><span class="p">)),</span>
  <span class="n">rowhead</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">fg_params</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">col</span><span class="o">=</span><span class="s">&quot;orange&quot;</span><span class="p">,</span> <span class="n">fontface</span><span class="o">=</span><span class="m">3L</span><span class="p">)))</span>

  <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span> <span class="s">&#39;_plot_table.pdf&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">width</span> <span class="o">=</span> <span class="m">7</span><span class="p">,</span> <span class="n">height</span> <span class="o">=</span> <span class="m">7</span><span class="p">)</span> <span class="c1"># Open a new pdf file</span>
  <span class="nf">grid.arrange</span><span class="p">(</span>
  <span class="nf">tableGrob</span><span class="p">(</span><span class="n">to_plot</span><span class="p">,</span> <span class="n">theme</span><span class="o">=</span><span class="n">tt3</span><span class="p">),</span>
  <span class="n">nrow</span><span class="o">=</span><span class="m">1</span><span class="p">)</span>
  <span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span> <span class="c1"># Close the file</span>

  <span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt</span><span class="p">[</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">gene_biotype</span><span class="o">%in%</span><span class="s">&quot;rRNA&quot;</span><span class="p">,]</span>
  <span class="nf">if</span><span class="p">(</span><span class="nf">dim</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">!=</span> <span class="m">0</span><span class="p">){</span>
    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;pie plot for RMBase evidence&quot;</span><span class="p">)</span>
    <span class="c1">#read human.hg38.Pseudo.result.col29.xlsx</span>
    <span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Detecting novel psi (human.hg38.Pseudo.result.col29.xlsx: all known pseudouridylation sites download from RMBase)\n&quot;</span><span class="p">)</span>
    <span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="o">&lt;-</span><span class="nf">read.xlsx</span><span class="p">(</span><span class="n">human.hg38.Pseudo.result.col29.xlsx.file</span><span class="p">)</span><span class="c1">#&quot;human.hg38.Pseudo.result.col29.xlsx&quot;</span>
    <span class="n">human.hg38.Pseudo.result.col29.xlsx.tab</span><span class="o">&lt;-</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="o">$</span><span class="n">uniq_id</span><span class="p">))</span>
    <span class="nf">colnames</span><span class="p">(</span><span class="n">human.hg38.Pseudo.result.col29.xlsx.tab</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;known&quot;</span><span class="p">,</span><span class="s">&quot;hit&quot;</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">human.hg38.Pseudo.result.col29.xlsx.tab</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
    <span class="n">novel_psi</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt</span><span class="p">[</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">==</span><span class="s">&quot;FALSE&quot;</span><span class="p">,]</span>
    <span class="n">common_psi</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt</span><span class="p">[</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">==</span><span class="s">&quot;TRUE&quot;</span><span class="p">,]</span>
    <span class="n">common_psi</span><span class="o">&lt;-</span><span class="n">common_psi</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;uniq_id&quot;</span><span class="o">=</span><span class="s">&quot;uniq_id&quot;</span><span class="p">))</span>
    <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">novel_psi</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;human.hg38.Pseudo.result.col29_novel.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
    <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">common_psi</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;human.hg38.Pseudo.result.col29_common.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

    <span class="n">known_num</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">human.hg38.Pseudo.result.col29.xlsx</span><span class="o">$</span><span class="n">uniq_id</span><span class="p">)),</span><span class="nf">desc</span><span class="p">(</span><span class="n">Freq</span><span class="p">))</span>
    <span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="o">&lt;-</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span><span class="s">&quot;TRUE&quot;</span><span class="p">,</span><span class="s">&quot;Known&quot;</span><span class="p">)</span>
    <span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="o">&lt;-</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span><span class="s">&quot;FALSE&quot;</span><span class="p">,</span><span class="s">&quot;Novel&quot;</span><span class="p">)</span>
    <span class="n">percent</span><span class="o">&lt;-</span><span class="nf">round</span><span class="p">(</span><span class="m">100</span><span class="o">*</span><span class="n">known_num</span><span class="o">$</span><span class="n">Freq</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">known_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">),</span><span class="m">2</span><span class="p">)</span>
    <span class="n">percent</span> <span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;(&#39;</span><span class="p">,</span><span class="n">percent</span><span class="p">,</span> <span class="s">&quot;%&quot;</span><span class="p">,</span> <span class="s">&quot;, &quot;</span><span class="p">,</span> <span class="n">known_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">,</span><span class="s">&#39;)&#39;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
    <span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">percent</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
    <span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">known_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">)</span>
    <span class="n">mycol</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Set3&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Paired&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span> <span class="s">&quot;Spectral&quot;</span><span class="p">))</span>
    <span class="n">pie_plot</span><span class="o">&lt;-</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">known_num</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;Content&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">Freq</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">Var1</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span> <span class="o">=</span> <span class="s">&#39;identity&#39;</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="s">&#39;stack&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">coord_polar</span><span class="p">(</span><span class="n">theta</span> <span class="o">=</span> <span class="s">&#39;y&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">theme</span><span class="p">(</span><span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">panel.background</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.title</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s">&quot;Total Known/Novel psi&quot;</span><span class="p">))</span><span class="o">+</span>
          <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycol</span><span class="p">)</span>

    <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_known_novel_psi_piechart.pdf&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">pie_plot</span><span class="p">)</span>
    <span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>


    <span class="nf">print</span><span class="p">(</span><span class="s">&quot;pie plot for known rRNA evidence&quot;</span><span class="p">)</span>
    <span class="c1">#read hg38.psiU.SingleSites.bed</span>
    <span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt</span> <span class="p">[</span><span class="n">add_seq_group_uniq_filt</span><span class="o">$</span><span class="n">gene_biotype</span><span class="o">==</span><span class="s">&quot;rRNA&quot;</span> <span class="p">,]</span>
    <span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Detecting novel rRNA psi (hg38.psiU.SingleSites.bed: all known pseudouridylation sites in rRNA)\n&quot;</span><span class="p">)</span>
    <span class="n">hg38.psiU.SingleSites.bed</span><span class="o">&lt;-</span><span class="nf">read.table</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed.file</span><span class="p">,</span><span class="n">head</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span><span class="c1">#&quot;hg38.psiU.SingleSites.bed&quot;</span>
    <span class="nf">colnames</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom&quot;</span><span class="p">,</span><span class="s">&quot;chromStart&quot;</span><span class="p">,</span><span class="s">&quot;chromEnd&quot;</span><span class="p">,</span><span class="s">&quot;rRNA_anno&quot;</span><span class="p">,</span><span class="s">&quot;score&quot;</span><span class="p">,</span><span class="s">&quot;strand&quot;</span><span class="p">)</span>
    <span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">chrom</span><span class="p">,</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">chromStart</span><span class="p">,</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">chromEnd</span><span class="p">,</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">strand</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>
    <span class="n">hg38.psiU.SingleSites.bed.tab</span><span class="o">&lt;-</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">uniq_id</span><span class="p">))</span>
    <span class="nf">colnames</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed.tab</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;known&quot;</span><span class="p">,</span><span class="s">&quot;hit&quot;</span><span class="p">)</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed.tab</span><span class="p">,</span> <span class="n">row.names</span> <span class="o">=</span> <span class="bp">F</span><span class="p">)</span>
    <span class="n">novel_rRNA_psi</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="p">[</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">==</span><span class="s">&quot;FALSE&quot;</span><span class="p">,]</span>
    <span class="n">common_rRNA_psi</span><span class="o">&lt;-</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="p">[</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">==</span><span class="s">&quot;TRUE&quot;</span><span class="p">,]</span>
    <span class="n">common_rRNA_psi</span><span class="o">&lt;-</span><span class="n">common_rRNA_psi</span> <span class="o">%&gt;%</span> <span class="nf">left_join</span><span class="p">(</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="p">,</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;uniq_id&quot;</span><span class="o">=</span><span class="s">&quot;uniq_id&quot;</span><span class="p">))</span>
    <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">novel_rRNA_psi</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;hg38.psiU.SingleSites_novel.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
    <span class="nf">write.xlsx</span><span class="p">(</span><span class="n">common_rRNA_psi</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;hg38.psiU.SingleSites_common.xlsx&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span> <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

    <span class="n">known_rRNA_num</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">table</span><span class="p">(</span><span class="n">add_seq_group_uniq_filt_rRNA</span><span class="o">$</span><span class="n">uniq_id</span><span class="o">%in%</span><span class="n">hg38.psiU.SingleSites.bed</span><span class="o">$</span><span class="n">uniq_id</span><span class="p">)),</span><span class="nf">desc</span><span class="p">(</span><span class="n">Freq</span><span class="p">))</span>
    <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="o">&lt;-</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span><span class="s">&quot;TRUE&quot;</span><span class="p">,</span><span class="s">&quot;Known&quot;</span><span class="p">)</span>
    <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="o">&lt;-</span><span class="nf">str_replace</span><span class="p">(</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span><span class="s">&quot;FALSE&quot;</span><span class="p">,</span><span class="s">&quot;Novel&quot;</span><span class="p">)</span>
    <span class="n">percent</span><span class="o">&lt;-</span><span class="nf">round</span><span class="p">(</span><span class="m">100</span><span class="o">*</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Freq</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">),</span><span class="m">2</span><span class="p">)</span>
    <span class="n">percent</span> <span class="o">&lt;-</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;(&#39;</span><span class="p">,</span><span class="n">percent</span><span class="p">,</span> <span class="s">&quot;%&quot;</span><span class="p">,</span> <span class="s">&quot;, &quot;</span><span class="p">,</span> <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Freq</span><span class="p">,</span><span class="s">&#39;)&#39;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&quot;&quot;</span><span class="p">)</span>
    <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">paste</span><span class="p">(</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">percent</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span>
    <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span> <span class="o">&lt;-</span> <span class="nf">factor</span><span class="p">(</span><span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">,</span> <span class="n">levels</span> <span class="o">=</span> <span class="n">known_rRNA_num</span><span class="o">$</span><span class="n">Var1</span><span class="p">)</span>
    <span class="n">mycol</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Set3&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">12</span><span class="p">,</span> <span class="s">&quot;Paired&quot;</span><span class="p">),</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span> <span class="s">&quot;Spectral&quot;</span><span class="p">))</span>
    <span class="n">pie_plot</span><span class="o">&lt;-</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">known_rRNA_num</span><span class="p">,</span> <span class="n">mapping</span> <span class="o">=</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;Content&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">Freq</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">Var1</span><span class="p">))</span> <span class="o">+</span>
        <span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span> <span class="o">=</span> <span class="s">&#39;identity&#39;</span><span class="p">,</span> <span class="n">position</span> <span class="o">=</span> <span class="s">&#39;stack&#39;</span><span class="p">,</span> <span class="n">width</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span> <span class="o">+</span> <span class="nf">coord_polar</span><span class="p">(</span><span class="n">theta</span> <span class="o">=</span> <span class="s">&#39;y&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="nf">labs</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">,</span> <span class="n">title</span> <span class="o">=</span> <span class="s">&#39;&#39;</span><span class="p">)</span> <span class="o">+</span>
        <span class="nf">theme</span><span class="p">(</span><span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">theme</span><span class="p">(</span><span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">theme</span><span class="p">(</span><span class="n">panel.grid</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">panel.background</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.text</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.title</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
          <span class="n">axis.ticks</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">())</span> <span class="o">+</span>
          <span class="nf">guides</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="nf">guide_legend</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s">&quot;rRNA Known/Novel psi&quot;</span><span class="p">))</span><span class="o">+</span>
          <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycol</span><span class="p">)</span>

    <span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile</span><span class="p">,</span><span class="s">&quot;_known_novel_rRNA_psi_piechart.pdf&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">))</span>
    <span class="nf">print</span><span class="p">(</span><span class="n">pie_plot</span><span class="p">)</span>
    <span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
    <span class="p">}</span><span class="n">else</span><span class="p">{</span>
      <span class="nf">print</span><span class="p">(</span><span class="s">&quot;No known/novel rRNA detected!&quot;</span><span class="p">)</span>
    <span class="p">}</span>

<span class="p">}</span><span class="n">else</span><span class="p">{</span>
  <span class="nf">print</span><span class="p">(</span><span class="s">&quot;No differentially expressed psi-sites detected!&quot;</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="optional-parameters">
<h2><a class="toc-backref" href="#id3">Optional Parameters</a><a class="headerlink" href="#optional-parameters" title="Permalink to this heading">¶</a></h2>
<section id="up-log2fc">
<h3><a class="toc-backref" href="#id4">up-log2FC</a><a class="headerlink" href="#up-log2fc" title="Permalink to this heading">¶</a></h3>
<p>Users can customize the up-regulated fold change threshold by setting <code class="docutils literal notranslate"><span class="pre">up-log2FC</span></code>, for example, if <code class="docutils literal notranslate"><span class="pre">up-log2FC</span></code> is set as <code class="docutils literal notranslate"><span class="pre">1.00</span></code>, then result with log2(fold change) &gt; 1.00 is retained.</p>
</section>
<section id="down-log2fc">
<h3><a class="toc-backref" href="#id5">down-log2FC</a><a class="headerlink" href="#down-log2fc" title="Permalink to this heading">¶</a></h3>
<p>Users can customize the down-regulated fold change threshold by setting <code class="docutils literal notranslate"><span class="pre">down-log2FC</span></code>, for example, if <code class="docutils literal notranslate"><span class="pre">down-log2FC</span></code> is set as <code class="docutils literal notranslate"><span class="pre">-1.00</span></code>, then result with log2(fold change) &lt; -1.00 is retained.</p>
</section>
<section id="p-value">
<h3><a class="toc-backref" href="#id6">P.Value</a><a class="headerlink" href="#p-value" title="Permalink to this heading">¶</a></h3>
<p>Users can customize the statistical test threshold by setting <code class="docutils literal notranslate"><span class="pre">P.Value</span></code>, for example, if <code class="docutils literal notranslate"><span class="pre">P.Value</span></code> is set as <code class="docutils literal notranslate"><span class="pre">0.05</span></code>, then result with p value &lt; 0.05 is retained.</p>
</section>
<section id="min-fold">
<h3><a class="toc-backref" href="#id7">min-fold</a><a class="headerlink" href="#min-fold" title="Permalink to this heading">¶</a></h3>
<p>Users can customize the minimum stopRatioFC threshold of each group by setting <code class="docutils literal notranslate"><span class="pre">min-fold</span></code>, for example, if <code class="docutils literal notranslate"><span class="pre">min-fold</span></code> is set as <code class="docutils literal notranslate"><span class="pre">2.00</span></code>, then result with minimum <code class="docutils literal notranslate"><span class="pre">stopRatioFC</span> <span class="pre">&gt;</span> <span class="pre">2.00</span></code> (each sample) is retained.</p>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id8">Output</a><a class="headerlink" href="#output" title="Permalink to this heading">¶</a></h2>
<section id="information">
<h3><a class="toc-backref" href="#id9">Information</a><a class="headerlink" href="#information" title="Permalink to this heading">¶</a></h3>
<p>Result with <code class="docutils literal notranslate"><span class="pre">_group_redundance_high_confid.xlsx</span></code> suffix is the final differential modification result, which is a filter result of <code class="docutils literal notranslate"><span class="pre">_group_redundance.xlsx</span></code> (filter by <code class="docutils literal notranslate"><span class="pre">Optional</span> <span class="pre">Parameters</span></code>).</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir
$ tree -L <span class="m">1</span>
.
├── B11_B12_svm_all.bed
├── B1_B2_svm_all.bed
├── B3_B4_svm_all.bed
├── B5_B6_svm_all.bed
├── B7_B8_svm_all.bed
├── B9_B10_svm_all.bed
├── polyaRNA_Day4-polyaRNA_Day0_limma_bed_annotation.log
├── polyaRNA_Day4-polyaRNA_Day0_limma.log
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_add_seq_group.bed
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_add_seq_group_uniq_filt.bed
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_add_seq_group_uniq_filt.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_add_seq_group_uniq.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_add_seq_group.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_anno_append.bed
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_anno.bed
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_anno_gene_biotype_num.sort
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_anno_group_redundance_high_confid.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_anno_group_redundance.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt.bed
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt.bed6
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_DGE_volcano_plot.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_gene_biotype_piechart.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_gene_feature_piechart.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filthg38.psiU.SingleSites_common.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filthg38.psiU.SingleSites_novel.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filthuman.hg38.Pseudo.result.col29_common.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filthuman.hg38.Pseudo.result.col29_novel.xlsx
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_known_novel_psi_piechart.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_known_novel_rRNA_psi_piechart.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt_plot_table.pdf
├── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall_filt.xlsx
└── polyaRNA_Day4-polyaRNA_Day0_treat_control_overall.xlsx

<span class="m">0</span> directories, <span class="m">32</span> files
</pre></div>
</div>
</section>
<section id="diagram">
<h3><a class="toc-backref" href="#id10">Diagram</a><a class="headerlink" href="#diagram" title="Permalink to this heading">¶</a></h3>
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_volcano_plot.pdf</span></code> is a volcano plot of Ψ-sites differential modification on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/Differential_Modification_volcano_plot.png" src="../_images/Differential_Modification_volcano_plot.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_pseUdiff_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="../index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Ψ-sites Differential Modification</a><ul>
<li><a class="reference internal" href="#input">Input</a></li>
<li><a class="reference internal" href="#compare-sites-modification-abundance">Compare Ψ-sites modification abundance</a></li>
<li><a class="reference internal" href="#optional-parameters">Optional Parameters</a><ul>
<li><a class="reference internal" href="#up-log2fc">up-log2FC</a></li>
<li><a class="reference internal" href="#down-log2fc">down-log2FC</a></li>
<li><a class="reference internal" href="#p-value">P.Value</a></li>
<li><a class="reference internal" href="#min-fold">min-fold</a></li>
</ul>
</li>
<li><a class="reference internal" href="#output">Output</a><ul>
<li><a class="reference internal" href="#information">Information</a></li>
<li><a class="reference internal" href="#diagram">Diagram</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html"
                          title="previous chapter">Ψ-sites SNPs and SNVs Searching</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="%CE%A8-sites%20Functional%20Enrichment.html"
                          title="next chapter">Ψ-sites Functional Enrichment</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/content/Ψ-sites Differential Modification.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20Functional%20Enrichment.html" title="Ψ-sites Functional Enrichment"
             >next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html" title="Ψ-sites SNPs and SNVs Searching"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Differential Modification</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 5.1.1.
    </div>
  </body>
</html>