<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Ψ-sites Distribution &mdash; psiFinder v1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Codon Preference" href="%CE%A8-sites%20Codon%20Preference.html" />
    <link rel="prev" title="Ψ-sites Annotation" href="%CE%A8-sites%20Annotation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> psiFinder
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Ψ-sites Distribution</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l2"><a class="reference internal" href="#profile-sites-distribution">Profile Ψ-sites distribution</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sites-file">Ψ-sites file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Target%20Prediction.html">Ψ-sites Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">psiFinder</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Ψ-sites Distribution</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Ψ-sites Distribution.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="sites-distribution">
<h1>Ψ-sites Distribution<a class="headerlink" href="#sites-distribution" title="Permalink to this heading"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#profile-sites-distribution" id="id2">Profile Ψ-sites distribution</a></p>
<ul>
<li><p><a class="reference internal" href="#sites-file" id="id3">Ψ-sites file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id4">Output</a></p></li>
</ul>
</div>
<p>We utilize <a class="reference external" href="https://github.com/olarerin/metaPlotR">MetaPlotR</a> to generate metagene plot for Ψ-sites distribution profile.</p>
<img alt="../_images/MetaGene.png" src="../_images/MetaGene.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading"></a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format, genome fasta file, genome annotation file, and transcript annotation file (the same file for bedAnnotator <code class="docutils literal notranslate"><span class="pre">--anno</span></code> input).</p>
</section>
<section id="profile-sites-distribution">
<h2><a class="toc-backref" href="#id2">Profile Ψ-sites distribution</a><a class="headerlink" href="#profile-sites-distribution" title="Permalink to this heading"></a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">metagene.sh</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">metagene.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

usage<span class="o">()</span> <span class="o">{</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; -i bedfile -g genome -a gtf [option] &#39;</span>
    <span class="nb">echo</span> <span class="s1">&#39; -A &lt;bedfile&gt; : bed6 file for annotating&#39;</span>
    <span class="nb">exit</span> -1
<span class="o">}</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">genome</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">gtf</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">annofile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>

<span class="k">while</span> <span class="nb">getopts</span> <span class="s1">&#39;i:g:a:A:&#39;</span> OPT<span class="p">;</span> <span class="k">do</span>
    <span class="k">case</span> <span class="nv">$OPT</span> <span class="k">in</span>
        i<span class="o">)</span> <span class="nv">bedfile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        g<span class="o">)</span> <span class="nv">genome</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        a<span class="o">)</span> <span class="nv">gtf</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        A<span class="o">)</span> <span class="nv">annofile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        h<span class="o">)</span> usage<span class="p">;;</span>
        ?<span class="o">)</span> usage<span class="p">;;</span>
    <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">echo</span> <span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
bedAnnotator --anno <span class="si">${</span><span class="nv">annofile</span><span class="si">}</span> --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed <span class="m">2</span>&gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.log <span class="c1"># hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6</span>

<span class="c1">#get data for metagene plot</span>
<span class="nb">echo</span> -e <span class="s2">&quot;get metagene data&quot;</span>
<span class="k">if</span> <span class="o">[</span> ! -f <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed <span class="o">]</span>
    <span class="k">then</span>
    <span class="k">if</span> <span class="o">[</span> ! -f <span class="s2">&quot;</span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">.genePred&quot;</span>  <span class="o">]</span>
    <span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate genePredfile&quot;</span>
    gtfToGenePred -genePredExt -geneNameAsName2 <span class="nv">$gtf</span> <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred
    <span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;genePredfile exist&quot;</span>
    <span class="k">fi</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate metagene annotation bed&quot;</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;perl </span><span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span><span class="s2">/make_annot_bed.pl --genomeDir ./script/genome/ --genePred </span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">.genePred &gt; </span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">_annot.bed&quot;</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/make_annot_bed.pl --genomeDir ./script/genome/ --genePred <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed

    bedtools sort -i <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed
    rm <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed

    <span class="nb">echo</span> <span class="s2">&quot;generate region sizes data&quot;</span>
    <span class="nb">echo</span> -e <span class="s2">&quot;perl </span><span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span><span class="s2">/size_of_cds_utrs.pl --annot </span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">_annot_sorted.bed &gt;</span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">_region_sizes.txt&quot;</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/size_of_cds_utrs.pl --annot <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt
<span class="k">fi</span>

<span class="nb">echo</span> -e <span class="s2">&quot;metagene annotaion&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/annotate_bed_file.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> --bed2 <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/rel_and_abs_dist_calc.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed --regions <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt

<span class="nb">echo</span> -e <span class="s2">&quot;generate metagene plot from </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_metagene.txt&quot;</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/metagene.r -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_pseudoU_standard.pdf <span class="p">&amp;</span>&gt; /dev/null

<span class="nb">echo</span> <span class="s2">&quot;metagene plot end&quot;</span>
<span class="nb">echo</span> -e <span class="s2">&quot;metagene result in ./</span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">metagene.r</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># reference https://github.com/olarerin/metaPlotR</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;ggplot2&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;optparse&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;RColorBrewer&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;scales&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;dplyr&quot;</span><span class="o">))</span>

<span class="nv">option_list</span> <span class="o">=</span> list<span class="o">(</span>
  make_option<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;-f&quot;</span>, <span class="s2">&quot;--metagenefile&quot;</span><span class="o">)</span>, <span class="nv">type</span><span class="o">=</span><span class="s2">&quot;character&quot;</span>, <span class="nv">default</span><span class="o">=</span>NULL,
              <span class="nv">help</span><span class="o">=</span><span class="s2">&quot;distance of single sites [file]&quot;</span>, <span class="nv">metavar</span><span class="o">=</span><span class="s2">&quot;character&quot;</span><span class="o">)</span>,
  make_option<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;-o&quot;</span>, <span class="s2">&quot;--outfile_prefix&quot;</span><span class="o">)</span>, <span class="nv">type</span><span class="o">=</span><span class="s2">&quot;character&quot;</span>, <span class="nv">default</span><span class="o">=</span>NULL,
              <span class="nv">help</span><span class="o">=</span><span class="s2">&quot;output file name [default= %default]&quot;</span>, <span class="nv">metavar</span><span class="o">=</span><span class="s2">&quot;character&quot;</span><span class="o">)</span>
<span class="o">)</span><span class="p">;</span>
<span class="nv">opt_parser</span> <span class="o">=</span> OptionParser<span class="o">(</span><span class="nv">option_list</span><span class="o">=</span>option_list<span class="o">)</span><span class="p">;</span>
<span class="nv">opt</span> <span class="o">=</span> parse_args<span class="o">(</span>opt_parser<span class="o">)</span><span class="p">;</span>

<span class="k">if</span> <span class="o">(</span>is.null<span class="o">(</span>opt<span class="nv">$metagenefile</span><span class="o">)||</span> is.null<span class="o">(</span>opt<span class="nv">$outfile_prefix</span><span class="o">)</span> <span class="o">){</span>
  print_help<span class="o">(</span>opt_parser<span class="o">)</span><span class="p">;</span>
  stop<span class="o">(</span><span class="s2">&quot;Please provide -f metagenefile and -o outfile_prefix option&quot;</span>, call.<span class="o">=</span>FALSE<span class="o">)</span><span class="p">;</span>
<span class="o">}</span>

<span class="nv">Metagenefile</span> <span class="o">=</span> opt<span class="nv">$metagenefile</span>
<span class="nv">outFile_prefix</span> <span class="o">=</span> opt<span class="nv">$outfile_prefix</span>

print<span class="o">(</span>Metagenefile<span class="o">)</span>
print<span class="o">(</span>outFile_prefix<span class="o">)</span>

pseudoU.dist &lt;- read.delim<span class="o">(</span>Metagenefile, <span class="nv">header</span> <span class="o">=</span> T<span class="o">)</span>
<span class="c1"># Determine longest length transcript for each gene</span>
trx_len &lt;- pseudoU.dist<span class="nv">$utr5_size</span> + pseudoU.dist<span class="nv">$cds_size</span> + pseudoU.dist<span class="nv">$utr3_size</span>
temp &lt;- data.frame<span class="o">(</span>paste<span class="o">(</span>pseudoU.dist<span class="nv">$chr</span>,pseudoU.dist<span class="nv">$coord</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span>, pseudoU.dist<span class="nv">$refseqID</span>, trx_len<span class="o">)</span>
colnames<span class="o">(</span>temp<span class="o">)</span> &lt;- c<span class="o">(</span><span class="s2">&quot;coord&quot;</span>, <span class="s2">&quot;gid&quot;</span>, <span class="s2">&quot;trx_len&quot;</span><span class="o">)</span>
temp.df &lt;- temp<span class="o">[</span>order<span class="o">(</span>temp<span class="nv">$coord</span>,  temp<span class="nv">$gid</span>, -temp<span class="nv">$trx_len</span><span class="o">)</span>,<span class="o">]</span>
temp.df &lt;- temp<span class="o">[</span>!duplicated<span class="o">(</span>temp<span class="nv">$coord</span><span class="o">)</span>,<span class="o">]</span>

<span class="c1"># m6a data to one transcript per gene (longest)</span>
pseudoU.dist &lt;- pseudoU.dist<span class="o">[</span>pseudoU.dist<span class="nv">$refseqID</span> %in% temp.df<span class="nv">$gid</span>,<span class="o">]</span>
pseudoU.dist<span class="nv">$metagene_feature</span>&lt;-case_when<span class="o">(</span>
  <span class="m">0</span> &lt;<span class="o">=</span> pseudoU.dist<span class="nv">$rel_location</span> <span class="p">&amp;</span> pseudoU.dist<span class="nv">$rel_location</span> &lt; <span class="m">1</span> ~ <span class="s2">&quot;5&#39;UTR&quot;</span>,
  <span class="m">1</span>&lt;<span class="o">=</span> pseudoU.dist<span class="nv">$rel_location</span> <span class="p">&amp;</span> pseudoU.dist<span class="nv">$rel_location</span> &lt; <span class="m">2</span> ~ <span class="s2">&quot;CDS&quot;</span>,
  <span class="m">2</span>&lt;<span class="o">=</span> pseudoU.dist<span class="nv">$rel_location</span> <span class="p">&amp;</span> pseudoU.dist<span class="nv">$rel_location</span> &lt;<span class="o">=</span> <span class="m">3</span> ~ <span class="s2">&quot;3&#39;UTR&quot;</span>
<span class="o">)</span>
table<span class="o">(</span>pseudoU.dist<span class="nv">$metagene_feature</span><span class="o">)</span>
write.table<span class="o">(</span>pseudoU.dist,paste<span class="o">(</span>outFile_prefix,<span class="s2">&quot;_pseudoU.dist.uniq.txt&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">)</span>,row.names<span class="o">=</span>F,quote<span class="o">=</span>F<span class="o">)</span>


<span class="c1">####standard#####</span>
metagene_pseudoU&lt;- ggplot<span class="o">(</span>pseudoU.dist,aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>rel_location<span class="o">))</span>+
geom_density<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span><span class="m">0</span>.8, <span class="nv">color</span> <span class="o">=</span> <span class="s2">&quot;black&quot;</span>,size<span class="o">=</span><span class="m">0</span>.6,fill <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="o">)</span>+
geom_vline<span class="o">(</span><span class="nv">xintercept</span> <span class="o">=</span> <span class="m">1</span>:2, <span class="nv">col</span> <span class="o">=</span> brewer.pal<span class="o">(</span><span class="m">3</span>, <span class="s2">&quot;Set1&quot;</span><span class="o">)[</span><span class="m">1</span><span class="o">]</span>,linetype<span class="o">=</span><span class="s2">&quot;dashed&quot;</span><span class="o">)</span>+
theme_classic<span class="o">()</span>+
theme<span class="o">(</span>legend.position<span class="o">=</span><span class="s2">&quot;top&quot;</span>,
  legend.title<span class="o">=</span>element_blank<span class="o">()</span>,
  panel.background <span class="o">=</span> element_blank<span class="o">()</span>,
  axis.title.x<span class="o">=</span>element_blank<span class="o">())</span>+
scale_x_continuous<span class="o">(</span><span class="nv">limits</span> <span class="o">=</span> c<span class="o">(</span><span class="m">0</span>, <span class="m">3</span><span class="o">)</span>,expand<span class="o">=</span>c<span class="o">(</span><span class="m">0</span>,0<span class="o">))</span>+
scale_y_continuous<span class="o">(</span><span class="nv">expand</span><span class="o">=</span>c<span class="o">(</span><span class="m">0</span>,0<span class="o">))</span>+
theme<span class="o">(</span>axis.text.x <span class="o">=</span> element_text<span class="o">(</span><span class="nv">face</span><span class="o">=</span><span class="s2">&quot;bold&quot;</span>,size<span class="o">=</span><span class="m">14</span>,hjust<span class="o">=</span><span class="m">1</span>.8<span class="o">))</span>+
theme<span class="o">(</span>plot.title <span class="o">=</span> element_text<span class="o">(</span><span class="nv">hjust</span> <span class="o">=</span> <span class="m">0</span>.5<span class="o">)</span>,plot.margin<span class="o">=</span>unit<span class="o">(</span>c<span class="o">(</span><span class="m">2</span>,2,2,2<span class="o">)</span>,units<span class="o">=</span><span class="s2">&quot;cm&quot;</span><span class="o">))</span>
<span class="c1"># +scale_x_discrete(limits = c(&#39;5_UTR&#39;,&#39;CDS  &#39;,&#39;3_UTR&#39;))</span>
print<span class="o">(</span>paste<span class="o">(</span>outFile_prefix,<span class="s2">&quot;_metagene_pseudoU_standard.pdf&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">))</span>
pdf<span class="o">(</span>paste<span class="o">(</span>outFile_prefix,<span class="s2">&quot;_metagene_pseudoU_standard.pdf&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">))</span>
metagene_pseudoU
dev.off<span class="o">()</span>


<span class="c1">####normalize by region length#####</span>
utr5.SF &lt;- median<span class="o">(</span>pseudoU.dist<span class="nv">$utr5_size</span>, na.rm <span class="o">=</span> T<span class="o">)</span>/median<span class="o">(</span>pseudoU.dist<span class="nv">$cds_size</span>, na.rm <span class="o">=</span> T<span class="o">)</span>
utr3.SF &lt;- median<span class="o">(</span>pseudoU.dist<span class="nv">$utr3_size</span>, na.rm <span class="o">=</span> T<span class="o">)</span>/median<span class="o">(</span>pseudoU.dist<span class="nv">$cds_size</span>, na.rm <span class="o">=</span> T<span class="o">)</span>

<span class="c1"># assign the regions to new dataframes</span>
utr5.pseudoU.dist &lt;- pseudoU.dist<span class="o">[</span>pseudoU.dist<span class="nv">$rel_location</span> &lt; <span class="m">1</span>, <span class="o">]</span>
cds.pseudoU.dist &lt;- pseudoU.dist <span class="o">[</span>pseudoU.dist<span class="nv">$rel_location</span> &lt; <span class="m">2</span> <span class="p">&amp;</span> pseudoU.dist<span class="nv">$rel_location</span> &gt;<span class="o">=</span> <span class="m">1</span>, <span class="o">]</span>
utr3.pseudoU.dist &lt;- pseudoU.dist<span class="o">[</span>pseudoU.dist<span class="nv">$rel_location</span> &gt;<span class="o">=</span> <span class="m">2</span>, <span class="o">]</span>


<span class="c1"># rescale 5&#39;UTR and 3&#39;UTR</span>
utr5.pseudoU.dist<span class="nv">$rel_location</span> &lt;- rescale<span class="o">(</span>utr5.pseudoU.dist<span class="nv">$rel_location</span>, <span class="nv">to</span> <span class="o">=</span> c<span class="o">(</span><span class="m">1</span>-utr5.SF, <span class="m">1</span><span class="o">)</span>, <span class="nv">from</span> <span class="o">=</span> c<span class="o">(</span><span class="m">0</span>,1<span class="o">))</span>
utr3.pseudoU.dist<span class="nv">$rel_location</span> &lt;- rescale<span class="o">(</span>utr3.pseudoU.dist<span class="nv">$rel_location</span>, <span class="nv">to</span> <span class="o">=</span> c<span class="o">(</span><span class="m">2</span>, <span class="m">2</span>+utr3.SF<span class="o">)</span>, <span class="nv">from</span> <span class="o">=</span> c<span class="o">(</span><span class="m">2</span>,3<span class="o">))</span>
pseudoU.metagene.coord &lt;- data.frame<span class="o">(</span><span class="nv">norm_value</span><span class="o">=</span>c<span class="o">(</span>utr5.pseudoU.dist<span class="nv">$rel_location</span>, cds.pseudoU.dist<span class="nv">$rel_location</span>, utr3.pseudoU.dist<span class="nv">$rel_location</span><span class="o">)</span>,metagene_feature<span class="o">=</span>c<span class="o">(</span>rep<span class="o">(</span><span class="s2">&quot;5&#39;UTR&quot;</span>,length<span class="o">(</span>utr5.pseudoU.dist<span class="nv">$rel_location</span><span class="o">))</span>,rep<span class="o">(</span><span class="s2">&quot;CDS&quot;</span>,length<span class="o">(</span>cds.pseudoU.dist<span class="nv">$rel_location</span><span class="o">))</span>,rep<span class="o">(</span><span class="s2">&quot;3&#39;UTR&quot;</span>,length<span class="o">(</span>utr3.pseudoU.dist<span class="nv">$rel_location</span><span class="o">))))</span>
pseudoU.metagene.coord&lt;-arrange<span class="o">(</span>pseudoU.metagene.coord,norm_value<span class="o">)</span>

metagene_pseudoU&lt;- ggplot<span class="o">(</span>pseudoU.metagene.coord,aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>norm_value<span class="o">))</span>+
geom_density<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span><span class="m">0</span>.8, <span class="nv">color</span> <span class="o">=</span> <span class="s2">&quot;black&quot;</span>,size<span class="o">=</span><span class="m">0</span>.6,fill <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="o">)</span>+
geom_density<span class="o">()</span>+
theme_classic<span class="o">()</span>+theme<span class="o">(</span>legend.position<span class="o">=</span><span class="s2">&quot;top&quot;</span>,legend.title<span class="o">=</span>element_blank<span class="o">()</span>,panel.background<span class="o">=</span>element_rect<span class="o">(</span><span class="nv">fill</span><span class="o">=</span><span class="s2">&quot;white&quot;</span>,color<span class="o">=</span><span class="s2">&quot;black&quot;</span><span class="o">))</span>+
theme<span class="o">(</span>axis.title.x<span class="o">=</span>element_blank<span class="o">())</span> +
<span class="c1"># scale_x_discrete(limits = c(&#39;5_UTR&#39;,&#39;CDS&#39;,&#39;3_UTR&#39;))+</span>
theme<span class="o">(</span>axis.text.x <span class="o">=</span> element_text<span class="o">(</span><span class="nv">face</span><span class="o">=</span><span class="s2">&quot;bold&quot;</span>,size<span class="o">=</span><span class="m">8</span><span class="o">)</span>,axis.text.y <span class="o">=</span> element_text<span class="o">(</span><span class="nv">face</span><span class="o">=</span><span class="s2">&quot;bold&quot;</span>,size<span class="o">=</span><span class="m">8</span><span class="o">))</span>+
scale_x_continuous<span class="o">(</span><span class="nv">expand</span> <span class="o">=</span> c<span class="o">(</span><span class="m">0</span>, <span class="m">0</span><span class="o">))</span> +
scale_y_continuous<span class="o">(</span><span class="nv">expand</span> <span class="o">=</span> c<span class="o">(</span><span class="m">0</span>, <span class="m">0</span><span class="o">))</span> +
theme<span class="o">(</span>plot.margin<span class="o">=</span>unit<span class="o">(</span>c<span class="o">(</span><span class="m">2</span>,2,2,2<span class="o">)</span>,units<span class="o">=</span><span class="s2">&quot;cm&quot;</span><span class="o">)</span>,legend.background <span class="o">=</span> element_rect<span class="o">(</span><span class="nv">colour</span> <span class="o">=</span> <span class="s1">&#39;grey&#39;</span>, <span class="nv">fill</span> <span class="o">=</span> <span class="s1">&#39;white&#39;</span>, <span class="nv">linetype</span><span class="o">=</span><span class="s1">&#39;dashed&#39;</span><span class="o">))</span>+
geom_vline<span class="o">(</span><span class="nv">xintercept</span> <span class="o">=</span> <span class="m">1</span>:2, <span class="nv">col</span> <span class="o">=</span> <span class="s2">&quot;grey&quot;</span>,size<span class="o">=</span><span class="m">0</span>.7,linetype<span class="o">=</span><span class="s2">&quot;dashed&quot;</span><span class="o">)</span>+
geom_area<span class="o">(</span>
    aes<span class="o">(</span><span class="nv">x</span> <span class="o">=</span> stage<span class="o">(</span>norm_value, <span class="nv">after_stat</span> <span class="o">=</span> oob_censor<span class="o">(</span>x, c<span class="o">(</span><span class="m">0</span>, <span class="m">1</span><span class="o">)))</span>,
    <span class="nv">fill</span><span class="o">=</span><span class="s2">&quot;UTR5&quot;</span><span class="o">)</span>,
    <span class="nv">stat</span> <span class="o">=</span> <span class="s2">&quot;density&quot;</span>
  <span class="o">)</span>+
geom_area<span class="o">(</span>
    aes<span class="o">(</span><span class="nv">x</span> <span class="o">=</span> stage<span class="o">(</span>norm_value, <span class="nv">after_stat</span> <span class="o">=</span> oob_censor<span class="o">(</span>x, c<span class="o">(</span><span class="m">1</span>, <span class="m">2</span><span class="o">)))</span>,
    <span class="nv">fill</span><span class="o">=</span><span class="s2">&quot;CDS&quot;</span><span class="o">)</span>,
    <span class="nv">stat</span> <span class="o">=</span> <span class="s2">&quot;density&quot;</span>
    <span class="c1"># fill=brewer.pal(6,&quot;Accent&quot;)[5],</span>
  <span class="o">)</span>+
geom_area<span class="o">(</span>
    aes<span class="o">(</span><span class="nv">x</span> <span class="o">=</span> stage<span class="o">(</span>norm_value, <span class="nv">after_stat</span> <span class="o">=</span> oob_censor<span class="o">(</span>x, c<span class="o">(</span><span class="m">2</span>, max<span class="o">(</span>pseudoU.metagene.coord<span class="nv">$norm_value</span><span class="o">))))</span>,
    <span class="nv">fill</span><span class="o">=</span><span class="s2">&quot;UTR3&quot;</span><span class="o">)</span>,
    <span class="nv">stat</span> <span class="o">=</span> <span class="s2">&quot;density&quot;</span>
  <span class="o">)</span>+
scale_fill_manual<span class="o">(</span><span class="nv">values</span><span class="o">=</span>c<span class="o">(</span><span class="nv">UTR5</span><span class="o">=</span>brewer.pal<span class="o">(</span><span class="m">6</span>,<span class="s2">&quot;Set3&quot;</span><span class="o">)[</span><span class="m">4</span><span class="o">]</span>,CDS<span class="o">=</span>brewer.pal<span class="o">(</span><span class="m">6</span>,<span class="s2">&quot;Set3&quot;</span><span class="o">)[</span><span class="m">5</span><span class="o">]</span>,UTR3<span class="o">=</span>brewer.pal<span class="o">(</span><span class="m">6</span>,<span class="s2">&quot;Set3&quot;</span><span class="o">)[</span><span class="m">6</span><span class="o">]))</span>


print<span class="o">(</span>paste<span class="o">(</span>outFile_prefix,<span class="s2">&quot;_metagene_pseudoU_norm_length.pdf&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">))</span>
pdf<span class="o">(</span>paste<span class="o">(</span>outFile_prefix,<span class="s2">&quot;_metagene_pseudoU_norm_length.pdf&quot;</span>,sep<span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">))</span>
metagene_pseudoU
dev.off<span class="o">()</span>
</pre></div>
</div>
<section id="sites-file">
<h3><a class="toc-backref" href="#id3">Ψ-sites file</a><a class="headerlink" href="#sites-file" title="Permalink to this heading"></a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-i</span> <span class="pre">bedfile</span></code> accept file in bed format and pass it to MetaPlotR pipline.</p>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id4">Output</a><a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>Result with <span class="blue">_pseudoU.dist.uniq.txt</span> suffix is the final MetaGene result.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir

<span class="c1"># see all files, don&#39;t run.</span>
$ tree -L <span class="m">1</span>
.
├── Day0_common_rep1_anno.bed
├── Day0_common_rep1_anno.log
├── Day0_common_rep1.bed
├── Day0_common_rep1.bed6
├── Day0_common_rep1_metagene_pseudoU_norm_length.pdf
├── Day0_common_rep1_metagene_pseudoU_standard.pdf
├── Day0_common_rep1_metagene.txt
├── Day0_common_rep1_pseudoU.dist.uniq.txt
└── Day0_common_rep1.sorted_annot.bed

<span class="m">0</span> directories, <span class="m">9</span> files
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <span class="blue">_metagene_config.txt</span> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="%CE%A8-sites%20Annotation.html" class="btn btn-neutral float-left" title="Ψ-sites Annotation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="%CE%A8-sites%20Codon%20Preference.html" class="btn btn-neutral float-right" title="Ψ-sites Codon Preference" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, chenzhr23.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>