
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ψ-sites snoRNA Target Prediction &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/haiku.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites miRNA Target Prediction" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html" />
    <link rel="prev" title="Ψ-sites Codon Preference" href="%CE%A8-sites%20Codon%20Preference.html" /> 
  </head><body>
      <div class="header" role="banner"><h1 class="heading"><a href="../index.html">
          <span>psiFinder v1.0 documentation</span></a></h1>
        <h2 class="heading"><span>Ψ-sites snoRNA Target Prediction</span></h2>
      </div>
      <div class="topnav" role="navigation" aria-label="top navigation">
      
        <p>
        «&#160;&#160;<a href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a>&#160;&#160;»
        </p>

      </div>
      <div class="content" role="main">
        
        
  <section id="sites-snorna-target-prediction">
<h1>Ψ-sites snoRNA Target Prediction<a class="headerlink" href="#sites-snorna-target-prediction" title="Permalink to this heading">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#predict-sites-snorna-target" id="id2">Predict Ψ-sites snoRNA target</a></p>
<ul>
<li><p><a class="reference internal" href="#snorna-sequence" id="id3">snoRNA sequence</a></p></li>
<li><p><a class="reference internal" href="#modified-file" id="id4">Modified file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id5">Output</a></p>
<ul>
<li><p><a class="reference internal" href="#information" id="id6">Information</a></p></li>
<li><p><a class="reference internal" href="#diagram" id="id7">Diagram</a></p></li>
</ul>
</li>
</ul>
</div>
<p>psiFinder <code class="docutils literal notranslate"><span class="pre">Ψ-sites</span> <span class="pre">snoRNA</span> <span class="pre">Target</span> <span class="pre">Prediction</span></code> was developed with a core C program–<code class="docutils literal notranslate"><span class="pre">ACAscan</span></code>, for Ψ-sites snoRNA target prediction.</p>
<img alt="../_images/snoRNA_target_prediction.png" src="../_images/snoRNA_target_prediction.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading">¶</a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format, corresponding annotation file of the bed file (with <cite>_anno.bed</cite> sufffix), and snoRNA dataset <code class="docutils literal notranslate"><span class="pre">human_hg38_snoRNABase_snoDB_rmRepeat.fa</span></code>, to predict snoRNA-(Ψ)target interaction.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Usage:  ACAscan <span class="o">[</span>options<span class="o">]</span> -f &lt;fasta file&gt; -m &lt;modified file&gt;
<span class="o">[</span>options<span class="o">]</span>
-f/--fa&lt;file&gt;          : sequence file&lt;FASTA format&gt;
-m/--mod&lt;file&gt;         : modified file
-o/--output &lt;file&gt;     : output file
-v/--verbose           : verbose information
-V/--version           : ACAscan version
-h/--help              : <span class="nb">help</span> informations
-s/--struct-score      : minimum score <span class="k">for</span> hairpin-Hbox-hairpin-ACAbox<span class="o">[</span><span class="nv">default</span><span class="o">=</span><span class="m">5</span>.0<span class="o">]</span>
-t/--target-score      : minimum score <span class="k">for</span> snoRNA-target duplex<span class="o">[</span><span class="nv">default</span><span class="o">=</span><span class="m">13</span>.5<span class="o">]</span>
</pre></div>
</div>
</section>
<section id="predict-sites-snorna-target">
<h2><a class="toc-backref" href="#id2">Predict Ψ-sites snoRNA target</a><a class="headerlink" href="#predict-sites-snorna-target" title="Permalink to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code>. psiFinder computes the <strong>Target_score/Modified_score/PairNum/total_UnPair</strong> of interactions between hairpin-Hbox-hairpin-ACAbox and snoRNA-target duplex based on the probabilistic model and conserved structure motifs, which makes it feasible to predict Ψ target for both guide snoRNAs and orphan snoRNAs comprehensively. snoRNA-target pairs that fit the following standard are regard as high-confidence interactions: <code class="docutils literal notranslate"><span class="pre">Modified_score</span> <span class="pre">&gt;=</span> <span class="pre">15</span> <span class="pre">&amp;</span> <span class="pre">PairNum</span> <span class="pre">&gt;=</span> <span class="pre">9</span> <span class="pre">&amp;</span> <span class="pre">total_UnPair</span> <span class="pre">&lt;=</span> <span class="pre">2</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -ne <span class="m">3</span> <span class="o">]]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; bedfile annobedfile acaboxseq&#39;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="k">fi</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">annobedfile</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">acaboxseq</span><span class="o">=</span><span class="nv">$3</span>

cat <span class="nv">$annobedfile</span> <span class="p">|</span> cut -f <span class="m">1</span>-6 &gt; <span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>
cat <span class="nv">$annobedfile</span> <span class="p">|</span>awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,$5,$7}&#39;</span> &gt; <span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>.append
bedtools intersect -a <span class="nv">$bedfile</span> -b <span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span> -s &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed
bedtools intersect -a <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed -b <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed -wb<span class="p">|</span>awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,$43,$39}&#39;</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.known.target
bedtools intersect -a <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed -b <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed -wb -v <span class="p">|</span>awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,&quot;unknown&quot;,$39}&#39;</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.unknown.target
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.known.target <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.unknown.target &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites
sed -i -e <span class="s1">&#39;s/T/U/g&#39;</span> <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites

<span class="nb">wait</span>

<span class="nb">echo</span> <span class="s2">&quot;generating ACAscan result(_fa.out)&quot;</span>
<span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/ACAscan -f <span class="nv">$acaboxseq</span> -m <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_fa.out


<span class="nb">echo</span> <span class="s2">&quot;generating text information for ACAscan result&quot;</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/ACA_scan.r -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_fa.out -a <span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>.append -b <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/human_hg38_snoRNABase_snoDB_rmRepeat_addorphaninfo.csv -c <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>

mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU_Orphan_piechart.pdf <span class="p">&amp;</span>&gt; /dev/null
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU_snoRNA_guided_RNA_piechart.pdf <span class="p">&amp;</span>&gt; /dev/null

<span class="nb">echo</span> -e <span class="s2">&quot;Finished: ACAscan done!\n&quot;</span>
<span class="nb">echo</span> -e <span class="s2">&quot;ACAscan result in </span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</div>
<section id="snorna-sequence">
<h3><a class="toc-backref" href="#id3">snoRNA sequence</a><a class="headerlink" href="#snorna-sequence" title="Permalink to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-f</span></code> option requires snoRNA sequence file in fasta format, default psiFinder provide built-in snoRNA dataset <code class="docutils literal notranslate"><span class="pre">human_hg38_snoRNABase_snoDB_rmRepeat.fa</span></code></p>
</section>
<section id="modified-file">
<h3><a class="toc-backref" href="#id4">Modified file</a><a class="headerlink" href="#modified-file" title="Permalink to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-m</span></code> option accept a three column file (with <code class="docutils literal notranslate"><span class="pre">_pseudoU.sites</span></code> suffix, generated by <code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code>) like below (genome_location snoRNA_info target_sequence)</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>chr1_10298995_10298996_+        U6-31-ACA65     GCACAUAUACUAAAAYUGGAACGAUACAGAG
chr1_10299004_10299005_+        U6-40-HBI-100/ACA12     CUAAAAUUGGAACGAYACAGAGAAGAUUAGC
chr1_11908193_11908194_+        U5-43-ACA57     AAAUCUUUCGCCUUUYACUAAAGAUUUCCGU
chr1_44731124_44731125_-        U5-46-U85/U89   UCUUUCGCCUUUUACYAAAGAUUUCCGUGGA
chr1_92700930_92700931_-        U4-4-SCARNA27   AGCYUUGCGCAGUGGCAGU
chr1_109591554_109591555_-      U6-86-ACA65     GGAUGACACGCAAAUYCGUGAAGCGUUCCAU
chr1_145465775_145465776_-      U1-5-ACA47      AUACYUACCUGGCAGGGGAG
chr1_148522605_148522606_+      U1-6-ACA47/U109 AUACUYACCUGGCAGGGGAGA
chr11_62841749_62841750_-       U2-60-unknown   CUUAUCAGUUUAAUAYCUGAUACGUCCUCUA
chr11_62841751_62841752_-       U2-58-unknown   UUCUUAUCAGUUUAAYAUCUGAUACGUCCUC
</pre></div>
</div>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id5">Output</a><a class="headerlink" href="#output" title="Permalink to this heading">¶</a></h2>
<section id="information">
<h3><a class="toc-backref" href="#id6">Information</a><a class="headerlink" href="#information" title="Permalink to this heading">¶</a></h3>
<p>Result with <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence.csv</span></code> suffix is the final snoRNA target prediction result, which is a filter result of <code class="docutils literal notranslate"><span class="pre">_pseudoU.csv</span></code> (filter by <code class="docutils literal notranslate"><span class="pre">Modified_score&gt;=15</span> <span class="pre">&amp;</span> <span class="pre">PairNum&gt;=9</span> <span class="pre">&amp;</span> <span class="pre">total_UnPair&lt;=2</span></code>).</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir
$ tree -L <span class="m">1</span>
.
├── Day0_common_rep1_anno.bed
├── Day0_common_rep1_anno.bed6
├── Day0_common_rep1_anno.bed6.append
├── Day0_common_rep1_anno_info.bed
├── Day0_common_rep1.bed
├── Day0_common_rep1_fa.out
├── Day0_common_rep1.known.target
├── Day0_common_rep1_pseudoU.csv
├── Day0_common_rep1_pseudoU_high_confidence.csv
├── Day0_common_rep1_pseudoU_high_confidence.pdf
├── Day0_common_rep1_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.pdf
├── Day0_common_rep1_pseudoU_high_confidence.txt
├── Day0_common_rep1_pseudoU_high_confidence.xlsx
├── Day0_common_rep1_pseudoU_Orphan_piechart.pdf
├── Day0_common_rep1_pseudoU.pdf
├── Day0_common_rep1_pseudoU.sites
├── Day0_common_rep1_pseudoU.txt
├── Day0_common_rep1_pseudoU.xlsx
└── Day0_common_rep1.unknown.target

<span class="m">0</span> directory, <span class="m">19</span> files
</pre></div>
</div>
</section>
<section id="diagram">
<h3><a class="toc-backref" href="#id7">Diagram</a><a class="headerlink" href="#diagram" title="Permalink to this heading">¶</a></h3>
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence.pdf</span></code> is a graphical summary of snoRNA-target pair on input Ψ-sites file (<strong>rtsSeeker result+bedAnnotator result</strong>).</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_high_confidence.png" src="../_images/snoRNA_target_prediction_pseudoU_high_confidence.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.pdf</span></code> is a pie chart graphical summary of snoRNA-guided RNA on input Ψ-sites file (<strong>rtsSeeker result+bedAnnotator result</strong>).</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.png" src="../_images/snoRNA_target_prediction_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_Orphan_piechart.pdf</span></code> is a pie chart graphical summary of orphan snoRNAs predicted with Ψ-modified targets on input Ψ-sites file.</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_Orphan_piechart.png" src="../_images/snoRNA_target_prediction_pseudoU_Orphan_piechart.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_ACAscan_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>
</section>


      </div>
      <div class="bottomnav" role="navigation" aria-label="bottom navigation">
      
        <p>
        «&#160;&#160;<a href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 5.1.1.
    </div>
  </body>
</html>