

<!doctype html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Ψ-sites Codon Preference &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/bizstyle.css" />
    
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites snoRNA Target Prediction" href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html" />
    <link rel="prev" title="Ψ-sites Distribution" href="%CE%A8-sites%20Distribution.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html" title="Ψ-sites snoRNA Target Prediction"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Distribution.html" title="Ψ-sites Distribution"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Codon Preference</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="sites-codon-preference">
<h1>Ψ-sites Codon Preference<a class="headerlink" href="#sites-codon-preference" title="Permalink to this heading">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#profile-sites-codon-preference" id="id2">Profile Ψ-sites codon preference</a></p>
<ul>
<li><p><a class="reference internal" href="#sites-file" id="id3">Ψ-sites file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id4">Output</a></p>
<ul>
<li><p><a class="reference internal" href="#information" id="id5">Information</a></p></li>
<li><p><a class="reference internal" href="#diagram" id="id6">Diagram</a></p></li>
</ul>
</li>
</ul>
</div>
<p>psiFinder <code class="docutils literal notranslate"><span class="pre">Ψ-sites</span> <span class="pre">Codon</span> <span class="pre">Preference</span></code> utilize <a class="reference external" href="https://github.com/olarerin/metaPlotR">MetaPlotR</a> to generate codon preference result for Ψ-sites.</p>
<img alt="../_images/CodonPre.png" src="../_images/CodonPre.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading">¶</a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result) to <code class="docutils literal notranslate"><span class="pre">CodonPre</span></code> QT widget in bed format, genome fasta file, genome annotation file, and transcript annotation file (the same file for bedAnnotator <code class="docutils literal notranslate"><span class="pre">--anno</span></code> input).</p>
</section>
<section id="profile-sites-codon-preference">
<h2><a class="toc-backref" href="#id2">Profile Ψ-sites codon preference</a><a class="headerlink" href="#profile-sites-codon-preference" title="Permalink to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">codon_preference.sh</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">codon_preference.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

usage<span class="o">()</span> <span class="o">{</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; -i bedfile -g genome -a gtf [option] &#39;</span>
    <span class="nb">echo</span> <span class="s1">&#39; -A &lt;bedfile&gt; : bed6 file for annotating&#39;</span>
    <span class="nb">exit</span> -1
<span class="o">}</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">genome</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">gtf</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">annofile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>

<span class="k">while</span> <span class="nb">getopts</span> <span class="s1">&#39;i:g:a:A:&#39;</span> OPT<span class="p">;</span> <span class="k">do</span>
    <span class="k">case</span> <span class="nv">$OPT</span> <span class="k">in</span>
        i<span class="o">)</span> <span class="nv">bedfile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        g<span class="o">)</span> <span class="nv">genome</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        a<span class="o">)</span> <span class="nv">gtf</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        A<span class="o">)</span> <span class="nv">annofile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        h<span class="o">)</span> usage<span class="p">;;</span>
        ?<span class="o">)</span> usage<span class="p">;;</span>
    <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">echo</span> <span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
bedAnnotator --anno <span class="si">${</span><span class="nv">annofile</span><span class="si">}</span> --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed <span class="m">2</span>&gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.log <span class="c1"># /public/home/chenlf/pseudoU/script/final_version/hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6</span>

<span class="c1">#get data for metagene plot</span>
<span class="nb">echo</span> -e <span class="s2">&quot;get metagene data&quot;</span>
<span class="k">if</span> <span class="o">[</span> ! -f <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed <span class="o">]</span>
    <span class="k">then</span>
    <span class="k">if</span> <span class="o">[</span> ! -f <span class="s2">&quot;</span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">.genePred&quot;</span>  <span class="o">]</span>
    <span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate genePredfile&quot;</span>
    gtfToGenePred -genePredExt -geneNameAsName2 <span class="nv">$gtf</span> <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred
    <span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;genePredfile exist&quot;</span>
    <span class="k">fi</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate metagene annotation bed&quot;</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate metagene annotation bed&quot;</span>
    <span class="k">if</span> <span class="o">[</span> -d <span class="s2">&quot;</span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span><span class="s2">/genome&quot;</span> <span class="o">]</span>
    <span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate </span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span><span class="s2">/genome directory&quot;</span>
    rsync -avzP rsync://hgdownload.cse.ucsc.edu/goldenPath/hg38/chromosomes/ <span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span>/genome
    gunzip <span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span>/genome/*.fa.gz
    <span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span><span class="s2">/genome directory exist&quot;</span>
    <span class="k">fi</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/make_annot_bed.pl --genomeDir <span class="k">$(</span>dirname <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="k">)</span>/genome/ --genePred <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed
    bedtools sort -i <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed
    rm <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed
    <span class="nb">echo</span> <span class="s2">&quot;generate region sizes data&quot;</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/size_of_cds_utrs.pl --annot <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt
<span class="k">fi</span>

<span class="nb">echo</span> -e <span class="s2">&quot;metagene annotaion&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/annotate_bed_file.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> --bed2 <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/rel_and_abs_dist_calc.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed --regions <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt

<span class="nb">echo</span> -e <span class="s2">&quot;get data for codon information&quot;</span>
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed <span class="p">|</span>awk <span class="s1">&#39;BEGIN{FS=&quot;|&quot;;OFS=&quot;\t&quot;} {if ($3==&quot;cds&quot;) {print $1,$3,$4%3}}&#39;</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.txt
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.txt <span class="p">|</span> awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {if ($6==&quot;+&quot;)  {if ($12==0)  {print $1,$2-2,$3,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6} else {print $1,$2+1-$12,$3+3-$12,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6}} else {if ($12==0)  {print $1,$2,$3+2,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6} else {print $1,$2-3+$12,$3-1+$12,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6}}}&#39;</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.bed6
bedtools getfasta -fi <span class="nv">$genome</span> -bed <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.bed6 -name+ -tab -s  &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa
sed -i -e <span class="s1">&#39;s/|/\t/g&#39;</span> -e <span class="s1">&#39;s/T/U/g&#39;</span> -e <span class="s1">&#39;s/t/u/g&#39;</span> -e <span class="s1">&#39;s/::/\t/&#39;</span> -e <span class="s1">&#39;s/ENSU/ENST/&#39;</span> -e <span class="s1">&#39;s/0\tchr/3\tchr/&#39;</span> <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa <span class="c1"># -e &#39;s/a/A/g&#39; -e &#39;s/t/T/g&#39; -e &#39;s/g/G/g&#39; -e &#39;s/c/C/g&#39;</span>

<span class="nb">echo</span> -e <span class="s2">&quot;generate codon_preference plot from </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_metagene_cds.fa&quot;</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/codon_preference.r -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt -m <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_codon_preference.pdf <span class="p">&amp;</span>&gt; /dev/null

<span class="nb">echo</span> <span class="s2">&quot;codon_preference plot end&quot;</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">codon_preference.r</span></code></p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/usr/bin/env Rscript</span>

<span class="c1"># reference https://github.com/olarerin/metaPlotR</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;optparse&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;dplyr&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;RColorBrewer&quot;</span><span class="p">))</span>
<span class="nf">suppressMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;stringr&quot;</span><span class="p">))</span>

<span class="n">option_list</span> <span class="o">=</span> <span class="nf">list</span><span class="p">(</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-f&quot;</span><span class="p">,</span> <span class="s">&quot;--metagenefile&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;metagenefile [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-m&quot;</span><span class="p">,</span> <span class="s">&quot;--metagenefile_cds&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;metagenefile_cds [file]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">),</span>
  <span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-o&quot;</span><span class="p">,</span> <span class="s">&quot;--outfile_prefix&quot;</span><span class="p">),</span> <span class="n">type</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
              <span class="n">help</span><span class="o">=</span><span class="s">&quot;output file name [default= %default]&quot;</span><span class="p">,</span> <span class="n">metavar</span><span class="o">=</span><span class="s">&quot;character&quot;</span><span class="p">)</span>
<span class="p">);</span>
<span class="n">opt_parser</span> <span class="o">=</span> <span class="nf">OptionParser</span><span class="p">(</span><span class="n">option_list</span><span class="o">=</span><span class="n">option_list</span><span class="p">);</span>
<span class="n">opt</span> <span class="o">=</span> <span class="nf">parse_args</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>

<span class="nf">if </span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">metagenefile</span><span class="p">)</span><span class="o">||</span> <span class="nf">is.null</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">outfile_prefix</span><span class="p">)</span> <span class="p">){</span>
  <span class="nf">print_help</span><span class="p">(</span><span class="n">opt_parser</span><span class="p">);</span>
  <span class="nf">stop</span><span class="p">(</span><span class="s">&quot;Please provide -f metagene file -m metagene_cds file and -o outfile_prefix option&quot;</span><span class="p">,</span> <span class="n">call.</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">);</span>
<span class="p">}</span>

<span class="n">metagenefile</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">metagenefile</span>
<span class="n">metagenefile_cds</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">metagenefile_cds</span>
<span class="n">outFile_prefix</span> <span class="o">=</span> <span class="n">opt</span><span class="o">$</span><span class="n">outfile_prefix</span>

<span class="nf">print</span><span class="p">(</span><span class="n">metagenefile</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">metagenefile_cds</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="n">outFile_prefix</span><span class="p">)</span>

<span class="n">pseudoU.dist</span> <span class="o">&lt;-</span> <span class="nf">read.delim </span><span class="p">(</span><span class="n">metagenefile</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">T</span><span class="p">)</span>
<span class="c1"># Determine longest length transcript for each gene</span>
<span class="n">trx_len</span> <span class="o">&lt;-</span> <span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">utr5_size</span> <span class="o">+</span> <span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">cds_size</span> <span class="o">+</span> <span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">utr3_size</span>
<span class="n">temp</span> <span class="o">&lt;-</span> <span class="nf">data.frame</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">chr</span><span class="p">,</span><span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">coord</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">),</span> <span class="n">pseudoU.dist</span><span class="o">$</span><span class="n">refseqID</span><span class="p">,</span> <span class="n">trx_len</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">temp</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;coord&quot;</span><span class="p">,</span> <span class="s">&quot;gid&quot;</span><span class="p">,</span> <span class="s">&quot;trx_len&quot;</span><span class="p">)</span>
<span class="n">temp.df</span> <span class="o">&lt;-</span> <span class="n">temp</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">temp</span><span class="o">$</span><span class="n">coord</span><span class="p">,</span>  <span class="n">temp</span><span class="o">$</span><span class="n">gid</span><span class="p">,</span> <span class="o">-</span><span class="n">temp</span><span class="o">$</span><span class="n">trx_len</span><span class="p">),]</span>
<span class="n">temp.df</span> <span class="o">&lt;-</span> <span class="n">temp</span><span class="p">[</span><span class="o">!</span><span class="nf">duplicated</span><span class="p">(</span><span class="n">temp</span><span class="o">$</span><span class="n">coord</span><span class="p">),]</span>

<span class="n">mycol</span><span class="o">=</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">7</span><span class="p">,</span> <span class="s">&quot;Set1&quot;</span><span class="p">)</span>
<span class="n">codon_info</span><span class="o">&lt;-</span><span class="nf">read.table</span><span class="p">(</span><span class="n">metagenefile_cds</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">codon_final</span><span class="o">&lt;-</span><span class="n">codon_info</span><span class="p">[</span><span class="nf">paste</span><span class="p">(</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V1</span><span class="p">,</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V2</span><span class="p">,</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V3</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span> <span class="o">%in%</span> <span class="nf">paste</span><span class="p">(</span><span class="n">temp.df</span><span class="o">$</span><span class="n">coord</span><span class="p">,</span><span class="n">temp.df</span><span class="o">$</span><span class="n">gid</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">),</span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">6</span><span class="p">)]</span>
<span class="n">codon_final_tmp</span><span class="o">&lt;-</span><span class="n">codon_info</span><span class="p">[</span><span class="nf">paste</span><span class="p">(</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V1</span><span class="p">,</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V2</span><span class="p">,</span><span class="n">codon_info</span><span class="o">$</span><span class="n">V3</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span> <span class="o">%in%</span> <span class="nf">paste</span><span class="p">(</span><span class="n">temp.df</span><span class="o">$</span><span class="n">coord</span><span class="p">,</span><span class="n">temp.df</span><span class="o">$</span><span class="n">gid</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">),]</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">codon_final_tmp</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile_prefix</span><span class="p">,</span> <span class="s">&#39;_codon_preference_bed.txt&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="n">codon_final</span><span class="o">$</span><span class="n">V6</span><span class="o">&lt;-</span><span class="nf">toupper</span><span class="p">(</span><span class="n">codon_final</span><span class="o">$</span><span class="n">V6</span><span class="p">)</span>
<span class="n">codon_final</span><span class="o">&lt;-</span><span class="n">plyr</span><span class="o">::</span><span class="nf">count</span><span class="p">(</span><span class="n">codon_final</span><span class="p">,</span> <span class="nf">names</span><span class="p">(</span><span class="n">codon_final</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">codon_final</span><span class="p">)</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;position&quot;</span><span class="p">,</span><span class="s">&quot;codon&quot;</span><span class="p">,</span><span class="s">&quot;number&quot;</span><span class="p">)</span>
<span class="n">codon_final</span><span class="o">$</span><span class="n">position</span><span class="o">=</span><span class="nf">as.character</span><span class="p">(</span><span class="n">codon_final</span><span class="o">$</span><span class="n">position</span><span class="p">)</span>
<span class="n">codon_final</span><span class="o">$</span><span class="n">position</span><span class="o">&lt;-</span><span class="nf">str_replace_all</span><span class="p">(</span><span class="n">codon_final</span><span class="o">$</span><span class="n">position</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;1&quot;</span> <span class="o">=</span> <span class="s">&quot;1st&quot;</span><span class="p">,</span> <span class="s">&quot;2&quot;</span> <span class="o">=</span> <span class="s">&quot;2nd&quot;</span><span class="p">,</span><span class="s">&quot;3&quot;</span><span class="o">=</span><span class="s">&quot;3rd&quot;</span><span class="p">))</span>
<span class="n">codon_final</span><span class="o">&lt;-</span><span class="nf">arrange</span><span class="p">(</span><span class="n">codon_final</span><span class="p">,</span><span class="nf">desc</span><span class="p">(</span><span class="n">number</span><span class="p">))</span>
<span class="n">codon_final</span><span class="o">&lt;-</span><span class="n">codon_final</span><span class="p">[</span><span class="n">codon_final</span><span class="o">$</span><span class="n">number</span><span class="o">&gt;</span><span class="m">1</span><span class="p">,]</span>
<span class="n">codon_plot</span><span class="o">&lt;-</span><span class="nf">ggplot</span><span class="p">(</span><span class="n">codon_final</span><span class="p">,</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="nf">factor</span><span class="p">(</span><span class="n">codon</span><span class="p">,</span><span class="n">levels</span><span class="o">=</span><span class="nf">unique</span><span class="p">(</span><span class="n">codon_final</span><span class="o">$</span><span class="n">codon</span><span class="p">)),</span><span class="n">y</span><span class="o">=</span><span class="n">number</span><span class="p">,</span><span class="n">fill</span><span class="o">=</span><span class="n">position</span><span class="p">))</span><span class="o">+</span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">alpha</span><span class="o">=</span><span class="m">0.8</span><span class="p">,</span><span class="n">width</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span><span class="n">stat</span><span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">,</span><span class="n">position</span><span class="o">=</span><span class="s">&quot;stack&quot;</span><span class="p">)</span><span class="o">+</span> <span class="nf">theme_bw</span><span class="p">()</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span>
     <span class="n">legend.position</span><span class="o">=</span><span class="s">&quot;top&quot;</span><span class="p">,</span>
     <span class="n">plot.margin</span> <span class="o">=</span> <span class="nf">unit</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">7</span><span class="p">,</span><span class="m">4</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">4</span><span class="p">),</span> <span class="s">&quot;cm&quot;</span><span class="p">),</span>
         <span class="n">axis.text.x</span> <span class="o">=</span> <span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="o">=</span><span class="m">90</span><span class="p">,</span> <span class="n">hjust</span><span class="o">=</span><span class="m">0</span><span class="p">,</span> <span class="n">vjust</span><span class="o">=</span><span class="m">0.3</span><span class="p">),</span>
         <span class="n">panel.border</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">panel.grid.major</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">panel.grid.minor</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">axis.line</span> <span class="o">=</span> <span class="nf">element_line</span><span class="p">(</span><span class="n">colour</span> <span class="o">=</span> <span class="s">&quot;black&quot;</span><span class="p">),</span>
         <span class="n">axis.title.x</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">axis.ticks.x</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">panel.grid</span> <span class="o">=</span> <span class="nf">element_blank</span><span class="p">(),</span>
         <span class="n">panel.background</span><span class="o">=</span><span class="nf">element_blank</span><span class="p">(),</span>
         <span class="p">)</span><span class="o">+</span><span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;Numbers&quot;</span><span class="p">)</span><span class="o">+</span>
         <span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span> <span class="o">=</span> <span class="n">mycol</span><span class="p">)</span><span class="o">+</span>
         <span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">expand</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">))</span>
<span class="nf">pdf</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile_prefix</span><span class="p">,</span> <span class="s">&#39;_codon_preference.pdf&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">width</span><span class="o">=</span> <span class="m">8</span><span class="p">,</span><span class="n">height</span><span class="o">=</span><span class="m">8</span><span class="p">)</span>
<span class="n">codon_plot</span>
<span class="nf">dev.off</span><span class="p">()</span>

<span class="nf">write.table</span><span class="p">(</span><span class="n">codon_final</span><span class="p">,</span><span class="nf">paste</span><span class="p">(</span><span class="n">outFile_prefix</span><span class="p">,</span> <span class="s">&#39;_codon_preference.txt&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>
</pre></div>
</div>
<section id="sites-file">
<h3><a class="toc-backref" href="#id3">Ψ-sites file</a><a class="headerlink" href="#sites-file" title="Permalink to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-i</span> <span class="pre">bedfile</span></code> accept file in bed format and pass it to MetaPlotR pipline.</p>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id4">Output</a><a class="headerlink" href="#output" title="Permalink to this heading">¶</a></h2>
<section id="information">
<h3><a class="toc-backref" href="#id5">Information</a><a class="headerlink" href="#information" title="Permalink to this heading">¶</a></h3>
<p>In the same directory of the input annotation gtf file, following prerequisite files will be outputed first:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> psiFinder/snakemake/script/metagene

$ tree -L <span class="m">1</span>
.
├── annotate_bed_file.pl
├── gencode.v32.chr_patch_hapl_scaff.annotation_annot_sorted.bed
├── gencode.v32.chr_patch_hapl_scaff.annotation.genePred
├── gencode.v32.chr_patch_hapl_scaff.annotation.gtf
├── gencode.v32.chr_patch_hapl_scaff.annotation_region_sizes.txt
├── genome
├── make_annot_bed.pl
├── metagene.r
├── metagene.sh
├── rel_and_abs_dist_calc.pl
└── size_of_cds_utrs.pl

<span class="m">1</span> directory, <span class="m">10</span> files

$ du -h *
<span class="m">4</span>.0K    annotate_bed_file.pl
21G     gencode.v32.chr_patch_hapl_scaff.annotation_annot_sorted.bed
50M     gencode.v32.chr_patch_hapl_scaff.annotation.genePred
<span class="m">1</span>.4G    gencode.v32.chr_patch_hapl_scaff.annotation.gtf
11M     gencode.v32.chr_patch_hapl_scaff.annotation_region_sizes.txt
<span class="m">3</span>.1G    genome
<span class="m">4</span>.0K    make_annot_bed.pl
<span class="m">8</span>.0K    metagene.r
<span class="m">4</span>.0K    metagene.sh
<span class="m">4</span>.0K    rel_and_abs_dist_calc.pl
<span class="m">4</span>.0K    size_of_cds_utrs.pl
</pre></div>
</div>
<p>Result with <code class="docutils literal notranslate"><span class="pre">_codon_preference.txt</span></code> suffix is the final CodonPre result.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir
$ tree -L <span class="m">1</span>
.
├── Day0_common_rep1_anno.bed
├── Day0_common_rep1_anno.log
├── Day0_common_rep1.bed
├── Day0_common_rep1.bed6
├── Day0_common_rep1_codon_preference_bed.txt
├── Day0_common_rep1_codon_preference.pdf
├── Day0_common_rep1_codon_preference.txt
├── Day0_common_rep1_metagene_cds.bed6
├── Day0_common_rep1_metagene_cds.fa
├── Day0_common_rep1_metagene_cds.txt
├── Day0_common_rep1_metagene.txt
└── Day0_common_rep1.sorted_annot.bed

<span class="m">0</span> directories, <span class="m">12</span> files
</pre></div>
</div>
</section>
<section id="diagram">
<h3><a class="toc-backref" href="#id6">Diagram</a><a class="headerlink" href="#diagram" title="Permalink to this heading">¶</a></h3>
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_codon_preference.pdf</span></code> is a histogram of Ψ-sites codon preference on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/CodonPre_codon_preference.png" src="../_images/CodonPre_codon_preference.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_codon_preference_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="../index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Ψ-sites Codon Preference</a><ul>
<li><a class="reference internal" href="#input">Input</a></li>
<li><a class="reference internal" href="#profile-sites-codon-preference">Profile Ψ-sites codon preference</a><ul>
<li><a class="reference internal" href="#sites-file">Ψ-sites file</a></li>
</ul>
</li>
<li><a class="reference internal" href="#output">Output</a><ul>
<li><a class="reference internal" href="#information">Information</a></li>
<li><a class="reference internal" href="#diagram">Diagram</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="%CE%A8-sites%20Distribution.html"
                          title="previous chapter">Ψ-sites Distribution</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html"
                          title="next chapter">Ψ-sites snoRNA Target Prediction</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/content/Ψ-sites Codon Preference.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html" title="Ψ-sites snoRNA Target Prediction"
             >next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Distribution.html" title="Ψ-sites Distribution"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Codon Preference</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 5.1.1.
    </div>
  </body>
</html>