<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Ψ-sites Annotation &mdash; psiFinder v1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
      <link rel="stylesheet" href="../_static/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Distribution" href="%CE%A8-sites%20Distribution.html" />
    <link rel="prev" title="Ψ-sites Identification" href="%CE%A8-sites%20Identification.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> psiFinder
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Ψ-sites Annotation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#reference-annotation-file">reference annotation file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sites-file">Ψ-sites file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#annotate-sites">Annotate Ψ-sites</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Target%20Prediction.html">Ψ-sites Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">psiFinder</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Ψ-sites Annotation</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Ψ-sites Annotation.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="sites-annotation">
<h1>Ψ-sites Annotation<a class="headerlink" href="#sites-annotation" title="Permalink to this heading"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p>
<ul>
<li><p><a class="reference internal" href="#reference-annotation-file" id="id2">reference annotation file</a></p></li>
<li><p><a class="reference internal" href="#sites-file" id="id3">Ψ-sites file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#annotate-sites" id="id4">Annotate Ψ-sites</a></p></li>
<li><p><a class="reference internal" href="#output" id="id5">Output</a></p></li>
</ul>
</div>
<p>We developed a core C program–<code class="docutils literal notranslate"><span class="pre">bedAnnotator</span></code>, for Ψ-sites annotation.</p>
<img alt="../_images/Annotation.png" src="../_images/Annotation.png" />
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format to generate non-redundant Ψ-sites and get corresponding annotation files.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Usage:  bedAnnotator <span class="o">[</span>options<span class="o">]</span> --anno &lt;bed6 file&gt; --bed/bam &lt;bed/bed6/bam file&gt;
<span class="o">[</span>options<span class="o">]</span>
-v/--verbose                   : verbose information
-V/--version                   : bedAnnotator version
-h/--help                      : <span class="nb">help</span> informations
-s/--strand                    : <span class="o">[</span><span class="m">0</span>,1,2<span class="o">]</span>, <span class="m">0</span> is both strand, <span class="m">1</span> is same strand,
                                  <span class="m">2</span> is different strand <span class="o">[</span>default is both strand<span class="o">]</span>
-S/--summary                   : output the summary informations of each gene types
-C/--collapser                 : reads were collapsed by fastx_collapser, default is <span class="nb">false</span>
--bed&lt;string&gt;                  : input is bed file
--bam&lt;string&gt;                  : input is bam file
--anno&lt;string&gt;                 : annotation file, it must be bed6
-c/--min-cov                   : minimum coverage with overlap<span class="o">[</span><span class="nv">default</span><span class="o">=</span>1e-10<span class="o">]</span>
-o/--output &lt;string&gt;           : output file
</pre></div>
</div>
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading"></a></h2>
<section id="reference-annotation-file">
<h3><a class="toc-backref" href="#id2">reference annotation file</a><a class="headerlink" href="#reference-annotation-file" title="Permalink to this heading"></a></h3>
<p><code class="docutils literal notranslate"><span class="pre">--anno</span></code> option require a annotation file in bed6 format, default psiFinder provides a build-in annotation file named <code class="docutils literal notranslate"><span class="pre">hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6</span></code></p>
</section>
<section id="sites-file">
<h3><a class="toc-backref" href="#id3">Ψ-sites file</a><a class="headerlink" href="#sites-file" title="Permalink to this heading"></a></h3>
<p><code class="docutils literal notranslate"><span class="pre">--bed/bam</span></code> option require a bed/bam file, e.g. rtsSeeker result in bed format.</p>
</section>
</section>
<section id="annotate-sites">
<h2><a class="toc-backref" href="#id4">Annotate Ψ-sites</a><a class="headerlink" href="#annotate-sites" title="Permalink to this heading"></a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">4_bed_annotation.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span> <span class="o">[[</span> <span class="nv">$#</span> -ne <span class="m">2</span> <span class="o">]]</span>
<span class="k">then</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; bedfile bed6file&#39;</span>
    <span class="nb">exit</span> <span class="m">1</span>
<span class="k">fi</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">bed6file</span><span class="o">=</span><span class="nv">$2</span>




<span class="nb">echo</span> <span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
<span class="nv">bedAnnotator_cmd1</span><span class="o">=</span><span class="s2">&quot;./script/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="s2">6 -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
<span class="nv">bedAnnotator_cmd2</span><span class="o">=</span><span class="s2">&quot;./script/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="nv">$bedfile</span><span class="s2"> -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno_append.bed&quot;</span>
<span class="nb">eval</span> <span class="nv">$bedAnnotator_cmd1</span> <span class="p">&amp;</span>
<span class="nb">eval</span> <span class="nv">$bedAnnotator_cmd2</span> <span class="p">&amp;</span>

<span class="nb">wait</span>

<span class="c1">#add gene biotype (get ${bedfile%.bed}_anno.biotype.bed)</span>
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/add_biotype.pl <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed

<span class="c1">#add sequence and remove redundancy</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/bed_annotation.r -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.biotype.bed -g <span class="nv">$bedfile</span> -s <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed -e <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/human.hg38.Pseudo.result.col29.xlsx -i ./snakemake/genome/hg38.fa -j <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/snoRNAbase_rRNA.fa -k <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/snoRNAbase_snRNA.fa -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>

<span class="nb">echo</span> <span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno_stopRatioFC.bed&quot;</span>
sort -r -g -k5 -k7 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.bed
awk <span class="s1">&#39;BEGIN{print &quot;#chrom\tchromStart\tchromEnd\tname\tfoldChange\tstrand\tgeneName\tgeneType\tolBaseNum\tqueryCov\tsampCov\tupDist\tdownDist&quot;}1&#39;</span> <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.bed &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.txt

<span class="nb">echo</span> <span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_uniq_chrloc.txt (Venn diagram input)&quot;</span>
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_add_seq_group_uniq.bed <span class="p">|</span>awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;_&quot;$7&quot;&quot;}&#39;</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_uniq_chrloc.txt

<span class="nb">echo</span> <span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_uniq_chrloc.txt (Venn diagram input)&quot;</span>
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_add_seq_group_uniq.bed <span class="p">|</span>awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {print $52}&#39;</span> &gt; <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_uniq_seq.txt

<span class="nb">echo</span> -e <span class="s2">&quot;Finished: bedAnnotator done\n&quot;</span>

<span class="nb">echo</span> -e <span class="s2">&quot;bedAnnotator result in </span><span class="k">$(</span>dirname <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>

mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_gene_biotype_piechart.pdf <span class="p">&amp;</span>&gt; /dev/null
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id5">Output</a><a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>Result with <span class="blue">_anno_group_redundance.xlsx</span> suffix is the final annoation result, which group candidate Ψ-sites by the same extended sequence and retain one copy of each group.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir

<span class="c1"># see all files, don&#39;t run.</span>
$ tree -L <span class="m">1</span>
.
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group_uniq.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group_uniq.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_append.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno.biotype.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_gene_biotype_num.sort
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_group_redundance.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_stopRatioFC.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_stopRatioFC.txt
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction.bed
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction.bed6
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_gene_biotype_piechart.pdf
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_gene_feature_piechart.pdf
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_common_rRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_common_snRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_novel_rRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_novel_snRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_total_rRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_total_snRNA.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhuman.hg38.Pseudo.result.col29_common.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_predictionhuman.hg38.Pseudo.result.col29_novel.xlsx
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_psi_piechart.pdf
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_rRNA_psi_piechart.pdf
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_snRNA_psi_piechart.pdf
├── A1_A2_svm_filt_totalRNA_svm_psi_prediction_uniq_chrloc.txt
└── A1_A2_svm_filt_totalRNA_svm_psi_prediction_uniq_seq.txt

<span class="m">0</span> directories, <span class="m">28</span> files
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <span class="blue">_bedAnnotator_config.txt</span> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="%CE%A8-sites%20Identification.html" class="btn btn-neutral float-left" title="Ψ-sites Identification" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="%CE%A8-sites%20Distribution.html" class="btn btn-neutral float-right" title="Ψ-sites Distribution" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, chenzhr23.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>