<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Ψ-sites Codon Preference &mdash; psiFinder v1.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom-style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Target Prediction" href="%CE%A8-sites%20Target%20Prediction.html" />
    <link rel="prev" title="Ψ-sites Distribution" href="%CE%A8-sites%20Distribution.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> psiFinder
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Ψ-sites Codon Preference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l2"><a class="reference internal" href="#profile-sites-codon-preference">Profile Ψ-sites codon preference</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sites-file">Ψ-sites file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Target%20Prediction.html">Ψ-sites Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">psiFinder</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Ψ-sites Codon Preference</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/content/Ψ-sites Codon Preference.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="sites-codon-preference">
<h1>Ψ-sites Codon Preference<a class="headerlink" href="#sites-codon-preference" title="Permalink to this heading"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#profile-sites-codon-preference" id="id2">Profile Ψ-sites codon preference</a></p>
<ul>
<li><p><a class="reference internal" href="#sites-file" id="id3">Ψ-sites file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id4">Output</a></p></li>
</ul>
</div>
<p>We utilize <a class="reference external" href="https://github.com/olarerin/metaPlotR">MetaPlotR</a> to generate metagene plot for Ψ-sites distribution profile.</p>
<img alt="../_images/CodonPre.png" src="../_images/CodonPre.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1">Input</a><a class="headerlink" href="#input" title="Permalink to this heading"></a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format, genome fasta file, genome annotation file, and transcript annotation file (the same file for bedAnnotator <code class="docutils literal notranslate"><span class="pre">--anno</span></code> input).</p>
</section>
<section id="profile-sites-codon-preference">
<h2><a class="toc-backref" href="#id2">Profile Ψ-sites codon preference</a><a class="headerlink" href="#profile-sites-codon-preference" title="Permalink to this heading"></a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">codon_preference.sh</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">codon_preference.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

usage<span class="o">()</span> <span class="o">{</span>
    <span class="nb">echo</span> <span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span> <span class="s1">&#39; -i bedfile -g genome -a gtf [option] &#39;</span>
    <span class="nb">echo</span> <span class="s1">&#39; -A &lt;bedfile&gt; : bed6 file for annotating&#39;</span>
    <span class="nb">exit</span> -1
<span class="o">}</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">genome</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">gtf</span><span class="o">=</span><span class="s1">&#39;&#39;</span>
<span class="nv">annofile</span><span class="o">=</span><span class="s1">&#39;&#39;</span>

<span class="k">while</span> <span class="nb">getopts</span> <span class="s1">&#39;i:g:a:A:&#39;</span> OPT<span class="p">;</span> <span class="k">do</span>
    <span class="k">case</span> <span class="nv">$OPT</span> <span class="k">in</span>
        i<span class="o">)</span> <span class="nv">bedfile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        g<span class="o">)</span> <span class="nv">genome</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        a<span class="o">)</span> <span class="nv">gtf</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        A<span class="o">)</span> <span class="nv">annofile</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$OPTARG</span><span class="s2">&quot;</span><span class="p">;;</span>
        h<span class="o">)</span> usage<span class="p">;;</span>
        ?<span class="o">)</span> usage<span class="p">;;</span>
    <span class="k">esac</span>
<span class="k">done</span>

<span class="nb">echo</span> <span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
bedAnnotator --anno <span class="si">${</span><span class="nv">annofile</span><span class="si">}</span> --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed <span class="m">2</span>&gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.log <span class="c1"># /public/home/chenlf/pseudoU/script/final_version/hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6</span>

<span class="c1">#get data for metagene plot</span>
<span class="nb">echo</span> -e <span class="s2">&quot;get metagene data&quot;</span>
<span class="k">if</span> <span class="o">[</span> ! -f <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed <span class="o">]</span>
    <span class="k">then</span>
    <span class="k">if</span> <span class="o">[</span> ! -f <span class="s2">&quot;</span><span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span><span class="s2">.genePred&quot;</span>  <span class="o">]</span>
    <span class="k">then</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate genePredfile&quot;</span>
    gtfToGenePred -genePredExt -geneNameAsName2 <span class="nv">$gtf</span> <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred
    <span class="k">else</span>
    <span class="nb">echo</span> <span class="s2">&quot;genePredfile exist&quot;</span>
    <span class="k">fi</span>
    <span class="nb">echo</span> <span class="s2">&quot;generate metagene annotation bed&quot;</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/make_annot_bed.pl --genomeDir <span class="si">${</span><span class="nv">genome</span><span class="p">%.fa</span><span class="si">}</span>_metagene --genePred <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>.genePred &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed
    bedtools sort -i <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed &gt; <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed
    rm <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot.bed
    <span class="nb">echo</span> <span class="s2">&quot;generate region sizes data&quot;</span>
    perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/size_of_cds_utrs.pl --annot <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt
<span class="k">fi</span>

<span class="nb">echo</span> -e <span class="s2">&quot;metagene annotaion&quot;</span>
cut -f <span class="m">1</span>-6 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/annotate_bed_file.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span> --bed2 <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_annot_sorted.bed &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed
perl <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/rel_and_abs_dist_calc.pl --bed <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed --regions <span class="si">${</span><span class="nv">gtf</span><span class="p">%.gtf</span><span class="si">}</span>_region_sizes.txt &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt

<span class="nb">echo</span> -e <span class="s2">&quot;get data for codon information&quot;</span>
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.sorted_annot.bed <span class="p">|</span>awk <span class="s1">&#39;BEGIN{FS=&quot;|&quot;;OFS=&quot;\t&quot;} {if ($3==&quot;cds&quot;) {print $1,$3,$4%3}}&#39;</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.txt
cat <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.txt <span class="p">|</span> awk <span class="s1">&#39;FS=OFS=&quot;\t&quot; {if ($6==&quot;+&quot;)  {if ($12==0)  {print $1,$2-2,$3,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6} else {print $1,$2+1-$12,$3+3-$12,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6}} else {if ($12==0)  {print $1,$2,$3+2,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6} else {print $1,$2-3+$12,$3-1+$12,&quot;&quot;$1&quot;|&quot;$3&quot;|&quot;$10&quot;|&quot;$12&quot;&quot;,$12,$6}}}&#39;</span> &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.bed6
bedtools getfasta -fi <span class="nv">$genome</span> -bed <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.bed6 -name+ -tab -s  &gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa
sed -i -e <span class="s1">&#39;s/|/\t/g&#39;</span> -e <span class="s1">&#39;s/T/U/g&#39;</span> -e <span class="s1">&#39;s/t/u/g&#39;</span> -e <span class="s1">&#39;s/::/\t/&#39;</span> -e <span class="s1">&#39;s/ENSU/ENST/&#39;</span> -e <span class="s1">&#39;s/0\tchr/3\tchr/&#39;</span> <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa <span class="c1"># -e &#39;s/a/A/g&#39; -e &#39;s/t/T/g&#39; -e &#39;s/g/G/g&#39; -e &#39;s/c/C/g&#39;</span>

<span class="nb">echo</span> -e <span class="s2">&quot;generate codon_preference plot from </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_metagene_cds.fa&quot;</span>
Rscript <span class="k">$(</span>dirname <span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/codon_preference.r -f <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene.txt -m <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_metagene_cds.fa -o <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>
mupdf-x11 <span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_codon_preference.pdf <span class="p">&amp;</span>&gt; /dev/null

<span class="nb">echo</span> <span class="s2">&quot;codon_preference plot end&quot;</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">codon_preference.r</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># reference https://github.com/olarerin/metaPlotR</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;ggplot2&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;optparse&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;dplyr&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;RColorBrewer&quot;</span><span class="o">))</span>
suppressMessages<span class="o">(</span>library<span class="o">(</span><span class="s2">&quot;stringr&quot;</span><span class="o">))</span>

<span class="nv">option_list</span> <span class="o">=</span> list<span class="o">(</span>
  make_option<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;-f&quot;</span>, <span class="s2">&quot;--metagenefile&quot;</span><span class="o">)</span>, <span class="nv">type</span><span class="o">=</span><span class="s2">&quot;character&quot;</span>, <span class="nv">default</span><span class="o">=</span>NULL,
              <span class="nv">help</span><span class="o">=</span><span class="s2">&quot;metagenefile [file]&quot;</span>, <span class="nv">metavar</span><span class="o">=</span><span class="s2">&quot;character&quot;</span><span class="o">)</span>,
  make_option<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;-m&quot;</span>, <span class="s2">&quot;--metagenefile_cds&quot;</span><span class="o">)</span>, <span class="nv">type</span><span class="o">=</span><span class="s2">&quot;character&quot;</span>, <span class="nv">default</span><span class="o">=</span>NULL,
              <span class="nv">help</span><span class="o">=</span><span class="s2">&quot;metagenefile_cds [file]&quot;</span>, <span class="nv">metavar</span><span class="o">=</span><span class="s2">&quot;character&quot;</span><span class="o">)</span>,
  make_option<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;-o&quot;</span>, <span class="s2">&quot;--outfile_prefix&quot;</span><span class="o">)</span>, <span class="nv">type</span><span class="o">=</span><span class="s2">&quot;character&quot;</span>, <span class="nv">default</span><span class="o">=</span>NULL,
              <span class="nv">help</span><span class="o">=</span><span class="s2">&quot;output file name [default= %default]&quot;</span>, <span class="nv">metavar</span><span class="o">=</span><span class="s2">&quot;character&quot;</span><span class="o">)</span>
<span class="o">)</span><span class="p">;</span>
<span class="nv">opt_parser</span> <span class="o">=</span> OptionParser<span class="o">(</span><span class="nv">option_list</span><span class="o">=</span>option_list<span class="o">)</span><span class="p">;</span>
<span class="nv">opt</span> <span class="o">=</span> parse_args<span class="o">(</span>opt_parser<span class="o">)</span><span class="p">;</span>

<span class="k">if</span> <span class="o">(</span>is.null<span class="o">(</span>opt<span class="nv">$metagenefile</span><span class="o">)||</span> is.null<span class="o">(</span>opt<span class="nv">$outfile_prefix</span><span class="o">)</span> <span class="o">){</span>
  print_help<span class="o">(</span>opt_parser<span class="o">)</span><span class="p">;</span>
  stop<span class="o">(</span><span class="s2">&quot;Please provide -f metagene file -m metagene_cds file and -o outfile_prefix option&quot;</span>, call.<span class="o">=</span>FALSE<span class="o">)</span><span class="p">;</span>
<span class="o">}</span>

<span class="nv">metagenefile</span> <span class="o">=</span> opt<span class="nv">$metagenefile</span>
<span class="nv">metagenefile_cds</span> <span class="o">=</span> opt<span class="nv">$metagenefile_cds</span>
<span class="nv">outFile_prefix</span> <span class="o">=</span> opt<span class="nv">$outfile_prefix</span>

print<span class="o">(</span>metagenefile<span class="o">)</span>
print<span class="o">(</span>metagenefile_cds<span class="o">)</span>
print<span class="o">(</span>outFile_prefix<span class="o">)</span>

pseudoU.dist &lt;- read.delim <span class="o">(</span>metagenefile, <span class="nv">header</span> <span class="o">=</span> T<span class="o">)</span>
<span class="c1"># Determine longest length transcript for each gene</span>
trx_len &lt;- pseudoU.dist<span class="nv">$utr5_size</span> + pseudoU.dist<span class="nv">$cds_size</span> + pseudoU.dist<span class="nv">$utr3_size</span>
temp &lt;- data.frame<span class="o">(</span>paste<span class="o">(</span>pseudoU.dist<span class="nv">$chr</span>,pseudoU.dist<span class="nv">$coord</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span>, pseudoU.dist<span class="nv">$refseqID</span>, trx_len<span class="o">)</span>
colnames<span class="o">(</span>temp<span class="o">)</span> &lt;- c<span class="o">(</span><span class="s2">&quot;coord&quot;</span>, <span class="s2">&quot;gid&quot;</span>, <span class="s2">&quot;trx_len&quot;</span><span class="o">)</span>
temp.df &lt;- temp<span class="o">[</span>order<span class="o">(</span>temp<span class="nv">$coord</span>,  temp<span class="nv">$gid</span>, -temp<span class="nv">$trx_len</span><span class="o">)</span>,<span class="o">]</span>
temp.df &lt;- temp<span class="o">[</span>!duplicated<span class="o">(</span>temp<span class="nv">$coord</span><span class="o">)</span>,<span class="o">]</span>

<span class="nv">mycol</span><span class="o">=</span>brewer.pal<span class="o">(</span><span class="m">7</span>, <span class="s2">&quot;Set1&quot;</span><span class="o">)</span>
codon_info&lt;-read.table<span class="o">(</span>metagenefile_cds, <span class="nv">header</span><span class="o">=</span>FALSE<span class="o">)</span>
codon_final&lt;-codon_info<span class="o">[</span>paste<span class="o">(</span>codon_info<span class="nv">$V1</span>,codon_info<span class="nv">$V2</span>,codon_info<span class="nv">$V3</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span> %in% paste<span class="o">(</span>temp.df<span class="nv">$coord</span>,temp.df<span class="nv">$gid</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span>,c<span class="o">(</span><span class="m">4</span>,6<span class="o">)]</span>
codon_final_tmp&lt;-codon_info<span class="o">[</span>paste<span class="o">(</span>codon_info<span class="nv">$V1</span>,codon_info<span class="nv">$V2</span>,codon_info<span class="nv">$V3</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span> %in% paste<span class="o">(</span>temp.df<span class="nv">$coord</span>,temp.df<span class="nv">$gid</span>,sep<span class="o">=</span><span class="s2">&quot;_&quot;</span><span class="o">)</span>,<span class="o">]</span>
write.table<span class="o">(</span>codon_final_tmp,paste<span class="o">(</span>outFile_prefix, <span class="s1">&#39;_codon_preference_bed.txt&#39;</span>, <span class="nv">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">)</span>,sep<span class="o">=</span><span class="s2">&quot;\t&quot;</span>,row.names<span class="o">=</span>FALSE, col.names<span class="o">=</span>TRUE,quote<span class="o">=</span>F<span class="o">)</span>
codon_final<span class="nv">$V6</span>&lt;-toupper<span class="o">(</span>codon_final<span class="nv">$V6</span><span class="o">)</span>
codon_final&lt;-plyr::count<span class="o">(</span>codon_final, names<span class="o">(</span>codon_final<span class="o">))</span>
colnames<span class="o">(</span>codon_final<span class="o">)</span>&lt;-c<span class="o">(</span><span class="s2">&quot;position&quot;</span>,<span class="s2">&quot;codon&quot;</span>,<span class="s2">&quot;number&quot;</span><span class="o">)</span>
codon_final<span class="nv">$position</span><span class="o">=</span>as.character<span class="o">(</span>codon_final<span class="nv">$position</span><span class="o">)</span>
codon_final<span class="nv">$position</span>&lt;-str_replace_all<span class="o">(</span>codon_final<span class="nv">$position</span>, c<span class="o">(</span><span class="s2">&quot;1&quot;</span> <span class="o">=</span> <span class="s2">&quot;1st&quot;</span>, <span class="s2">&quot;2&quot;</span> <span class="o">=</span> <span class="s2">&quot;2nd&quot;</span>,<span class="s2">&quot;3&quot;</span><span class="o">=</span><span class="s2">&quot;3rd&quot;</span><span class="o">))</span>
codon_final&lt;-arrange<span class="o">(</span>codon_final,desc<span class="o">(</span>number<span class="o">))</span>
codon_final&lt;-codon_final<span class="o">[</span>codon_final<span class="nv">$number</span>&gt;1,<span class="o">]</span>
codon_plot&lt;-ggplot<span class="o">(</span>codon_final,aes<span class="o">(</span><span class="nv">x</span><span class="o">=</span>factor<span class="o">(</span>codon,levels<span class="o">=</span>unique<span class="o">(</span>codon_final<span class="nv">$codon</span><span class="o">))</span>,y<span class="o">=</span>number,fill<span class="o">=</span>position<span class="o">))</span>+geom_bar<span class="o">(</span><span class="nv">alpha</span><span class="o">=</span><span class="m">0</span>.8,width<span class="o">=</span><span class="m">0</span>.5,stat<span class="o">=</span><span class="s2">&quot;identity&quot;</span>,position<span class="o">=</span><span class="s2">&quot;stack&quot;</span><span class="o">)</span>+ theme_bw<span class="o">()</span>+theme<span class="o">(</span>
     legend.position<span class="o">=</span><span class="s2">&quot;top&quot;</span>,
     plot.margin <span class="o">=</span> unit<span class="o">(</span>c<span class="o">(</span><span class="m">7</span>,4,7,4<span class="o">)</span>, <span class="s2">&quot;cm&quot;</span><span class="o">)</span>,
         axis.text.x <span class="o">=</span> element_text<span class="o">(</span><span class="nv">angle</span><span class="o">=</span><span class="m">90</span>, <span class="nv">hjust</span><span class="o">=</span><span class="m">0</span>, <span class="nv">vjust</span><span class="o">=</span><span class="m">0</span>.3<span class="o">)</span>,
         panel.border <span class="o">=</span> element_blank<span class="o">()</span>,
         panel.grid.major <span class="o">=</span> element_blank<span class="o">()</span>,
         panel.grid.minor <span class="o">=</span> element_blank<span class="o">()</span>,
         axis.line <span class="o">=</span> element_line<span class="o">(</span><span class="nv">colour</span> <span class="o">=</span> <span class="s2">&quot;black&quot;</span><span class="o">)</span>,
         axis.title.x<span class="o">=</span>element_blank<span class="o">()</span>,
         axis.ticks.x<span class="o">=</span>element_blank<span class="o">()</span>,
         panel.grid <span class="o">=</span> element_blank<span class="o">()</span>,
         panel.background<span class="o">=</span>element_blank<span class="o">()</span>,
         <span class="o">)</span>+ylab<span class="o">(</span><span class="s2">&quot;Numbers&quot;</span><span class="o">)</span>+
         scale_fill_manual<span class="o">(</span><span class="nv">values</span> <span class="o">=</span> mycol<span class="o">)</span>+
         scale_y_continuous<span class="o">(</span><span class="nv">expand</span> <span class="o">=</span> c<span class="o">(</span><span class="m">0</span>,0<span class="o">))</span>
pdf<span class="o">(</span>paste<span class="o">(</span>outFile_prefix, <span class="s1">&#39;_codon_preference.pdf&#39;</span>, <span class="nv">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">)</span>,width<span class="o">=</span> <span class="m">8</span>,height<span class="o">=</span><span class="m">8</span><span class="o">)</span>
codon_plot
dev.off<span class="o">()</span>

write.table<span class="o">(</span>codon_final,paste<span class="o">(</span>outFile_prefix, <span class="s1">&#39;_codon_preference.txt&#39;</span>, <span class="nv">sep</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="o">)</span>,sep<span class="o">=</span><span class="s2">&quot;\t&quot;</span>,row.names<span class="o">=</span>FALSE, col.names<span class="o">=</span>TRUE,quote<span class="o">=</span>F<span class="o">)</span>
</pre></div>
</div>
<section id="sites-file">
<h3><a class="toc-backref" href="#id3">Ψ-sites file</a><a class="headerlink" href="#sites-file" title="Permalink to this heading"></a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-i</span> <span class="pre">bedfile</span></code> accept file in bed format and pass it to MetaPlotR pipline.</p>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id4">Output</a><a class="headerlink" href="#output" title="Permalink to this heading"></a></h2>
<p>Result with <span class="blue">_codon_preference.txt</span> suffix is the final CodonPre result.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> /the/directory/of/out_file_dir

<span class="c1"># see all files, don&#39;t run.</span>
$ tree -L <span class="m">1</span>
.
├── Day0_common_rep1_anno.bed
├── Day0_common_rep1_anno.log
├── Day0_common_rep1.bed
├── Day0_common_rep1.bed6
├── Day0_common_rep1_codon_preference_bed.txt
├── Day0_common_rep1_codon_preference.pdf
├── Day0_common_rep1_codon_preference.txt
├── Day0_common_rep1_metagene_cds.bed6
├── Day0_common_rep1_metagene_cds.fa
├── Day0_common_rep1_metagene_cds.txt
├── Day0_common_rep1_metagene.txt
└── Day0_common_rep1.sorted_annot.bed

<span class="m">0</span> directories, <span class="m">12</span> files
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <span class="blue">_codon_preference_config.txt</span> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="%CE%A8-sites%20Distribution.html" class="btn btn-neutral float-left" title="Ψ-sites Distribution" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="%CE%A8-sites%20Target%20Prediction.html" class="btn btn-neutral float-right" title="Ψ-sites Target Prediction" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, chenzhr23.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>