
<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Ψ-sites Annotation &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d75fae25" />
    <link rel="stylesheet" type="text/css" href="../_static/bizstyle.css?v=5283bb3d" />
    
    <script src="../_static/documentation_options.js?v=e160b93e"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites Distribution" href="%CE%A8-sites%20Distribution.html" />
    <link rel="prev" title="Ψ-sites Identification" href="%CE%A8-sites%20Identification.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20Distribution.html" title="Ψ-sites Distribution"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Identification.html" title="Ψ-sites Identification"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Annotation</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="sites-annotation">
<h1>Ψ-sites Annotation<a class="headerlink" href="#sites-annotation" title="Link to this heading">¶</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p>
<ul>
<li><p><a class="reference internal" href="#reference-annotation-file" id="id2">reference annotation file</a></p></li>
<li><p><a class="reference internal" href="#sites-file" id="id3">Ψ-sites file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#annotate-sites" id="id4">Annotate Ψ-sites</a></p></li>
<li><p><a class="reference internal" href="#output" id="id5">Output</a></p>
<ul>
<li><p><a class="reference internal" href="#information" id="id6">Information</a></p></li>
<li><p><a class="reference internal" href="#diagram" id="id7">Diagram</a></p></li>
</ul>
</li>
</ul>
</nav>
<p>psiFinder <code class="docutils literal notranslate"><span class="pre">Ψ-sites</span> <span class="pre">Annotation</span></code> was developed with a core C program–<code class="docutils literal notranslate"><span class="pre">bedAnnotator</span></code>, for Ψ-sites annotation.</p>
<img alt="../_images/annotation.png" src="../_images/annotation.png" />
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format to generate non-redundant Ψ-sites and get corresponding annotation files.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Usage:<span class="w">  </span>bedAnnotator<span class="w"> </span><span class="o">[</span>options<span class="o">]</span><span class="w"> </span>--anno<span class="w"> </span>&lt;bed6<span class="w"> </span>file&gt;<span class="w"> </span>--bed/bam<span class="w"> </span>&lt;bed/bed6/bam<span class="w"> </span>file&gt;
<span class="o">[</span>options<span class="o">]</span>
-v/--verbose<span class="w">                   </span>:<span class="w"> </span>verbose<span class="w"> </span>information
-V/--version<span class="w">                   </span>:<span class="w"> </span>bedAnnotator<span class="w"> </span>version
-h/--help<span class="w">                      </span>:<span class="w"> </span><span class="nb">help</span><span class="w"> </span>informations
-s/--strand<span class="w">                    </span>:<span class="w"> </span><span class="o">[</span><span class="m">0</span>,1,2<span class="o">]</span>,<span class="w"> </span><span class="m">0</span><span class="w"> </span>is<span class="w"> </span>both<span class="w"> </span>strand,<span class="w"> </span><span class="m">1</span><span class="w"> </span>is<span class="w"> </span>same<span class="w"> </span>strand,
<span class="w">                                  </span><span class="m">2</span><span class="w"> </span>is<span class="w"> </span>different<span class="w"> </span>strand<span class="w"> </span><span class="o">[</span>default<span class="w"> </span>is<span class="w"> </span>both<span class="w"> </span>strand<span class="o">]</span>
-S/--summary<span class="w">                   </span>:<span class="w"> </span>output<span class="w"> </span>the<span class="w"> </span>summary<span class="w"> </span>informations<span class="w"> </span>of<span class="w"> </span>each<span class="w"> </span>gene<span class="w"> </span>types
-C/--collapser<span class="w">                 </span>:<span class="w"> </span>reads<span class="w"> </span>were<span class="w"> </span>collapsed<span class="w"> </span>by<span class="w"> </span>fastx_collapser,<span class="w"> </span>default<span class="w"> </span>is<span class="w"> </span><span class="nb">false</span>
--bed&lt;string&gt;<span class="w">                  </span>:<span class="w"> </span>input<span class="w"> </span>is<span class="w"> </span>bed<span class="w"> </span>file
--bam&lt;string&gt;<span class="w">                  </span>:<span class="w"> </span>input<span class="w"> </span>is<span class="w"> </span>bam<span class="w"> </span>file
--anno&lt;string&gt;<span class="w">                 </span>:<span class="w"> </span>annotation<span class="w"> </span>file,<span class="w"> </span>it<span class="w"> </span>must<span class="w"> </span>be<span class="w"> </span>bed6
-c/--min-cov<span class="w">                   </span>:<span class="w"> </span>minimum<span class="w"> </span>coverage<span class="w"> </span>with<span class="w"> </span>overlap<span class="o">[</span><span class="nv">default</span><span class="o">=</span>1e-10<span class="o">]</span>
-o/--output<span class="w"> </span>&lt;string&gt;<span class="w">           </span>:<span class="w"> </span>output<span class="w"> </span>file
</pre></div>
</div>
<section id="input">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Input</a><a class="headerlink" href="#input" title="Link to this heading">¶</a></h2>
<section id="reference-annotation-file">
<h3><a class="toc-backref" href="#id2" role="doc-backlink">reference annotation file</a><a class="headerlink" href="#reference-annotation-file" title="Link to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">--anno</span></code> option require a annotation file in bed6 format, default psiFinder provides a build-in annotation file named <code class="docutils literal notranslate"><span class="pre">hg38.genecode.v30.tRNA.snoRNA.miRNA.rmsk.exonFeatures.bed6</span></code></p>
</section>
<section id="sites-file">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">Ψ-sites file</a><a class="headerlink" href="#sites-file" title="Link to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">--bed/bam</span></code> option require a bed/bam file, e.g. rtsSeeker result in bed format.</p>
</section>
</section>
<section id="annotate-sites">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">Annotate Ψ-sites</a><a class="headerlink" href="#annotate-sites" title="Link to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">4_bed_annotation.sh</span></code> to: <strong>(1)</strong> generate <code class="docutils literal notranslate"><span class="pre">bedAnnotator</span></code> result for input <code class="docutils literal notranslate"><span class="pre">rtsSeeker</span></code> result; <strong>(2)</strong> add gene biotype information; <strong>(3)</strong> group candidate Ψ-sites by sequence, select the candidate Ψ-site with highest stopRatioFC(foldChange) in each group, unique multiple copies of sequence, and finally ouput unique annotation result.</p>
<p><code class="docutils literal notranslate"><span class="pre">4_bed_annotation.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="nv">$#</span><span class="w"> </span>-ne<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span><span class="w"> </span><span class="s1">&#39; bedfile bed6file&#39;</span>
<span class="w">    </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
<span class="k">fi</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">bed6file</span><span class="o">=</span><span class="nv">$2</span>




<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generating </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-6<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="m">6</span>
<span class="nv">bedAnnotator_cmd1</span><span class="o">=</span><span class="s2">&quot;./script/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="s2">6 -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno.bed&quot;</span>
<span class="nv">bedAnnotator_cmd2</span><span class="o">=</span><span class="s2">&quot;./script/bedAnnotator -s 1 --anno </span><span class="nv">$bed6file</span><span class="s2"> --bed </span><span class="nv">$bedfile</span><span class="s2"> -o </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno_append.bed&quot;</span>
<span class="nb">eval</span><span class="w"> </span><span class="nv">$bedAnnotator_cmd1</span><span class="w"> </span><span class="p">&amp;</span>
<span class="nb">eval</span><span class="w"> </span><span class="nv">$bedAnnotator_cmd2</span><span class="w"> </span><span class="p">&amp;</span>

<span class="nb">wait</span>

<span class="c1">#add gene biotype (get ${bedfile%.bed}_anno.biotype.bed)</span>
perl<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/add_biotype.pl<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed

<span class="c1">#add sequence and remove redundancy</span>
Rscript<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/bed_annotation.r<span class="w"> </span>-f<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.biotype.bed<span class="w"> </span>-g<span class="w"> </span><span class="nv">$bedfile</span><span class="w"> </span>-s<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed<span class="w"> </span>-e<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/human.hg38.Pseudo.result.col29.xlsx<span class="w"> </span>-i<span class="w"> </span>./snakemake/genome/hg38.fa<span class="w"> </span>-j<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/snoRNAbase_rRNA.fa<span class="w"> </span>-k<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/snoRNAbase_snRNA.fa<span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>

<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_anno_stopRatioFC.bed&quot;</span>
sort<span class="w"> </span>-r<span class="w"> </span>-g<span class="w"> </span>-k5<span class="w"> </span>-k7<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno.bed<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.bed
awk<span class="w"> </span><span class="s1">&#39;BEGIN{print &quot;#chrom\tchromStart\tchromEnd\tname\tfoldChange\tstrand\tgeneName\tgeneType\tolBaseNum\tqueryCov\tsampCov\tupDist\tdownDist&quot;}1&#39;</span><span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.bed<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_stopRatioFC.txt

<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_uniq_chrloc.txt (Venn diagram input)&quot;</span>
cat<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_add_seq_group_uniq.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;_&quot;$7&quot;&quot;}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_uniq_chrloc.txt

<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generate </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="s2">_uniq_chrloc.txt (Venn diagram input)&quot;</span>
cat<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_add_seq_group_uniq.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;FS=OFS=&quot;\t&quot; {print $52}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_uniq_seq.txt

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Finished: bedAnnotator done\n&quot;</span>

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;bedAnnotator result in </span><span class="k">$(</span>dirname<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>

mupdf-x11<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_gene_biotype_piechart.pdf<span class="w"> </span><span class="p">&amp;</span>&gt;<span class="w"> </span>/dev/null
</pre></div>
</div>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading">¶</a></h2>
<section id="information">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Information</a><a class="headerlink" href="#information" title="Link to this heading">¶</a></h3>
<p>Result with <code class="docutils literal notranslate"><span class="pre">_anno_group_redundance.xlsx</span></code> suffix is the final annoation result, which group candidate Ψ-sites by the same extended sequence and retain one copy of each group.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>/the/directory/of/out_file_dir
$<span class="w"> </span>tree<span class="w"> </span>-L<span class="w"> </span><span class="m">1</span>
.
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group_uniq.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group_uniq.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_add_seq_group.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_append.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno.biotype.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_gene_biotype_num.sort
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_group_redundance.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_stopRatioFC.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_anno_stopRatioFC.txt
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction.bed
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction.bed6
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_gene_biotype_piechart.pdf
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_gene_feature_piechart.pdf
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_common_rRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_common_snRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_novel_rRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_novel_snRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_total_rRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhg38.psiU.SingleSites_total_snRNA.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhuman.hg38.Pseudo.result.col29_common.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_predictionhuman.hg38.Pseudo.result.col29_novel.xlsx
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_psi_piechart.pdf
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_rRNA_psi_piechart.pdf
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_known_novel_snRNA_psi_piechart.pdf
├──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_uniq_chrloc.txt
└──<span class="w"> </span>A1_A2_svm_filt_totalRNA_svm_psi_prediction_uniq_seq.txt

<span class="m">0</span><span class="w"> </span>directories,<span class="w"> </span><span class="m">28</span><span class="w"> </span>files
</pre></div>
</div>
</section>
<section id="diagram">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Diagram</a><a class="headerlink" href="#diagram" title="Link to this heading">¶</a></h3>
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_gene_biotype_piechart.pdf</span></code> is a pie chart graphical summary of gene biological types <a class="reference external" href="https://www.ensembl.org/info/genome/genebuild/biotypes.html">(Biotypes)</a> on input Ψ-sites file (rtsSeeker result). <code class="docutils literal notranslate"><span class="pre">4_bed_annotation.sh</span></code> firstly run <code class="docutils literal notranslate"><span class="pre">add_biotype.pl</span></code> to add gene Biotype information to bedAnnotator result (with <code class="docutils literal notranslate"><span class="pre">_anno.bed</span></code> suffix).</p>
<div class="highlight-perl notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/perl</span>

<span class="k">use</span><span class="w"> </span><span class="nn">strict</span><span class="p">;</span>
<span class="k">use</span><span class="w"> </span><span class="nn">warnings</span><span class="p">;</span>

<span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="w"> </span><span class="nb">scalar</span><span class="p">(</span><span class="nv">@ARGV</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="mi">1</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nb">die</span><span class="w"> </span><span class="s">&quot;Usage: perl $0 &lt;annoFile&gt;\n&quot;</span><span class="p">;</span>
<span class="w">    </span><span class="nb">die</span><span class="w"> </span><span class="s">&quot;Example: perl $0 bedAnnotator_res.bed &gt;out 2&gt;log\n&quot;</span><span class="p">;</span>
<span class="p">}</span>

<span class="k">my</span><span class="w"> </span><span class="nv">$annoFile</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="nv">$ARGV</span><span class="p">[</span><span class="mi">0</span><span class="p">];</span>

<span class="k">my</span><span class="w"> </span><span class="nv">@protein_coding</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;protein_coding&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@pseudogene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;rRNA_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;transcribed_unprocessed_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;translated_processed_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;unprocessed_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;processed_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;transcribed_processed_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;unitary_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;polymorphic_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;transcribed_unitary_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TR_V_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TR_J_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IG_V_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IG_C_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IG_D_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IG_pseudogene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;IG_J_pseudogene&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@lncRNA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;processed_transcript&quot;</span><span class="p">,</span><span class="s">&quot;lincRNA&quot;</span><span class="p">,</span><span class="s">&quot;non_coding&quot;</span><span class="p">,</span><span class="s">&quot;3prime_overlapping_ncRNA&quot;</span><span class="p">,</span><span class="s">&quot;3prime_overlapping_ncrna&quot;</span><span class="p">,</span><span class="s">&quot;sense_intronic&quot;</span><span class="p">,</span><span class="s">&quot;antisense&quot;</span><span class="p">,</span><span class="s">&quot;sense_overlapping&quot;</span><span class="p">,</span><span class="s">&quot;known_ncrna&quot;</span><span class="p">,</span><span class="s">&quot;macro_lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;bidirectional_promoter_lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;retained_intron&quot;</span><span class="p">,</span><span class="s">&quot;TEC&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@sncRNA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;snRNA&quot;</span><span class="p">,</span><span class="s">&quot;snoRNA&quot;</span><span class="p">,</span><span class="s">&quot;misc_RNA&quot;</span><span class="p">,</span><span class="s">&quot;miscRNA&quot;</span><span class="p">,</span><span class="s">&quot;miRNA&quot;</span><span class="p">,</span><span class="s">&quot;ribozyme&quot;</span><span class="p">,</span><span class="s">&quot;sRNA&quot;</span><span class="p">,</span><span class="s">&quot;scRNA&quot;</span><span class="p">,</span><span class="s">&quot;scaRNA&quot;</span><span class="p">,</span><span class="s">&quot;srpRNA&quot;</span><span class="p">,</span><span class="s">&quot;tRNA-Deu&quot;</span><span class="p">,</span><span class="s">&quot;tRNA-RTE&quot;</span><span class="p">,</span><span class="s">&quot;piRNA&quot;</span><span class="p">,</span><span class="s">&quot;siRNA&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@rRNA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;rRNA&quot;</span><span class="p">,</span><span class="s">&quot;Mt_rRNA&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@tRNA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;tRNA&quot;</span><span class="p">,</span><span class="s">&quot;Mt_tRNA&quot;</span><span class="p">,</span><span class="s">&quot;vaultRNA&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@IG_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;IG_C_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_D_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_J_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_LV_gene&quot;</span><span class="p">,</span><span class="s">&quot;IG_V_gene&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@TR_gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;TR_C_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_J_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_V_gene&quot;</span><span class="p">,</span><span class="s">&quot;TR_D_gene&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@repeatMasker</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;5S-Deu-L2&quot;</span><span class="p">,</span><span class="s">&quot;Alu&quot;</span><span class="p">,</span><span class="s">&quot;centr&quot;</span><span class="p">,</span><span class="s">&quot;CR1&quot;</span><span class="p">,</span><span class="s">&quot;DNA&quot;</span><span class="p">,</span><span class="s">&quot;DNA?&quot;</span><span class="p">,</span><span class="s">&quot;ERV1&quot;</span><span class="p">,</span><span class="s">&quot;ERV1?&quot;</span><span class="p">,</span><span class="s">&quot;ERVK&quot;</span><span class="p">,</span><span class="s">&quot;ERVL&quot;</span><span class="p">,</span><span class="s">&quot;ERVL?&quot;</span><span class="p">,</span><span class="s">&quot;ERVL-MaLR&quot;</span><span class="p">,</span><span class="s">&quot;Gypsy&quot;</span><span class="p">,</span><span class="s">&quot;Gypsy?&quot;</span><span class="p">,</span><span class="s">&quot;hAT&quot;</span><span class="p">,</span><span class="s">&quot;hAT?&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Ac&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Blackjack&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Charlie&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tag1&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tip100&quot;</span><span class="p">,</span><span class="s">&quot;hAT-Tip100?&quot;</span><span class="p">,</span><span class="s">&quot;Helitron&quot;</span><span class="p">,</span><span class="s">&quot;Helitron?&quot;</span><span class="p">,</span><span class="s">&quot;L1&quot;</span><span class="p">,</span><span class="s">&quot;L2&quot;</span><span class="p">,</span><span class="s">&quot;Low_complexity&quot;</span><span class="p">,</span><span class="s">&quot;LTR&quot;</span><span class="p">,</span><span class="s">&quot;LTR?&quot;</span><span class="p">,</span><span class="s">&quot;MIR&quot;</span><span class="p">,</span><span class="s">&quot;MULE-MuDR&quot;</span><span class="p">,</span><span class="s">&quot;nonsense_mediated_decay&quot;</span><span class="p">,</span><span class="s">&quot;non_stop_decay&quot;</span><span class="p">,</span><span class="s">&quot;Penelope&quot;</span><span class="p">,</span><span class="s">&quot;PiggyBac&quot;</span><span class="p">,</span><span class="s">&quot;PiggyBac?&quot;</span><span class="p">,</span><span class="s">&quot;RNA&quot;</span><span class="p">,</span><span class="s">&quot;RTE-BovB&quot;</span><span class="p">,</span><span class="s">&quot;RTE-X&quot;</span><span class="p">,</span><span class="s">&quot;Satellite&quot;</span><span class="p">,</span><span class="s">&quot;Simple_repeat&quot;</span><span class="p">,</span><span class="s">&quot;SVA&quot;</span><span class="p">,</span><span class="s">&quot;TcMar?&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Mariner&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tc2&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tigger&quot;</span><span class="p">,</span><span class="s">&quot;telo&quot;</span><span class="p">,</span><span class="s">&quot;Unknown&quot;</span><span class="p">,</span><span class="s">&quot;acro&quot;</span><span class="p">,</span><span class="s">&quot;Crypton&quot;</span><span class="p">,</span><span class="s">&quot;Dong-R4&quot;</span><span class="p">,</span><span class="s">&quot;I-Jockey&quot;</span><span class="p">,</span><span class="s">&quot;Kolobok&quot;</span><span class="p">,</span><span class="s">&quot;L1-Tx1&quot;</span><span class="p">,</span><span class="s">&quot;Merlin&quot;</span><span class="p">,</span><span class="s">&quot;MULE-MuDR?&quot;</span><span class="p">,</span><span class="s">&quot;PIF-Harbinger&quot;</span><span class="p">,</span><span class="s">&quot;SINE?&quot;</span><span class="p">,</span><span class="s">&quot;TcMar&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Pogo&quot;</span><span class="p">,</span><span class="s">&quot;TcMar-Tc1&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@intergenic</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;intergenic&quot;</span><span class="p">);</span>
<span class="k">my</span><span class="w"> </span><span class="nv">@circRNA</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;circRNA&quot;</span><span class="p">);</span>

<span class="k">my</span><span class="w"> </span><span class="nv">$outFile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nv">$annoFile</span><span class="p">;</span>
<span class="nv">$outFile</span><span class="w"> </span><span class="o">=~</span><span class="w"> </span><span class="sr">s/\.bed/\.biotype\.bed/</span><span class="p">;</span>
<span class="nb">open</span><span class="p">(</span><span class="n">IN</span><span class="p">,</span><span class="s">&quot;&lt;$annoFile&quot;</span><span class="p">)</span><span class="w"> </span><span class="ow">or</span><span class="w"> </span><span class="nb">die</span><span class="w"> </span><span class="s">&quot;can&#39;t open $annoFile:$!\n&quot;</span><span class="p">;</span>
<span class="nb">open</span><span class="p">(</span><span class="n">OUT</span><span class="p">,</span><span class="s">&quot;&gt;$outFile&quot;</span><span class="p">)</span><span class="w"> </span><span class="ow">or</span><span class="w"> </span><span class="nb">die</span><span class="w"> </span><span class="s">&quot;can&#39;t open $outFile:$!\n&quot;</span><span class="p">;</span>
<span class="k">while</span><span class="p">(</span><span class="sr">&lt;IN&gt;</span><span class="p">){</span>
<span class="w">    </span><span class="nb">chomp</span><span class="p">;</span>
<span class="w">    </span><span class="k">if</span><span class="p">(</span><span class="nv">$_</span><span class="w"> </span><span class="o">!~</span><span class="w"> </span><span class="sr">/^#/</span><span class="p">){</span>
<span class="w">        </span><span class="k">my</span><span class="w"> </span><span class="nv">@lines</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">split</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">$_</span><span class="p">);</span>
<span class="w">        </span><span class="k">my</span><span class="w"> </span><span class="nv">$info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nv">$lines</span><span class="p">[</span><span class="mi">6</span><span class="p">];</span>
<span class="w">        </span><span class="k">if</span><span class="p">(</span><span class="nv">$info</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="s">&quot;intergenic&quot;</span><span class="p">){</span>
<span class="w">            </span><span class="k">print</span><span class="w">  </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">6</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;intergenic&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;intergenic&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">7</span><span class="o">..</span><span class="mi">11</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">        </span><span class="p">}</span><span class="k">else</span><span class="p">{</span>
<span class="w">            </span><span class="k">my</span><span class="w"> </span><span class="nv">@infoArr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">split</span><span class="p">(</span><span class="sr">/\|/</span><span class="p">,</span><span class="w"> </span><span class="nv">$info</span><span class="p">);</span>
<span class="w">            </span><span class="k">my</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nv">$infoArr</span><span class="p">[</span><span class="mi">4</span><span class="p">];</span>
<span class="w">            </span><span class="k">if</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@circRNA</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;circRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@intergenic</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;intergenic&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@protein_coding</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;mRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@IG_gene</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;mRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@TR_gene</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;mRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@lncRNA</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;lncRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@sncRNA</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;sncRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@rRNA</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;rRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">                </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@tRNA</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;tRNA&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@repeatMasker</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;repeatMasker&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span><span class="k">elsif</span><span class="p">(</span><span class="nb">grep</span><span class="w"> </span><span class="p">{</span><span class="w"> </span><span class="nv">$_</span><span class="w"> </span><span class="ow">eq</span><span class="w"> </span><span class="nv">$geneType</span><span class="w"> </span><span class="p">}</span><span class="w"> </span><span class="nv">@pseudogene</span><span class="p">){</span>
<span class="w">                </span><span class="k">print</span><span class="w"> </span><span class="n">OUT</span><span class="w"> </span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">0</span><span class="o">..</span><span class="mi">7</span><span class="p">]),</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="s">&quot;pseudogene&quot;</span><span class="p">,</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="nb">join</span><span class="p">(</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="nv">@lines</span><span class="p">[</span><span class="mi">8</span><span class="o">..</span><span class="mi">12</span><span class="p">]),</span><span class="s">&quot;\n&quot;</span><span class="p">;</span>
<span class="w">            </span><span class="p">}</span>
<span class="w">        </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="p">}</span>
<span class="nb">close</span><span class="p">(</span><span class="n">IN</span><span class="p">);</span>
<span class="nb">close</span><span class="p">(</span><span class="n">OUT</span><span class="p">);</span>
</pre></div>
</div>
<img alt="../_images/annotation_gene_biotype_piechart.png" src="../_images/annotation_gene_biotype_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_gene_feature_piechart.pdf</span></code> is a pie chart graphical summary of gene feature type on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/annotation_gene_feature_piechart.png" src="../_images/annotation_gene_feature_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_known_novel_rRNA_psi_piechart.pdf</span></code> is a pie chart graphical summary of known/novel rRNA Ψ-sites on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/annotation_known_novel_rRNA_psi_piechart.png" src="../_images/annotation_known_novel_rRNA_psi_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_known_novel_snRNA_psi_piechart.pdf</span></code> is a pie chart graphical summary of known/novel snRNA Ψ-sites on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/annotation_known_novel_snRNA_psi_piechart.png" src="../_images/annotation_known_novel_snRNA_psi_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_known_novel_psi_piechart.pdf</span></code> is a pie chart graphical summary of all known/novel Ψ-sites (compare with reference annotation from RMBase database) on input Ψ-sites file (rtsSeeker result).</p>
<img alt="../_images/annotation_known_novel_psi_piechart.png" src="../_images/annotation_known_novel_psi_piechart.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_bedAnnotator_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Ψ-sites Annotation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#reference-annotation-file">reference annotation file</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sites-file">Ψ-sites file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#annotate-sites">Annotate Ψ-sites</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#information">Information</a></li>
<li class="toctree-l3"><a class="reference internal" href="#diagram">Diagram</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20snoRNA%20Target%20Prediction.html">Ψ-sites snoRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20and%20snoRNA%20Interaction.html">Ψ-sites and snoRNA Interaction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Motif%20Searching.html">Ψ-sites Motif Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html">Ψ-sites SNPs and SNVs Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Differential%20Modification.html">Ψ-sites Differential Modification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="Accessory%20Tools.html">Accessory Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20Distribution.html" title="Ψ-sites Distribution"
             >next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Identification.html" title="Ψ-sites Identification"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites Annotation</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
    &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 8.2.3.
    </div>
  </body>
</html>