
<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Ψ-sites snoRNA Target Prediction &#8212; psiFinder v1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d75fae25" />
    <link rel="stylesheet" type="text/css" href="../_static/bizstyle.css?v=5283bb3d" />
    
    <script src="../_static/documentation_options.js?v=e160b93e"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Ψ-sites miRNA Target Prediction" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html" />
    <link rel="prev" title="Ψ-sites Codon Preference" href="%CE%A8-sites%20Codon%20Preference.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0" />
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20miRNA%20Target%20Prediction.html" title="Ψ-sites miRNA Target Prediction"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Codon%20Preference.html" title="Ψ-sites Codon Preference"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites snoRNA Target Prediction</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="sites-snorna-target-prediction">
<h1>Ψ-sites snoRNA Target Prediction<a class="headerlink" href="#sites-snorna-target-prediction" title="Link to this heading">¶</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#input" id="id1">Input</a></p></li>
<li><p><a class="reference internal" href="#predict-sites-snorna-target" id="id2">Predict Ψ-sites snoRNA target</a></p>
<ul>
<li><p><a class="reference internal" href="#snorna-sequence" id="id3">snoRNA sequence</a></p></li>
<li><p><a class="reference internal" href="#modified-file" id="id4">Modified file</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output" id="id5">Output</a></p>
<ul>
<li><p><a class="reference internal" href="#information" id="id6">Information</a></p></li>
<li><p><a class="reference internal" href="#diagram" id="id7">Diagram</a></p></li>
</ul>
</li>
</ul>
</nav>
<p>psiFinder <code class="docutils literal notranslate"><span class="pre">Ψ-sites</span> <span class="pre">snoRNA</span> <span class="pre">Target</span> <span class="pre">Prediction</span></code> was developed with a core C program–<code class="docutils literal notranslate"><span class="pre">ACAscan</span></code>, for Ψ-sites snoRNA target prediction.</p>
<img alt="../_images/snoRNA_target_prediction.png" src="../_images/snoRNA_target_prediction.png" />
<section id="input">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">Input</a><a class="headerlink" href="#input" title="Link to this heading">¶</a></h2>
<p>Users should choose to upload files (i.e. rtsSeeker result) in bed format, corresponding annotation file of the bed file (with <cite>_anno.bed</cite> sufffix), and snoRNA dataset <code class="docutils literal notranslate"><span class="pre">human_hg38_snoRNABase_snoDB_rmRepeat.fa</span></code>, to predict snoRNA-(Ψ)target interaction.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Usage:<span class="w">  </span>ACAscan<span class="w"> </span><span class="o">[</span>options<span class="o">]</span><span class="w"> </span>-f<span class="w"> </span>&lt;fasta<span class="w"> </span>file&gt;<span class="w"> </span>-m<span class="w"> </span>&lt;modified<span class="w"> </span>file&gt;
<span class="o">[</span>options<span class="o">]</span>
-f/--fa&lt;file&gt;<span class="w">          </span>:<span class="w"> </span>sequence<span class="w"> </span>file&lt;FASTA<span class="w"> </span>format&gt;
-m/--mod&lt;file&gt;<span class="w">         </span>:<span class="w"> </span>modified<span class="w"> </span>file
-o/--output<span class="w"> </span>&lt;file&gt;<span class="w">     </span>:<span class="w"> </span>output<span class="w"> </span>file
-v/--verbose<span class="w">           </span>:<span class="w"> </span>verbose<span class="w"> </span>information
-V/--version<span class="w">           </span>:<span class="w"> </span>ACAscan<span class="w"> </span>version
-h/--help<span class="w">              </span>:<span class="w"> </span><span class="nb">help</span><span class="w"> </span>informations
-s/--struct-score<span class="w">      </span>:<span class="w"> </span>minimum<span class="w"> </span>score<span class="w"> </span><span class="k">for</span><span class="w"> </span>hairpin-Hbox-hairpin-ACAbox<span class="o">[</span><span class="nv">default</span><span class="o">=</span><span class="m">5</span>.0<span class="o">]</span>
-t/--target-score<span class="w">      </span>:<span class="w"> </span>minimum<span class="w"> </span>score<span class="w"> </span><span class="k">for</span><span class="w"> </span>snoRNA-target<span class="w"> </span>duplex<span class="o">[</span><span class="nv">default</span><span class="o">=</span><span class="m">13</span>.5<span class="o">]</span>
</pre></div>
</div>
</section>
<section id="predict-sites-snorna-target">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Predict Ψ-sites snoRNA target</a><a class="headerlink" href="#predict-sites-snorna-target" title="Link to this heading">¶</a></h2>
<p>Once click <code class="docutils literal notranslate"><span class="pre">START</span></code>, psiFindeer will run <code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code>. psiFinder computes the <strong>Target_score/Modified_score/PairNum/total_UnPair</strong> of interactions between hairpin-Hbox-hairpin-ACAbox and snoRNA-target duplex based on the probabilistic model and conserved structure motifs, which makes it feasible to predict Ψ target for both guide snoRNAs and orphan snoRNAs comprehensively. snoRNA-target pairs that fit the following standard are regard as high-confidence interactions: <code class="docutils literal notranslate"><span class="pre">Modified_score</span> <span class="pre">&gt;=</span> <span class="pre">15</span> <span class="pre">&amp;</span> <span class="pre">PairNum</span> <span class="pre">&gt;=</span> <span class="pre">9</span> <span class="pre">&amp;</span> <span class="pre">total_UnPair</span> <span class="pre">&lt;=</span> <span class="pre">2</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash</span>

<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span><span class="nv">$#</span><span class="w"> </span>-ne<span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="s1">&#39;Usage: ./&#39;</span><span class="nv">$0</span><span class="w"> </span><span class="s1">&#39; bedfile annobedfile acaboxseq&#39;</span>
<span class="w">    </span><span class="nb">exit</span><span class="w"> </span><span class="m">1</span>
<span class="k">fi</span>

<span class="nv">bedfile</span><span class="o">=</span><span class="nv">$1</span>
<span class="nv">annobedfile</span><span class="o">=</span><span class="nv">$2</span>
<span class="nv">acaboxseq</span><span class="o">=</span><span class="nv">$3</span>

cat<span class="w"> </span><span class="nv">$annobedfile</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>-6<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>
cat<span class="w"> </span><span class="nv">$annobedfile</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,$5,$7}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>.append
bedtools<span class="w"> </span>intersect<span class="w"> </span>-a<span class="w"> </span><span class="nv">$bedfile</span><span class="w"> </span>-b<span class="w"> </span><span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span><span class="w"> </span>-s<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed
bedtools<span class="w"> </span>intersect<span class="w"> </span>-a<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed<span class="w"> </span>-b<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed<span class="w"> </span>-wb<span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,$43,$39}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.known.target
bedtools<span class="w"> </span>intersect<span class="w"> </span>-a<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_anno_info.bed<span class="w"> </span>-b<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed<span class="w"> </span>-wb<span class="w"> </span>-v<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;FS=OFS=&quot;\t&quot; {print &quot;&quot;$1&quot;_&quot;$2&quot;_&quot;$3&quot;_&quot;$6&quot;&quot;,&quot;unknown&quot;,$39}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.unknown.target
cat<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.known.target<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>.unknown.target<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites
sed<span class="w"> </span>-i<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/T/U/g&#39;</span><span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites

<span class="nb">wait</span>

<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generating ACAscan result(_fa.out)&quot;</span>
<span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/ACAscan<span class="w"> </span>-f<span class="w"> </span><span class="nv">$acaboxseq</span><span class="w"> </span>-m<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU.sites<span class="w"> </span>&gt;<span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_fa.out


<span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;generating text information for ACAscan result&quot;</span>
Rscript<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/ACA_scan.r<span class="w"> </span>-f<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_fa.out<span class="w"> </span>-a<span class="w"> </span><span class="si">${</span><span class="nv">annobedfile</span><span class="si">}</span><span class="m">6</span>.append<span class="w"> </span>-b<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/human_hg38_snoRNABase_snoDB_rmRepeat_addorphaninfo.csv<span class="w"> </span>-c<span class="w"> </span><span class="k">$(</span>dirname<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$0</span><span class="s2">&quot;</span><span class="k">)</span>/hg38.psiU.SingleSites.bed<span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>

mupdf-x11<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU_Orphan_piechart.pdf<span class="w"> </span><span class="p">&amp;</span>&gt;<span class="w"> </span>/dev/null
mupdf-x11<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="p">%.bed</span><span class="si">}</span>_pseudoU_snoRNA_guided_RNA_piechart.pdf<span class="w"> </span><span class="p">&amp;</span>&gt;<span class="w"> </span>/dev/null

<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;Finished: ACAscan done!\n&quot;</span>
<span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;ACAscan result in </span><span class="k">$(</span>dirname<span class="w"> </span><span class="si">${</span><span class="nv">bedfile</span><span class="si">}</span><span class="k">)</span><span class="s2">&quot;</span>
</pre></div>
</div>
<section id="snorna-sequence">
<h3><a class="toc-backref" href="#id3" role="doc-backlink">snoRNA sequence</a><a class="headerlink" href="#snorna-sequence" title="Link to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-f</span></code> option requires snoRNA sequence file in fasta format, default psiFinder provide built-in snoRNA dataset <code class="docutils literal notranslate"><span class="pre">human_hg38_snoRNABase_snoDB_rmRepeat.fa</span></code></p>
</section>
<section id="modified-file">
<h3><a class="toc-backref" href="#id4" role="doc-backlink">Modified file</a><a class="headerlink" href="#modified-file" title="Link to this heading">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">-m</span></code> option accept a three column file (with <code class="docutils literal notranslate"><span class="pre">_pseudoU.sites</span></code> suffix, generated by <code class="docutils literal notranslate"><span class="pre">5_ACA_scan.sh</span></code>) like below (genome_location snoRNA_info target_sequence)</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>chr1_10298995_10298996_+<span class="w">        </span>U6-31-ACA65<span class="w">     </span>GCACAUAUACUAAAAYUGGAACGAUACAGAG
chr1_10299004_10299005_+<span class="w">        </span>U6-40-HBI-100/ACA12<span class="w">     </span>CUAAAAUUGGAACGAYACAGAGAAGAUUAGC
chr1_11908193_11908194_+<span class="w">        </span>U5-43-ACA57<span class="w">     </span>AAAUCUUUCGCCUUUYACUAAAGAUUUCCGU
chr1_44731124_44731125_-<span class="w">        </span>U5-46-U85/U89<span class="w">   </span>UCUUUCGCCUUUUACYAAAGAUUUCCGUGGA
chr1_92700930_92700931_-<span class="w">        </span>U4-4-SCARNA27<span class="w">   </span>AGCYUUGCGCAGUGGCAGU
chr1_109591554_109591555_-<span class="w">      </span>U6-86-ACA65<span class="w">     </span>GGAUGACACGCAAAUYCGUGAAGCGUUCCAU
chr1_145465775_145465776_-<span class="w">      </span>U1-5-ACA47<span class="w">      </span>AUACYUACCUGGCAGGGGAG
chr1_148522605_148522606_+<span class="w">      </span>U1-6-ACA47/U109<span class="w"> </span>AUACUYACCUGGCAGGGGAGA
chr11_62841749_62841750_-<span class="w">       </span>U2-60-unknown<span class="w">   </span>CUUAUCAGUUUAAUAYCUGAUACGUCCUCUA
chr11_62841751_62841752_-<span class="w">       </span>U2-58-unknown<span class="w">   </span>UUCUUAUCAGUUUAAYAUCUGAUACGUCCUC
</pre></div>
</div>
</section>
</section>
<section id="output">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">Output</a><a class="headerlink" href="#output" title="Link to this heading">¶</a></h2>
<section id="information">
<h3><a class="toc-backref" href="#id6" role="doc-backlink">Information</a><a class="headerlink" href="#information" title="Link to this heading">¶</a></h3>
<p>Result with <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence.csv</span></code> suffix is the final snoRNA target prediction result, which is a filter result of <code class="docutils literal notranslate"><span class="pre">_pseudoU.csv</span></code> (filter by <code class="docutils literal notranslate"><span class="pre">Modified_score&gt;=15</span> <span class="pre">&amp;</span> <span class="pre">PairNum&gt;=9</span> <span class="pre">&amp;</span> <span class="pre">total_UnPair&lt;=2</span></code>).</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>/the/directory/of/out_file_dir
$<span class="w"> </span>tree<span class="w"> </span>-L<span class="w"> </span><span class="m">1</span>
.
├──<span class="w"> </span>Day0_common_rep1_anno.bed
├──<span class="w"> </span>Day0_common_rep1_anno.bed6
├──<span class="w"> </span>Day0_common_rep1_anno.bed6.append
├──<span class="w"> </span>Day0_common_rep1_anno_info.bed
├──<span class="w"> </span>Day0_common_rep1.bed
├──<span class="w"> </span>Day0_common_rep1_fa.out
├──<span class="w"> </span>Day0_common_rep1.known.target
├──<span class="w"> </span>Day0_common_rep1_pseudoU.csv
├──<span class="w"> </span>Day0_common_rep1_pseudoU_high_confidence.csv
├──<span class="w"> </span>Day0_common_rep1_pseudoU_high_confidence.pdf
├──<span class="w"> </span>Day0_common_rep1_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.pdf
├──<span class="w"> </span>Day0_common_rep1_pseudoU_high_confidence.txt
├──<span class="w"> </span>Day0_common_rep1_pseudoU_high_confidence.xlsx
├──<span class="w"> </span>Day0_common_rep1_pseudoU_Orphan_piechart.pdf
├──<span class="w"> </span>Day0_common_rep1_pseudoU.pdf
├──<span class="w"> </span>Day0_common_rep1_pseudoU.sites
├──<span class="w"> </span>Day0_common_rep1_pseudoU.txt
├──<span class="w"> </span>Day0_common_rep1_pseudoU.xlsx
└──<span class="w"> </span>Day0_common_rep1.unknown.target

<span class="m">0</span><span class="w"> </span>directory,<span class="w"> </span><span class="m">19</span><span class="w"> </span>files
</pre></div>
</div>
</section>
<section id="diagram">
<h3><a class="toc-backref" href="#id7" role="doc-backlink">Diagram</a><a class="headerlink" href="#diagram" title="Link to this heading">¶</a></h3>
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence.pdf</span></code> is a graphical summary of snoRNA-target pair on input Ψ-sites file (<strong>rtsSeeker result+bedAnnotator result</strong>).</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_high_confidence.png" src="../_images/snoRNA_target_prediction_pseudoU_high_confidence.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.pdf</span></code> is a pie chart graphical summary of snoRNA-guided RNA on input Ψ-sites file (<strong>rtsSeeker result+bedAnnotator result</strong>).</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.png" src="../_images/snoRNA_target_prediction_pseudoU_high_confidence_snoRNA_guided_RNA_piechart.png" />
<p>File with suffix <code class="docutils literal notranslate"><span class="pre">_pseudoU_Orphan_piechart.pdf</span></code> is a pie chart graphical summary of orphan snoRNAs predicted with Ψ-modified targets on input Ψ-sites file.</p>
<img alt="../_images/snoRNA_target_prediction_pseudoU_Orphan_piechart.png" src="../_images/snoRNA_target_prediction_pseudoU_Orphan_piechart.png" />
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All user input will be recorded in a plain text file with suffix <code class="docutils literal notranslate"><span class="pre">_ACAscan_config.txt</span></code> in psiFinder/config and help users to easily reload the previous config (by simply clicking <code class="docutils literal notranslate"><span class="pre">CONFIG</span></code> button).</p>
</div>
</section>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality%20Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Transcripts%20Alignment.html">Transcripts Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Identification.html">Ψ-sites Identification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Annotation.html">Ψ-sites Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Distribution.html">Ψ-sites Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Codon%20Preference.html">Ψ-sites Codon Preference</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Ψ-sites snoRNA Target Prediction</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input">Input</a></li>
<li class="toctree-l2"><a class="reference internal" href="#predict-sites-snorna-target">Predict Ψ-sites snoRNA target</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#snorna-sequence">snoRNA sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="#modified-file">Modified file</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#information">Information</a></li>
<li class="toctree-l3"><a class="reference internal" href="#diagram">Diagram</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20miRNA%20Target%20Prediction.html">Ψ-sites miRNA Target Prediction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20and%20snoRNA%20Interaction.html">Ψ-sites and snoRNA Interaction</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Motif%20Searching.html">Ψ-sites Motif Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20SNPs%20and%20SNVs%20Searching.html">Ψ-sites SNPs and SNVs Searching</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Differential%20Modification.html">Ψ-sites Differential Modification</a></li>
<li class="toctree-l1"><a class="reference internal" href="%CE%A8-sites%20Functional%20Enrichment.html">Ψ-sites Functional Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="snakemake%20pipline.html">snakemake pipline</a></li>
<li class="toctree-l1"><a class="reference internal" href="Accessory%20Tools.html">Accessory Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="About%20psiFinder.html">About psiFinder</a></li>
</ul>

        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="Related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="%CE%A8-sites%20miRNA%20Target%20Prediction.html" title="Ψ-sites miRNA Target Prediction"
             >next</a> |</li>
        <li class="right" >
          <a href="%CE%A8-sites%20Codon%20Preference.html" title="Ψ-sites Codon Preference"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">psiFinder v1.0 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Ψ-sites snoRNA Target Prediction</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
    &#169; Copyright 2022, chenzhr23.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 8.2.3.
    </div>
  </body>
</html>